{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "61ed1a24",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.2 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.6      \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.8      \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.10\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.1      \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.1 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2      \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.2 \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f9d9c81d",
   "metadata": {},
   "outputs": [],
   "source": [
    "PROJECT_DIR = Sys.getenv(\"PROJECT_DIR\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9b0fc4a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "source(paste0(PROJECT_DIR, 'code/resolve2xe/LoadResolveBaysor.R'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "d48297cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m15041\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m8\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \",\"\n",
      "\u001b[31mchr\u001b[39m (1): cell\n",
      "\u001b[32mdbl\u001b[39m (7): x, y, n_transcripts, density, elongation, area, avg_confidence\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "An object of class Seurat \n",
       "400 features across 15042 samples within 4 assays \n",
       "Active assay: Xenium (100 features, 0 variable features)\n",
       " 3 other assays present: BlankCodeword, ControlCodeword, ControlProbe\n",
       " 1 spatial field of view present: fov"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "obj = LoadResolveBaysor('/data/petar/fgf1/resolve/xe/32810-1377-slide3_A2-1/')\n",
    "obj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "51b60b91",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_Xenium</th><th scope=col>nFeature_Xenium</th><th scope=col>nCount_BlankCodeword</th><th scope=col>nFeature_BlankCodeword</th><th scope=col>nCount_ControlCodeword</th><th scope=col>nFeature_ControlCodeword</th><th scope=col>nCount_ControlProbe</th><th scope=col>nFeature_ControlProbe</th><th scope=col>density</th><th scope=col>elongation</th><th scope=col>avg_confidence</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>0</th><td>SeuratProject</td><td>818150</td><td>100</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>     NA</td><td>   NA</td><td>    NA</td></tr>\n",
       "\t<tr><th scope=row>Cell10000_6303_6302</th><td>SeuratProject</td><td>   115</td><td> 42</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.04981</td><td>1.178</td><td>0.9905</td></tr>\n",
       "\t<tr><th scope=row>Cell10001_4431_4179</th><td>SeuratProject</td><td>    10</td><td>  7</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.08032</td><td>2.160</td><td>0.9817</td></tr>\n",
       "\t<tr><th scope=row>Cell10002_5865_7477</th><td>SeuratProject</td><td>    34</td><td>  6</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.03700</td><td>7.902</td><td>0.9860</td></tr>\n",
       "\t<tr><th scope=row>Cell10003_7028_1933</th><td>SeuratProject</td><td>    14</td><td> 12</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.08833</td><td>3.136</td><td>0.9893</td></tr>\n",
       "\t<tr><th scope=row>Cell10004_4228_10692</th><td>SeuratProject</td><td>    88</td><td> 32</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.04916</td><td>1.688</td><td>0.9892</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & orig.ident & nCount\\_Xenium & nFeature\\_Xenium & nCount\\_BlankCodeword & nFeature\\_BlankCodeword & nCount\\_ControlCodeword & nFeature\\_ControlCodeword & nCount\\_ControlProbe & nFeature\\_ControlProbe & density & elongation & avg\\_confidence\\\\\n",
       "  & <fct> & <dbl> & <int> & <dbl> & <int> & <dbl> & <int> & <dbl> & <int> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t0 & SeuratProject & 818150 & 100 & 0 & 0 & 0 & 0 & 0 & 0 &      NA &    NA &     NA\\\\\n",
       "\tCell10000\\_6303\\_6302 & SeuratProject &    115 &  42 & 0 & 0 & 0 & 0 & 0 & 0 & 0.04981 & 1.178 & 0.9905\\\\\n",
       "\tCell10001\\_4431\\_4179 & SeuratProject &     10 &   7 & 0 & 0 & 0 & 0 & 0 & 0 & 0.08032 & 2.160 & 0.9817\\\\\n",
       "\tCell10002\\_5865\\_7477 & SeuratProject &     34 &   6 & 0 & 0 & 0 & 0 & 0 & 0 & 0.03700 & 7.902 & 0.9860\\\\\n",
       "\tCell10003\\_7028\\_1933 & SeuratProject &     14 &  12 & 0 & 0 & 0 & 0 & 0 & 0 & 0.08833 & 3.136 & 0.9893\\\\\n",
       "\tCell10004\\_4228\\_10692 & SeuratProject &     88 &  32 & 0 & 0 & 0 & 0 & 0 & 0 & 0.04916 & 1.688 & 0.9892\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | orig.ident &lt;fct&gt; | nCount_Xenium &lt;dbl&gt; | nFeature_Xenium &lt;int&gt; | nCount_BlankCodeword &lt;dbl&gt; | nFeature_BlankCodeword &lt;int&gt; | nCount_ControlCodeword &lt;dbl&gt; | nFeature_ControlCodeword &lt;int&gt; | nCount_ControlProbe &lt;dbl&gt; | nFeature_ControlProbe &lt;int&gt; | density &lt;dbl&gt; | elongation &lt;dbl&gt; | avg_confidence &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0 | SeuratProject | 818150 | 100 | 0 | 0 | 0 | 0 | 0 | 0 |      NA |    NA |     NA |\n",
       "| Cell10000_6303_6302 | SeuratProject |    115 |  42 | 0 | 0 | 0 | 0 | 0 | 0 | 0.04981 | 1.178 | 0.9905 |\n",
       "| Cell10001_4431_4179 | SeuratProject |     10 |   7 | 0 | 0 | 0 | 0 | 0 | 0 | 0.08032 | 2.160 | 0.9817 |\n",
       "| Cell10002_5865_7477 | SeuratProject |     34 |   6 | 0 | 0 | 0 | 0 | 0 | 0 | 0.03700 | 7.902 | 0.9860 |\n",
       "| Cell10003_7028_1933 | SeuratProject |     14 |  12 | 0 | 0 | 0 | 0 | 0 | 0 | 0.08833 | 3.136 | 0.9893 |\n",
       "| Cell10004_4228_10692 | SeuratProject |     88 |  32 | 0 | 0 | 0 | 0 | 0 | 0 | 0.04916 | 1.688 | 0.9892 |\n",
       "\n"
      ],
      "text/plain": [
       "                     orig.ident    nCount_Xenium nFeature_Xenium\n",
       "0                    SeuratProject 818150        100            \n",
       "Cell10000_6303_6302  SeuratProject    115         42            \n",
       "Cell10001_4431_4179  SeuratProject     10          7            \n",
       "Cell10002_5865_7477  SeuratProject     34          6            \n",
       "Cell10003_7028_1933  SeuratProject     14         12            \n",
       "Cell10004_4228_10692 SeuratProject     88         32            \n",
       "                     nCount_BlankCodeword nFeature_BlankCodeword\n",
       "0                    0                    0                     \n",
       "Cell10000_6303_6302  0                    0                     \n",
       "Cell10001_4431_4179  0                    0                     \n",
       "Cell10002_5865_7477  0                    0                     \n",
       "Cell10003_7028_1933  0                    0                     \n",
       "Cell10004_4228_10692 0                    0                     \n",
       "                     nCount_ControlCodeword nFeature_ControlCodeword\n",
       "0                    0                      0                       \n",
       "Cell10000_6303_6302  0                      0                       \n",
       "Cell10001_4431_4179  0                      0                       \n",
       "Cell10002_5865_7477  0                      0                       \n",
       "Cell10003_7028_1933  0                      0                       \n",
       "Cell10004_4228_10692 0                      0                       \n",
       "                     nCount_ControlProbe nFeature_ControlProbe density\n",
       "0                    0                   0                          NA\n",
       "Cell10000_6303_6302  0                   0                     0.04981\n",
       "Cell10001_4431_4179  0                   0                     0.08032\n",
       "Cell10002_5865_7477  0                   0                     0.03700\n",
       "Cell10003_7028_1933  0                   0                     0.08833\n",
       "Cell10004_4228_10692 0                   0                     0.04916\n",
       "                     elongation avg_confidence\n",
       "0                       NA          NA        \n",
       "Cell10000_6303_6302  1.178      0.9905        \n",
       "Cell10001_4431_4179  2.160      0.9817        \n",
       "Cell10002_5865_7477  7.902      0.9860        \n",
       "Cell10003_7028_1933  3.136      0.9893        \n",
       "Cell10004_4228_10692 1.688      0.9892        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "obj %>% `[[` %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "6bbede34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for Seurat-methods {SeuratObject}\"><tr><td>Seurat-methods {SeuratObject}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Seurat Methods</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>Methods for <code>Seurat</code> objects for generics defined in other\n",
       "packages\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "## S3 method for class 'Seurat'\n",
       ".DollarNames(x, pattern = \"\")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "x$i, ...\n",
       "\n",
       "## S3 replacement method for class 'Seurat'\n",
       "x$i, ... &lt;- value\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "x[i, j, ...]\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "x[[i, ..., drop = FALSE]]\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "dim(x)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "dimnames(x)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "head(x, n = 10L, ...)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "merge(\n",
       "  x = NULL,\n",
       "  y = NULL,\n",
       "  add.cell.ids = NULL,\n",
       "  merge.data = TRUE,\n",
       "  merge.dr = NULL,\n",
       "  project = \"SeuratProject\",\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "names(x)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "subset(\n",
       "  x,\n",
       "  subset,\n",
       "  cells = NULL,\n",
       "  features = NULL,\n",
       "  idents = NULL,\n",
       "  return.null = FALSE,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "tail(x, n = 10L, ...)\n",
       "\n",
       "## S4 replacement method for signature 'Seurat,ANY,ANY,ANY'\n",
       "x[[i, j, ...]] &lt;- value\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "colMeans(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "colSums(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "rowMeans(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "rowSums(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "show(object)\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>x, object</code></td>\n",
       "<td>\n",
       "<p>A <code>Seurat</code> object</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>pattern</code></td>\n",
       "<td>\n",
       "<p> A regular expression.  Only matching names are\n",
       "returned.\n",
       "</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>i, features</code></td>\n",
       "<td>\n",
       "<p>Depends on the method\n",
       "</p>\n",
       "\n",
       "<dl>\n",
       "<dt><code>[</code>, <code>subset</code></dt><dd><p>Feature names or indices</p>\n",
       "</dd>\n",
       "<dt><code>$</code>, <code>$&lt;-</code></dt><dd><p>Name of a single metadata column</p>\n",
       "</dd>\n",
       "<dt><code>[[</code>, <code>[[&lt;-</code></dt><dd>\n",
       "<p>Name of one or more metadata columns or an associated object; associated\n",
       "objects include <code>Assay</code>, <code>DimReduc</code>,\n",
       "<code>Graph</code>, <code>SeuratCommand</code>, or\n",
       "<code>SpatialImage</code> objects\n",
       "</p>\n",
       "</dd>\n",
       "</dl>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>...</code></td>\n",
       "<td>\n",
       "<p>Arguments passed to other methods</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>value</code></td>\n",
       "<td>\n",
       "<p>Additional metadata or associated objects to add; <strong>note</strong>:\n",
       "can pass <code>NULL</code> to remove metadata or an associated object</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>j, cells</code></td>\n",
       "<td>\n",
       "<p>Cell names or indices</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>drop</code></td>\n",
       "<td>\n",
       "<p>See <code>drop</code></p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>n</code></td>\n",
       "<td>\n",
       "<p>The number of rows of metadata to return</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>y</code></td>\n",
       "<td>\n",
       "<p>A single <code>Seurat</code> object or a list of <code>Seurat</code> objects</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>add.cell.ids</code></td>\n",
       "<td>\n",
       "<p>A character vector of <code>length(x = c(x, y))</code>;\n",
       "appends the corresponding values to the start of each objects' cell names</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>merge.data</code></td>\n",
       "<td>\n",
       "<p>Merge the data slots instead of just merging the counts\n",
       "(which requires renormalization); this is recommended if the same\n",
       "normalization approach was applied to all objects</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>merge.dr</code></td>\n",
       "<td>\n",
       "<p>Merge specified DimReducs that are present in all objects;\n",
       "will only merge the embeddings slots for the first <code>N</code> dimensions that\n",
       "are shared across all objects.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>project</code></td>\n",
       "<td>\n",
       "<p>Project name for the <code>Seurat</code> object</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset</code></td>\n",
       "<td>\n",
       "<p>Logical expression indicating features/variables to keep</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>idents</code></td>\n",
       "<td>\n",
       "<p>A vector of identity classes to keep</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>return.null</code></td>\n",
       "<td>\n",
       "<p>If no cells are request, return a <code>NULL</code>;\n",
       "by default, throws an error</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>na.rm</code></td>\n",
       "<td>\n",
       "<p>logical.  Should missing values (including <code>NaN</code>)\n",
       "be omitted from the calculations?</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>dims</code></td>\n",
       "<td>\n",
       "<p>completely ignored by the <code>Matrix</code> methods.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>slot</code></td>\n",
       "<td>\n",
       "<p>Name of assay expression matrix to calculate column/row\n",
       "means/sums on</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<p><code>$</code>: metadata column <code>i</code> for object <code>x</code>;\n",
       "<strong>note</strong>: unlike <code>[[</code>, <code>$</code> drops the shape of the metadata\n",
       "to return a vector instead of a data frame\n",
       "</p>\n",
       "<p><code>$&lt;-</code>: object <code>x</code> with metadata <code>value</code> saved as\n",
       "<code>i</code>\n",
       "</p>\n",
       "<p><code>[</code>: object <code>x</code> with features <code>i</code> and cells <code>j</code>\n",
       "</p>\n",
       "<p><code>[[</code>: If <code>i</code> is missing, the metadata data frame; if\n",
       "<code>i</code> is a vector of metadata names, a data frame with the requested\n",
       "metadata, otherwise, the requested associated object\n",
       "</p>\n",
       "<p><code>dim</code>: The number of features (<code>nrow</code>) and cells\n",
       "(<code>ncol</code>) for the default assay; <strong>note</strong>: while the number of\n",
       "features changes depending on the active assay, the number of cells remains\n",
       "the same across all assays\n",
       "</p>\n",
       "<p><code>dimnames</code>: The feature (row) and cell (column) names;\n",
       "<strong>note</strong>: while the features change depending on the active assay, the\n",
       "cell names remain the same across all assays\n",
       "</p>\n",
       "<p><code>head</code>: The first <code>n</code> rows of cell-level metadata\n",
       "</p>\n",
       "<p><code>merge</code>: Merged object\n",
       "</p>\n",
       "<p><code>names</code>: The names of all <code>Assay</code>,\n",
       "<code>DimReduc</code>, <code>Graph</code>, and <code>SpatialImage</code>\n",
       "objects in the <code>Seurat</code> object\n",
       "</p>\n",
       "<p><code>subset</code>: A subsetted <code>Seurat</code> object\n",
       "</p>\n",
       "<p><code>tail</code>: The last <code>n</code> rows of cell-level metadata\n",
       "</p>\n",
       "<p><code>[[&lt;-</code>: <code>x</code> with the metadata or associated objects added\n",
       "as <code>i</code>; if <code>value</code> is <code>NULL</code>, removes metadata or associated\n",
       "object <code>i</code> from object <code>x</code>\n",
       "</p>\n",
       "<p><code>show</code>: Prints summary to <code>stdout</code> and\n",
       "invisibly returns <code>NULL</code>\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Functions</h3>\n",
       "\n",
       "\n",
       "<ul>\n",
       "<li> <p><code>.DollarNames(Seurat)</code>: Autocompletion for <code>$</code> access on a\n",
       "<code>Seurat</code> object\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>$</code>: Metadata access for <code>Seurat</code> objects\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>`$`(Seurat) &lt;- value</code>: Metadata setter for <code>Seurat</code> objects\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>[</code>: Simple subsetter for <code>Seurat</code> objects\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>[[</code>: Metadata and associated object accessor\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>dim(Seurat)</code>: Number of cells and features for the active assay\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>dimnames(Seurat)</code>: The cell and feature names for the active assay\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>head(Seurat)</code>: Get the first rows of cell-level metadata\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>merge(Seurat)</code>: Merge two or more <code>Seurat</code> objects together\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>names(Seurat)</code>: Common associated objects\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>subset(Seurat)</code>: Subset a <code>Seurat</code> object\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>tail(Seurat)</code>: Get the last rows of cell-level metadata\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>`[[`(x = Seurat, i = ANY, j = ANY) &lt;- value</code>: Add cell-level metadata or associated objects\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>colMeans(Seurat)</code>: Calculate <code>colMeans</code> on a\n",
       "<code>Seurat</code> object\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>colSums(Seurat)</code>: Calculate <code>colSums</code> on a\n",
       "<code>Seurat</code> object\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>rowMeans(Seurat)</code>: Calculate <code>rowMeans</code> on a\n",
       "<code>rowMeans</code> object\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>rowSums(Seurat)</code>: Calculate <code>rowSums</code> on a\n",
       "<code>Seurat</code> object\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>show(Seurat)</code>: Overview of a <code>Seurat</code> object\n",
       "</p>\n",
       "</li></ul>\n",
       "\n",
       "\n",
       "<h3>Merge Details</h3>\n",
       "\n",
       "<p>When merging Seurat objects, the merge procedure will merge the Assay level\n",
       "counts and potentially the data slots (depending on the merge.data parameter).\n",
       "It will also merge the cell-level meta data that was stored with each object\n",
       "and preserve the cell identities that were active in the objects pre-merge.\n",
       "The merge will optionally merge reductions depending on the values passed to\n",
       "<code>merge.dr</code> if they have the same name across objects. Here the\n",
       "embeddings slots will be merged and if there are differing numbers of\n",
       "dimensions across objects, only the first N shared dimensions will be merged.\n",
       "The feature loadings slots will be filled by the values present in the first\n",
       "object.The merge will not preserve graphs, logged commands, or feature-level\n",
       "metadata that were present in the original objects. If add.cell.ids isn't\n",
       "specified and any cell names are duplicated, cell names will be appended\n",
       "with _X, where X is the numeric index of the object in c(x, y).\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>See Also</h3>\n",
       "\n",
       "<p><code>subset</code> <code>WhichCells</code>\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "# Get metadata using `$'\n",
       "head(pbmc_small$groups)\n",
       "\n",
       "# Add metadata using the `$' operator\n",
       "set.seed(42)\n",
       "pbmc_small$value &lt;- sample(1:3, size = ncol(pbmc_small), replace = TRUE)\n",
       "head(pbmc_small[[\"value\"]])\n",
       "\n",
       "# `[' examples\n",
       "pbmc_small[VariableFeatures(object = pbmc_small), ]\n",
       "pbmc_small[, 1:10]\n",
       "\n",
       "# Get the cell-level metadata data frame\n",
       "head(pbmc_small[[]])\n",
       "\n",
       "# Pull specific metadata information\n",
       "head(pbmc_small[[c(\"letter.idents\", \"groups\")]])\n",
       "head(pbmc_small[[\"groups\", drop = TRUE]])\n",
       "\n",
       "# Get a sub-object (eg. an `Assay' or `DimReduc')\n",
       "pbmc_small[[\"RNA\"]]\n",
       "pbmc_small[[\"pca\"]]\n",
       "\n",
       "# Get the number of features in an object\n",
       "nrow(pbmc_small)\n",
       "\n",
       "# Get the number of cells in an object\n",
       "ncol(pbmc_small)\n",
       "\n",
       "# Get the feature names of an object\n",
       "rownames(pbmc_small)\n",
       "\n",
       "# Get the cell names of an object\n",
       "colnames(pbmc_small)\n",
       "\n",
       "# Get the first 10 rows of cell-level metadata\n",
       "head(pbmc_small)\n",
       "\n",
       "# `merge' examples\n",
       "# merge two objects\n",
       "merge(pbmc_small, y = pbmc_small)\n",
       "# to merge more than two objects, pass one to x and a list of objects to y\n",
       "merge(pbmc_small, y = c(pbmc_small, pbmc_small))\n",
       "\n",
       "names(pbmc_small)\n",
       "\n",
       "# `subset' examples\n",
       "subset(pbmc_small, subset = MS4A1 &gt; 4)\n",
       "subset(pbmc_small, subset = `DLGAP1-AS1` &gt; 2)\n",
       "subset(pbmc_small, idents = '0', invert = TRUE)\n",
       "subset(pbmc_small, subset = MS4A1 &gt; 3, slot = 'counts')\n",
       "subset(pbmc_small, features = VariableFeatures(object = pbmc_small))\n",
       "\n",
       "# Get the last 10 rows of cell-level metadata\n",
       "tail(pbmc_small)\n",
       "\n",
       "head(colMeans(pbmc_small))\n",
       "\n",
       "head(colSums(pbmc_small))\n",
       "\n",
       "head(rowMeans(pbmc_small))\n",
       "\n",
       "head(rowSums(pbmc_small))\n",
       "\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>SeuratObject</em> version 4.1.3 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{Seurat-methods}{Seurat Methods}{Seurat.Rdash.methods}\n",
       "\\aliasA{.DollarNames.Seurat}{Seurat-methods}{.DollarNames.Seurat}\n",
       "\\aliasA{[.Seurat}{Seurat-methods}{[.Seurat}\n",
       "\\aliasA{[[.Seurat}{Seurat-methods}{[[.Seurat}\n",
       "\\aliasA{[[<\\Rdash{},Seurat,ANY,ANY,ANY-method}{Seurat-methods}{[[<.Rdash.,Seurat,ANY,ANY,ANY.Rdash.method}\n",
       "\\aliasA{\\$.Seurat}{Seurat-methods}{.Rdol..Seurat}\n",
       "\\aliasA{\\$<\\Rdash{}.Seurat}{Seurat-methods}{.Rdol.<.Rdash..Seurat}\n",
       "\\aliasA{AddSamples}{Seurat-methods}{AddSamples}\n",
       "\\aliasA{colMeans,Seurat-method}{Seurat-methods}{colMeans,Seurat.Rdash.method}\n",
       "\\aliasA{colSums,Seurat-method}{Seurat-methods}{colSums,Seurat.Rdash.method}\n",
       "\\aliasA{dim.Seurat}{Seurat-methods}{dim.Seurat}\n",
       "\\aliasA{dimnames.Seurat}{Seurat-methods}{dimnames.Seurat}\n",
       "\\aliasA{head.Seurat}{Seurat-methods}{head.Seurat}\n",
       "\\aliasA{merge}{Seurat-methods}{merge}\n",
       "\\methaliasA{merge.Seurat}{Seurat-methods}{merge.Seurat}\n",
       "\\aliasA{MergeSeurat}{Seurat-methods}{MergeSeurat}\n",
       "\\aliasA{names.Seurat}{Seurat-methods}{names.Seurat}\n",
       "\\aliasA{rowMeans,Seurat-method}{Seurat-methods}{rowMeans,Seurat.Rdash.method}\n",
       "\\aliasA{rowSums,Seurat-method}{Seurat-methods}{rowSums,Seurat.Rdash.method}\n",
       "\\aliasA{show,Seurat-method}{Seurat-methods}{show,Seurat.Rdash.method}\n",
       "\\aliasA{subset}{Seurat-methods}{subset}\n",
       "\\methaliasA{subset.Seurat}{Seurat-methods}{subset.Seurat}\n",
       "\\aliasA{tail.Seurat}{Seurat-methods}{tail.Seurat}\n",
       "\\keyword{seurat}{Seurat-methods}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "Methods for \\code{\\LinkA{Seurat}{Seurat}} objects for generics defined in other\n",
       "packages\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "## S3 method for class 'Seurat'\n",
       ".DollarNames(x, pattern = \"\")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "x$i, ...\n",
       "\n",
       "## S3 replacement method for class 'Seurat'\n",
       "x$i, ... <- value\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "x[i, j, ...]\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "x[[i, ..., drop = FALSE]]\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "dim(x)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "dimnames(x)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "head(x, n = 10L, ...)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "merge(\n",
       "  x = NULL,\n",
       "  y = NULL,\n",
       "  add.cell.ids = NULL,\n",
       "  merge.data = TRUE,\n",
       "  merge.dr = NULL,\n",
       "  project = \"SeuratProject\",\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "names(x)\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "subset(\n",
       "  x,\n",
       "  subset,\n",
       "  cells = NULL,\n",
       "  features = NULL,\n",
       "  idents = NULL,\n",
       "  return.null = FALSE,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "tail(x, n = 10L, ...)\n",
       "\n",
       "## S4 replacement method for signature 'Seurat,ANY,ANY,ANY'\n",
       "x[[i, j, ...]] <- value\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "colMeans(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "colSums(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "rowMeans(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "rowSums(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "\n",
       "## S4 method for signature 'Seurat'\n",
       "show(object)\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{x, object}] A \\code{\\LinkA{Seurat}{Seurat}} object\n",
       "\n",
       "\\item[\\code{pattern}]  A regular expression.  Only matching names are\n",
       "returned.\n",
       "\n",
       "\n",
       "\\item[\\code{i, features}] Depends on the method\n",
       "\\begin{description}\n",
       "\n",
       "\\item[\\code{[}, \\code{subset}] Feature names or indices\n",
       "\\item[\\code{\\$}, \\code{\\$<-}] Name of a single metadata column\n",
       "\\item[\\code{[[}, \\code{[[<-}] \n",
       "Name of one or more metadata columns or an associated object; associated\n",
       "objects include \\code{\\LinkA{Assay}{Assay}}, \\code{\\LinkA{DimReduc}{DimReduc}},\n",
       "\\code{\\LinkA{Graph}{Graph}}, \\code{\\LinkA{SeuratCommand}{SeuratCommand}}, or\n",
       "\\code{\\LinkA{SpatialImage}{SpatialImage}} objects\n",
       "\n",
       "\n",
       "\\end{description}\n",
       "\n",
       "\n",
       "\\item[\\code{...}] Arguments passed to other methods\n",
       "\n",
       "\\item[\\code{value}] Additional metadata or associated objects to add; \\strong{note}:\n",
       "can pass \\code{NULL} to remove metadata or an associated object\n",
       "\n",
       "\\item[\\code{j, cells}] Cell names or indices\n",
       "\n",
       "\\item[\\code{drop}] See \\code{\\LinkA{drop}{drop}}\n",
       "\n",
       "\\item[\\code{n}] The number of rows of metadata to return\n",
       "\n",
       "\\item[\\code{y}] A single \\code{Seurat} object or a list of \\code{Seurat} objects\n",
       "\n",
       "\\item[\\code{add.cell.ids}] A character vector of \\code{length(x = c(x, y))};\n",
       "appends the corresponding values to the start of each objects' cell names\n",
       "\n",
       "\\item[\\code{merge.data}] Merge the data slots instead of just merging the counts\n",
       "(which requires renormalization); this is recommended if the same\n",
       "normalization approach was applied to all objects\n",
       "\n",
       "\\item[\\code{merge.dr}] Merge specified DimReducs that are present in all objects;\n",
       "will only merge the embeddings slots for the first \\code{N} dimensions that\n",
       "are shared across all objects.\n",
       "\n",
       "\\item[\\code{project}] \\LinkA{Project}{Project} name for the \\code{Seurat} object\n",
       "\n",
       "\\item[\\code{subset}] Logical expression indicating features/variables to keep\n",
       "\n",
       "\\item[\\code{idents}] A vector of identity classes to keep\n",
       "\n",
       "\\item[\\code{return.null}] If no cells are request, return a \\code{NULL};\n",
       "by default, throws an error\n",
       "\n",
       "\\item[\\code{na.rm}] logical.  Should missing values (including \\code{NaN})\n",
       "be omitted from the calculations?\n",
       "\n",
       "\\item[\\code{dims}] completely ignored by the \\code{Matrix} methods.\n",
       "\n",
       "\\item[\\code{slot}] Name of assay expression matrix to calculate column/row\n",
       "means/sums on\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Value}\n",
       "\\code{\\$}: metadata column \\code{i} for object \\code{x};\n",
       "\\strong{note}: unlike \\code{[[}, \\code{\\$} drops the shape of the metadata\n",
       "to return a vector instead of a data frame\n",
       "\n",
       "\\code{\\$<-}: object \\code{x} with metadata \\code{value} saved as\n",
       "\\code{i}\n",
       "\n",
       "\\code{[}: object \\code{x} with features \\code{i} and cells \\code{j}\n",
       "\n",
       "\\code{[[}: If \\code{i} is missing, the metadata data frame; if\n",
       "\\code{i} is a vector of metadata names, a data frame with the requested\n",
       "metadata, otherwise, the requested associated object\n",
       "\n",
       "\\code{dim}: The number of features (\\code{nrow}) and cells\n",
       "(\\code{ncol}) for the default assay; \\strong{note}: while the number of\n",
       "features changes depending on the active assay, the number of cells remains\n",
       "the same across all assays\n",
       "\n",
       "\\code{dimnames}: The feature (row) and cell (column) names;\n",
       "\\strong{note}: while the features change depending on the active assay, the\n",
       "cell names remain the same across all assays\n",
       "\n",
       "\\code{head}: The first \\code{n} rows of cell-level metadata\n",
       "\n",
       "\\code{merge}: Merged object\n",
       "\n",
       "\\code{names}: The names of all \\code{\\LinkA{Assay}{Assay}},\n",
       "\\code{\\LinkA{DimReduc}{DimReduc}}, \\code{\\LinkA{Graph}{Graph}}, and \\code{\\LinkA{SpatialImage}{SpatialImage}}\n",
       "objects in the \\code{Seurat} object\n",
       "\n",
       "\\code{subset}: A subsetted \\code{Seurat} object\n",
       "\n",
       "\\code{tail}: The last \\code{n} rows of cell-level metadata\n",
       "\n",
       "\\code{[[<-}: \\code{x} with the metadata or associated objects added\n",
       "as \\code{i}; if \\code{value} is \\code{NULL}, removes metadata or associated\n",
       "object \\code{i} from object \\code{x}\n",
       "\n",
       "\\code{show}: Prints summary to \\code{\\LinkA{stdout}{stdout}} and\n",
       "invisibly returns \\code{NULL}\n",
       "\\end{Value}\n",
       "%\n",
       "\\begin{Section}{Functions}\n",
       "\\begin{itemize}\n",
       "\n",
       "\\item{} \\code{.DollarNames(Seurat)}: Autocompletion for \\code{\\$} access on a\n",
       "\\code{Seurat} object\n",
       "\n",
       "\\item{} \\code{\\$}: Metadata access for \\code{Seurat} objects\n",
       "\n",
       "\\item{} \\code{`\\$`(Seurat) <- value}: Metadata setter for \\code{Seurat} objects\n",
       "\n",
       "\\item{} \\code{[}: Simple subsetter for \\code{Seurat} objects\n",
       "\n",
       "\\item{} \\code{[[}: Metadata and associated object accessor\n",
       "\n",
       "\\item{} \\code{dim(Seurat)}: Number of cells and features for the active assay\n",
       "\n",
       "\\item{} \\code{dimnames(Seurat)}: The cell and feature names for the active assay\n",
       "\n",
       "\\item{} \\code{head(Seurat)}: Get the first rows of cell-level metadata\n",
       "\n",
       "\\item{} \\code{merge(Seurat)}: Merge two or more \\code{Seurat} objects together\n",
       "\n",
       "\\item{} \\code{names(Seurat)}: Common associated objects\n",
       "\n",
       "\\item{} \\code{subset(Seurat)}: Subset a \\code{\\LinkA{Seurat}{Seurat}} object\n",
       "\n",
       "\\item{} \\code{tail(Seurat)}: Get the last rows of cell-level metadata\n",
       "\n",
       "\\item{} \\code{`[[`(x = Seurat, i = ANY, j = ANY) <- value}: Add cell-level metadata or associated objects\n",
       "\n",
       "\\item{} \\code{colMeans(Seurat)}: Calculate \\code{\\LinkA{colMeans}{colMeans}} on a\n",
       "\\code{Seurat} object\n",
       "\n",
       "\\item{} \\code{colSums(Seurat)}: Calculate \\code{\\LinkA{colSums}{colSums}} on a\n",
       "\\code{Seurat} object\n",
       "\n",
       "\\item{} \\code{rowMeans(Seurat)}: Calculate \\code{\\LinkA{rowMeans}{rowMeans}} on a\n",
       "\\code{rowMeans} object\n",
       "\n",
       "\\item{} \\code{rowSums(Seurat)}: Calculate \\code{\\LinkA{rowSums}{rowSums}} on a\n",
       "\\code{Seurat} object\n",
       "\n",
       "\\item{} \\code{show(Seurat)}: Overview of a \\code{Seurat} object\n",
       "\n",
       "\n",
       "\\end{itemize}\n",
       "\\end{Section}\n",
       "%\n",
       "\\begin{Section}{Merge Details}\n",
       "\n",
       "When merging Seurat objects, the merge procedure will merge the Assay level\n",
       "counts and potentially the data slots (depending on the merge.data parameter).\n",
       "It will also merge the cell-level meta data that was stored with each object\n",
       "and preserve the cell identities that were active in the objects pre-merge.\n",
       "The merge will optionally merge reductions depending on the values passed to\n",
       "\\code{merge.dr} if they have the same name across objects. Here the\n",
       "embeddings slots will be merged and if there are differing numbers of\n",
       "dimensions across objects, only the first N shared dimensions will be merged.\n",
       "The feature loadings slots will be filled by the values present in the first\n",
       "object.The merge will not preserve graphs, logged commands, or feature-level\n",
       "metadata that were present in the original objects. If add.cell.ids isn't\n",
       "specified and any cell names are duplicated, cell names will be appended\n",
       "with \\_X, where X is the numeric index of the object in c(x, y).\n",
       "\\end{Section}\n",
       "%\n",
       "\\begin{SeeAlso}\\relax\n",
       "\\code{\\LinkA{subset}{subset}} \\code{\\LinkA{WhichCells}{WhichCells}}\n",
       "\\end{SeeAlso}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "# Get metadata using `$'\n",
       "head(pbmc_small$groups)\n",
       "\n",
       "# Add metadata using the `$' operator\n",
       "set.seed(42)\n",
       "pbmc_small$value <- sample(1:3, size = ncol(pbmc_small), replace = TRUE)\n",
       "head(pbmc_small[[\"value\"]])\n",
       "\n",
       "# `[' examples\n",
       "pbmc_small[VariableFeatures(object = pbmc_small), ]\n",
       "pbmc_small[, 1:10]\n",
       "\n",
       "# Get the cell-level metadata data frame\n",
       "head(pbmc_small[[]])\n",
       "\n",
       "# Pull specific metadata information\n",
       "head(pbmc_small[[c(\"letter.idents\", \"groups\")]])\n",
       "head(pbmc_small[[\"groups\", drop = TRUE]])\n",
       "\n",
       "# Get a sub-object (eg. an `Assay' or `DimReduc')\n",
       "pbmc_small[[\"RNA\"]]\n",
       "pbmc_small[[\"pca\"]]\n",
       "\n",
       "# Get the number of features in an object\n",
       "nrow(pbmc_small)\n",
       "\n",
       "# Get the number of cells in an object\n",
       "ncol(pbmc_small)\n",
       "\n",
       "# Get the feature names of an object\n",
       "rownames(pbmc_small)\n",
       "\n",
       "# Get the cell names of an object\n",
       "colnames(pbmc_small)\n",
       "\n",
       "# Get the first 10 rows of cell-level metadata\n",
       "head(pbmc_small)\n",
       "\n",
       "# `merge' examples\n",
       "# merge two objects\n",
       "merge(pbmc_small, y = pbmc_small)\n",
       "# to merge more than two objects, pass one to x and a list of objects to y\n",
       "merge(pbmc_small, y = c(pbmc_small, pbmc_small))\n",
       "\n",
       "names(pbmc_small)\n",
       "\n",
       "# `subset' examples\n",
       "subset(pbmc_small, subset = MS4A1 > 4)\n",
       "subset(pbmc_small, subset = `DLGAP1-AS1` > 2)\n",
       "subset(pbmc_small, idents = '0', invert = TRUE)\n",
       "subset(pbmc_small, subset = MS4A1 > 3, slot = 'counts')\n",
       "subset(pbmc_small, features = VariableFeatures(object = pbmc_small))\n",
       "\n",
       "# Get the last 10 rows of cell-level metadata\n",
       "tail(pbmc_small)\n",
       "\n",
       "head(colMeans(pbmc_small))\n",
       "\n",
       "head(colSums(pbmc_small))\n",
       "\n",
       "head(rowMeans(pbmc_small))\n",
       "\n",
       "head(rowSums(pbmc_small))\n",
       "\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "Seurat-methods          package:SeuratObject           R Documentation\n",
       "\n",
       "_\bS_\be_\bu_\br_\ba_\bt _\bM_\be_\bt_\bh_\bo_\bd_\bs\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     Methods for ‘Seurat’ objects for generics defined in other\n",
       "     packages\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     ## S3 method for class 'Seurat'\n",
       "     .DollarNames(x, pattern = \"\")\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     x$i, ...\n",
       "     \n",
       "     ## S3 replacement method for class 'Seurat'\n",
       "     x$i, ... <- value\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     x[i, j, ...]\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     x[[i, ..., drop = FALSE]]\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     dim(x)\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     dimnames(x)\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     head(x, n = 10L, ...)\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     merge(\n",
       "       x = NULL,\n",
       "       y = NULL,\n",
       "       add.cell.ids = NULL,\n",
       "       merge.data = TRUE,\n",
       "       merge.dr = NULL,\n",
       "       project = \"SeuratProject\",\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     names(x)\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     subset(\n",
       "       x,\n",
       "       subset,\n",
       "       cells = NULL,\n",
       "       features = NULL,\n",
       "       idents = NULL,\n",
       "       return.null = FALSE,\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     tail(x, n = 10L, ...)\n",
       "     \n",
       "     ## S4 replacement method for signature 'Seurat,ANY,ANY,ANY'\n",
       "     x[[i, j, ...]] <- value\n",
       "     \n",
       "     ## S4 method for signature 'Seurat'\n",
       "     colMeans(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "     \n",
       "     ## S4 method for signature 'Seurat'\n",
       "     colSums(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "     \n",
       "     ## S4 method for signature 'Seurat'\n",
       "     rowMeans(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "     \n",
       "     ## S4 method for signature 'Seurat'\n",
       "     rowSums(x, na.rm = FALSE, dims = 1, ..., slot = \"data\")\n",
       "     \n",
       "     ## S4 method for signature 'Seurat'\n",
       "     show(object)\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "x, object: A ‘Seurat’ object\n",
       "\n",
       " pattern: A regular expression.  Only matching names are returned.\n",
       "\n",
       "i, features: Depends on the method\n",
       "\n",
       "          ‘[’, ‘subset’ Feature names or indices\n",
       "\n",
       "          ‘$’, ‘$<-’ Name of a single metadata column\n",
       "\n",
       "          ‘[[’, ‘[[<-’ Name of one or more metadata columns or an\n",
       "              associated object; associated objects include ‘Assay’,\n",
       "              ‘DimReduc’, ‘Graph’, ‘SeuratCommand’, or ‘SpatialImage’\n",
       "              objects\n",
       "\n",
       "     ...: Arguments passed to other methods\n",
       "\n",
       "   value: Additional metadata or associated objects to add; *note*: can\n",
       "          pass ‘NULL’ to remove metadata or an associated object\n",
       "\n",
       "j, cells: Cell names or indices\n",
       "\n",
       "    drop: See ‘drop’\n",
       "\n",
       "       n: The number of rows of metadata to return\n",
       "\n",
       "       y: A single ‘Seurat’ object or a list of ‘Seurat’ objects\n",
       "\n",
       "add.cell.ids: A character vector of ‘length(x = c(x, y))’; appends the\n",
       "          corresponding values to the start of each objects' cell names\n",
       "\n",
       "merge.data: Merge the data slots instead of just merging the counts\n",
       "          (which requires renormalization); this is recommended if the\n",
       "          same normalization approach was applied to all objects\n",
       "\n",
       "merge.dr: Merge specified DimReducs that are present in all objects;\n",
       "          will only merge the embeddings slots for the first ‘N’\n",
       "          dimensions that are shared across all objects.\n",
       "\n",
       " project: Project name for the ‘Seurat’ object\n",
       "\n",
       "  subset: Logical expression indicating features/variables to keep\n",
       "\n",
       "  idents: A vector of identity classes to keep\n",
       "\n",
       "return.null: If no cells are request, return a ‘NULL’; by default,\n",
       "          throws an error\n",
       "\n",
       "   na.rm: logical.  Should missing values (including ‘NaN’) be omitted\n",
       "          from the calculations?\n",
       "\n",
       "    dims: completely ignored by the ‘Matrix’ methods.\n",
       "\n",
       "    slot: Name of assay expression matrix to calculate column/row\n",
       "          means/sums on\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     ‘$’: metadata column ‘i’ for object ‘x’; *note*: unlike ‘[[’, ‘$’\n",
       "     drops the shape of the metadata to return a vector instead of a\n",
       "     data frame\n",
       "\n",
       "     ‘$<-’: object ‘x’ with metadata ‘value’ saved as ‘i’\n",
       "\n",
       "     ‘[’: object ‘x’ with features ‘i’ and cells ‘j’\n",
       "\n",
       "     ‘[[’: If ‘i’ is missing, the metadata data frame; if ‘i’ is a\n",
       "     vector of metadata names, a data frame with the requested\n",
       "     metadata, otherwise, the requested associated object\n",
       "\n",
       "     ‘dim’: The number of features (‘nrow’) and cells (‘ncol’) for the\n",
       "     default assay; *note*: while the number of features changes\n",
       "     depending on the active assay, the number of cells remains the\n",
       "     same across all assays\n",
       "\n",
       "     ‘dimnames’: The feature (row) and cell (column) names; *note*:\n",
       "     while the features change depending on the active assay, the cell\n",
       "     names remain the same across all assays\n",
       "\n",
       "     ‘head’: The first ‘n’ rows of cell-level metadata\n",
       "\n",
       "     ‘merge’: Merged object\n",
       "\n",
       "     ‘names’: The names of all ‘Assay’, ‘DimReduc’, ‘Graph’, and\n",
       "     ‘SpatialImage’ objects in the ‘Seurat’ object\n",
       "\n",
       "     ‘subset’: A subsetted ‘Seurat’ object\n",
       "\n",
       "     ‘tail’: The last ‘n’ rows of cell-level metadata\n",
       "\n",
       "     ‘[[<-’: ‘x’ with the metadata or associated objects added as ‘i’;\n",
       "     if ‘value’ is ‘NULL’, removes metadata or associated object ‘i’\n",
       "     from object ‘x’\n",
       "\n",
       "     ‘show’: Prints summary to ‘stdout’ and invisibly returns ‘NULL’\n",
       "\n",
       "_\bF_\bu_\bn_\bc_\bt_\bi_\bo_\bn_\bs:\n",
       "\n",
       "        • ‘.DollarNames(Seurat)’: Autocompletion for ‘$’ access on a\n",
       "          ‘Seurat’ object\n",
       "\n",
       "        • ‘$’: Metadata access for ‘Seurat’ objects\n",
       "\n",
       "        • ‘`$`(Seurat) <- value’: Metadata setter for ‘Seurat’ objects\n",
       "\n",
       "        • ‘[’: Simple subsetter for ‘Seurat’ objects\n",
       "\n",
       "        • ‘[[’: Metadata and associated object accessor\n",
       "\n",
       "        • ‘dim(Seurat)’: Number of cells and features for the active\n",
       "          assay\n",
       "\n",
       "        • ‘dimnames(Seurat)’: The cell and feature names for the active\n",
       "          assay\n",
       "\n",
       "        • ‘head(Seurat)’: Get the first rows of cell-level metadata\n",
       "\n",
       "        • ‘merge(Seurat)’: Merge two or more ‘Seurat’ objects together\n",
       "\n",
       "        • ‘names(Seurat)’: Common associated objects\n",
       "\n",
       "        • ‘subset(Seurat)’: Subset a ‘Seurat’ object\n",
       "\n",
       "        • ‘tail(Seurat)’: Get the last rows of cell-level metadata\n",
       "\n",
       "        • ‘`[[`(x = Seurat, i = ANY, j = ANY) <- value’: Add cell-level\n",
       "          metadata or associated objects\n",
       "\n",
       "        • ‘colMeans(Seurat)’: Calculate ‘colMeans’ on a ‘Seurat’ object\n",
       "\n",
       "        • ‘colSums(Seurat)’: Calculate ‘colSums’ on a ‘Seurat’ object\n",
       "\n",
       "        • ‘rowMeans(Seurat)’: Calculate ‘rowMeans’ on a ‘rowMeans’\n",
       "          object\n",
       "\n",
       "        • ‘rowSums(Seurat)’: Calculate ‘rowSums’ on a ‘Seurat’ object\n",
       "\n",
       "        • ‘show(Seurat)’: Overview of a ‘Seurat’ object\n",
       "\n",
       "_\bM_\be_\br_\bg_\be _\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     When merging Seurat objects, the merge procedure will merge the\n",
       "     Assay level counts and potentially the data slots (depending on\n",
       "     the merge.data parameter). It will also merge the cell-level meta\n",
       "     data that was stored with each object and preserve the cell\n",
       "     identities that were active in the objects pre-merge. The merge\n",
       "     will optionally merge reductions depending on the values passed to\n",
       "     ‘merge.dr’ if they have the same name across objects. Here the\n",
       "     embeddings slots will be merged and if there are differing numbers\n",
       "     of dimensions across objects, only the first N shared dimensions\n",
       "     will be merged. The feature loadings slots will be filled by the\n",
       "     values present in the first object.The merge will not preserve\n",
       "     graphs, logged commands, or feature-level metadata that were\n",
       "     present in the original objects. If add.cell.ids isn't specified\n",
       "     and any cell names are duplicated, cell names will be appended\n",
       "     with _X, where X is the numeric index of the object in c(x, y).\n",
       "\n",
       "_\bS_\be_\be _\bA_\bl_\bs_\bo:\n",
       "\n",
       "     ‘subset’ ‘WhichCells’\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     # Get metadata using `$'\n",
       "     head(pbmc_small$groups)\n",
       "     \n",
       "     # Add metadata using the `$' operator\n",
       "     set.seed(42)\n",
       "     pbmc_small$value <- sample(1:3, size = ncol(pbmc_small), replace = TRUE)\n",
       "     head(pbmc_small[[\"value\"]])\n",
       "     \n",
       "     # `[' examples\n",
       "     pbmc_small[VariableFeatures(object = pbmc_small), ]\n",
       "     pbmc_small[, 1:10]\n",
       "     \n",
       "     # Get the cell-level metadata data frame\n",
       "     head(pbmc_small[[]])\n",
       "     \n",
       "     # Pull specific metadata information\n",
       "     head(pbmc_small[[c(\"letter.idents\", \"groups\")]])\n",
       "     head(pbmc_small[[\"groups\", drop = TRUE]])\n",
       "     \n",
       "     # Get a sub-object (eg. an `Assay' or `DimReduc')\n",
       "     pbmc_small[[\"RNA\"]]\n",
       "     pbmc_small[[\"pca\"]]\n",
       "     \n",
       "     # Get the number of features in an object\n",
       "     nrow(pbmc_small)\n",
       "     \n",
       "     # Get the number of cells in an object\n",
       "     ncol(pbmc_small)\n",
       "     \n",
       "     # Get the feature names of an object\n",
       "     rownames(pbmc_small)\n",
       "     \n",
       "     # Get the cell names of an object\n",
       "     colnames(pbmc_small)\n",
       "     \n",
       "     # Get the first 10 rows of cell-level metadata\n",
       "     head(pbmc_small)\n",
       "     \n",
       "     # `merge' examples\n",
       "     # merge two objects\n",
       "     merge(pbmc_small, y = pbmc_small)\n",
       "     # to merge more than two objects, pass one to x and a list of objects to y\n",
       "     merge(pbmc_small, y = c(pbmc_small, pbmc_small))\n",
       "     \n",
       "     names(pbmc_small)\n",
       "     \n",
       "     # `subset' examples\n",
       "     subset(pbmc_small, subset = MS4A1 > 4)\n",
       "     subset(pbmc_small, subset = `DLGAP1-AS1` > 2)\n",
       "     subset(pbmc_small, idents = '0', invert = TRUE)\n",
       "     subset(pbmc_small, subset = MS4A1 > 3, slot = 'counts')\n",
       "     subset(pbmc_small, features = VariableFeatures(object = pbmc_small))\n",
       "     \n",
       "     # Get the last 10 rows of cell-level metadata\n",
       "     tail(pbmc_small)\n",
       "     \n",
       "     head(colMeans(pbmc_small))\n",
       "     \n",
       "     head(colSums(pbmc_small))\n",
       "     \n",
       "     head(rowMeans(pbmc_small))\n",
       "     \n",
       "     head(rowSums(pbmc_small))\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?SeuratObject::merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "fea2c57e",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in parse(text = x, srcfile = src): <text>:1:106: unexpected symbol\n1: baysor_cell_stats = read_csv('/data/petar/fgf1/resolve/xe/32810-1377-slide3_A2-1/baysor_cell_stats.csv') magrittr\n                                                                                                             ^\n",
     "output_type": "error",
     "traceback": [
      "Error in parse(text = x, srcfile = src): <text>:1:106: unexpected symbol\n1: baysor_cell_stats = read_csv('/data/petar/fgf1/resolve/xe/32810-1377-slide3_A2-1/baysor_cell_stats.csv') magrittr\n                                                                                                             ^\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "baysor_cell_stats = read_csv('/data/petar/fgf1/resolve/xe/32810-1377-slide3_A2-1/baysor_cell_stats.csv') magrittr::`%>%`\n",
    "column_to_rownames(\"cell\") magrittr::`%>%`\n",
    "select(density, elongation, avg_confidence)\n",
    "\n",
    "baysor_cell_stats %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "b90b33c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "obj = Seurat::AddMetaData(obj, metadata = baysor_cell_stats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "32d4cb1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_Xenium</th><th scope=col>nFeature_Xenium</th><th scope=col>nCount_BlankCodeword</th><th scope=col>nFeature_BlankCodeword</th><th scope=col>nCount_ControlCodeword</th><th scope=col>nFeature_ControlCodeword</th><th scope=col>nCount_ControlProbe</th><th scope=col>nFeature_ControlProbe</th><th scope=col>density</th><th scope=col>elongation</th><th scope=col>avg_confidence</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>0</th><td>SeuratProject</td><td>818150</td><td>100</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>     NA</td><td>   NA</td><td>    NA</td></tr>\n",
       "\t<tr><th scope=row>Cell10000_6303_6302</th><td>SeuratProject</td><td>   115</td><td> 42</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.04981</td><td>1.178</td><td>0.9905</td></tr>\n",
       "\t<tr><th scope=row>Cell10001_4431_4179</th><td>SeuratProject</td><td>    10</td><td>  7</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.08032</td><td>2.160</td><td>0.9817</td></tr>\n",
       "\t<tr><th scope=row>Cell10002_5865_7477</th><td>SeuratProject</td><td>    34</td><td>  6</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.03700</td><td>7.902</td><td>0.9860</td></tr>\n",
       "\t<tr><th scope=row>Cell10003_7028_1933</th><td>SeuratProject</td><td>    14</td><td> 12</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.08833</td><td>3.136</td><td>0.9893</td></tr>\n",
       "\t<tr><th scope=row>Cell10004_4228_10692</th><td>SeuratProject</td><td>    88</td><td> 32</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.04916</td><td>1.688</td><td>0.9892</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & orig.ident & nCount\\_Xenium & nFeature\\_Xenium & nCount\\_BlankCodeword & nFeature\\_BlankCodeword & nCount\\_ControlCodeword & nFeature\\_ControlCodeword & nCount\\_ControlProbe & nFeature\\_ControlProbe & density & elongation & avg\\_confidence\\\\\n",
       "  & <fct> & <dbl> & <int> & <dbl> & <int> & <dbl> & <int> & <dbl> & <int> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t0 & SeuratProject & 818150 & 100 & 0 & 0 & 0 & 0 & 0 & 0 &      NA &    NA &     NA\\\\\n",
       "\tCell10000\\_6303\\_6302 & SeuratProject &    115 &  42 & 0 & 0 & 0 & 0 & 0 & 0 & 0.04981 & 1.178 & 0.9905\\\\\n",
       "\tCell10001\\_4431\\_4179 & SeuratProject &     10 &   7 & 0 & 0 & 0 & 0 & 0 & 0 & 0.08032 & 2.160 & 0.9817\\\\\n",
       "\tCell10002\\_5865\\_7477 & SeuratProject &     34 &   6 & 0 & 0 & 0 & 0 & 0 & 0 & 0.03700 & 7.902 & 0.9860\\\\\n",
       "\tCell10003\\_7028\\_1933 & SeuratProject &     14 &  12 & 0 & 0 & 0 & 0 & 0 & 0 & 0.08833 & 3.136 & 0.9893\\\\\n",
       "\tCell10004\\_4228\\_10692 & SeuratProject &     88 &  32 & 0 & 0 & 0 & 0 & 0 & 0 & 0.04916 & 1.688 & 0.9892\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 12\n",
       "\n",
       "| <!--/--> | orig.ident &lt;fct&gt; | nCount_Xenium &lt;dbl&gt; | nFeature_Xenium &lt;int&gt; | nCount_BlankCodeword &lt;dbl&gt; | nFeature_BlankCodeword &lt;int&gt; | nCount_ControlCodeword &lt;dbl&gt; | nFeature_ControlCodeword &lt;int&gt; | nCount_ControlProbe &lt;dbl&gt; | nFeature_ControlProbe &lt;int&gt; | density &lt;dbl&gt; | elongation &lt;dbl&gt; | avg_confidence &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0 | SeuratProject | 818150 | 100 | 0 | 0 | 0 | 0 | 0 | 0 |      NA |    NA |     NA |\n",
       "| Cell10000_6303_6302 | SeuratProject |    115 |  42 | 0 | 0 | 0 | 0 | 0 | 0 | 0.04981 | 1.178 | 0.9905 |\n",
       "| Cell10001_4431_4179 | SeuratProject |     10 |   7 | 0 | 0 | 0 | 0 | 0 | 0 | 0.08032 | 2.160 | 0.9817 |\n",
       "| Cell10002_5865_7477 | SeuratProject |     34 |   6 | 0 | 0 | 0 | 0 | 0 | 0 | 0.03700 | 7.902 | 0.9860 |\n",
       "| Cell10003_7028_1933 | SeuratProject |     14 |  12 | 0 | 0 | 0 | 0 | 0 | 0 | 0.08833 | 3.136 | 0.9893 |\n",
       "| Cell10004_4228_10692 | SeuratProject |     88 |  32 | 0 | 0 | 0 | 0 | 0 | 0 | 0.04916 | 1.688 | 0.9892 |\n",
       "\n"
      ],
      "text/plain": [
       "                     orig.ident    nCount_Xenium nFeature_Xenium\n",
       "0                    SeuratProject 818150        100            \n",
       "Cell10000_6303_6302  SeuratProject    115         42            \n",
       "Cell10001_4431_4179  SeuratProject     10          7            \n",
       "Cell10002_5865_7477  SeuratProject     34          6            \n",
       "Cell10003_7028_1933  SeuratProject     14         12            \n",
       "Cell10004_4228_10692 SeuratProject     88         32            \n",
       "                     nCount_BlankCodeword nFeature_BlankCodeword\n",
       "0                    0                    0                     \n",
       "Cell10000_6303_6302  0                    0                     \n",
       "Cell10001_4431_4179  0                    0                     \n",
       "Cell10002_5865_7477  0                    0                     \n",
       "Cell10003_7028_1933  0                    0                     \n",
       "Cell10004_4228_10692 0                    0                     \n",
       "                     nCount_ControlCodeword nFeature_ControlCodeword\n",
       "0                    0                      0                       \n",
       "Cell10000_6303_6302  0                      0                       \n",
       "Cell10001_4431_4179  0                      0                       \n",
       "Cell10002_5865_7477  0                      0                       \n",
       "Cell10003_7028_1933  0                      0                       \n",
       "Cell10004_4228_10692 0                      0                       \n",
       "                     nCount_ControlProbe nFeature_ControlProbe density\n",
       "0                    0                   0                          NA\n",
       "Cell10000_6303_6302  0                   0                     0.04981\n",
       "Cell10001_4431_4179  0                   0                     0.08032\n",
       "Cell10002_5865_7477  0                   0                     0.03700\n",
       "Cell10003_7028_1933  0                   0                     0.08833\n",
       "Cell10004_4228_10692 0                   0                     0.04916\n",
       "                     elongation avg_confidence\n",
       "0                       NA          NA        \n",
       "Cell10000_6303_6302  1.178      0.9905        \n",
       "Cell10001_4431_4179  2.160      0.9817        \n",
       "Cell10002_5865_7477  7.902      0.9860        \n",
       "Cell10003_7028_1933  3.136      0.9893        \n",
       "Cell10004_4228_10692 1.688      0.9892        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "obj %>% `[[` %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "fa018e40",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (data.dir, fov = \"fov\", assay = \"Xenium\") \n",
       "{\n",
       "<span style=white-space:pre-wrap>    data &lt;- ReadXenium(data.dir = data.dir, type = c(\"centroids\", </span>\n",
       "<span style=white-space:pre-wrap>        \"segmentations\"), )</span>\n",
       "<span style=white-space:pre-wrap>    segmentations.data &lt;- list(centroids = CreateCentroids(data$centroids), </span>\n",
       "<span style=white-space:pre-wrap>        segmentation = CreateSegmentation(data$segmentations))</span>\n",
       "<span style=white-space:pre-wrap>    coords &lt;- CreateFOV(coords = segmentations.data, type = c(\"segmentation\", </span>\n",
       "<span style=white-space:pre-wrap>        \"centroids\"), molecules = data$microns, assay = assay)</span>\n",
       "<span style=white-space:pre-wrap>    xenium.obj &lt;- CreateSeuratObject(counts = data$matrix[[\"Gene Expression\"]], </span>\n",
       "<span style=white-space:pre-wrap>        assay = assay)</span>\n",
       "<span style=white-space:pre-wrap>    if (\"Blank Codeword\" %in% names(data$matrix)) </span>\n",
       "<span style=white-space:pre-wrap>        xenium.obj[[\"BlankCodeword\"]] &lt;- CreateAssayObject(counts = data$matrix[[\"Blank Codeword\"]])</span>\n",
       "<span style=white-space:pre-wrap>    else xenium.obj[[\"BlankCodeword\"]] &lt;- CreateAssayObject(counts = data$matrix[[\"Unassigned Codeword\"]])</span>\n",
       "<span style=white-space:pre-wrap>    xenium.obj[[\"ControlCodeword\"]] &lt;- CreateAssayObject(counts = data$matrix[[\"Negative Control Codeword\"]])</span>\n",
       "<span style=white-space:pre-wrap>    xenium.obj[[\"ControlProbe\"]] &lt;- CreateAssayObject(counts = data$matrix[[\"Negative Control Probe\"]])</span>\n",
       "<span style=white-space:pre-wrap>    xenium.obj[[fov]] &lt;- coords</span>\n",
       "<span style=white-space:pre-wrap>    return(xenium.obj)</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (data.dir, fov = \"fov\", assay = \"Xenium\") \n",
       "\\{\n",
       "    data <- ReadXenium(data.dir = data.dir, type = c(\"centroids\", \n",
       "        \"segmentations\"), )\n",
       "    segmentations.data <- list(centroids = CreateCentroids(data\\$centroids), \n",
       "        segmentation = CreateSegmentation(data\\$segmentations))\n",
       "    coords <- CreateFOV(coords = segmentations.data, type = c(\"segmentation\", \n",
       "        \"centroids\"), molecules = data\\$microns, assay = assay)\n",
       "    xenium.obj <- CreateSeuratObject(counts = data\\$matrix{[}{[}\"Gene Expression\"{]}{]}, \n",
       "        assay = assay)\n",
       "    if (\"Blank Codeword\" \\%in\\% names(data\\$matrix)) \n",
       "        xenium.obj{[}{[}\"BlankCodeword\"{]}{]} <- CreateAssayObject(counts = data\\$matrix{[}{[}\"Blank Codeword\"{]}{]})\n",
       "    else xenium.obj{[}{[}\"BlankCodeword\"{]}{]} <- CreateAssayObject(counts = data\\$matrix{[}{[}\"Unassigned Codeword\"{]}{]})\n",
       "    xenium.obj{[}{[}\"ControlCodeword\"{]}{]} <- CreateAssayObject(counts = data\\$matrix{[}{[}\"Negative Control Codeword\"{]}{]})\n",
       "    xenium.obj{[}{[}\"ControlProbe\"{]}{]} <- CreateAssayObject(counts = data\\$matrix{[}{[}\"Negative Control Probe\"{]}{]})\n",
       "    xenium.obj{[}{[}fov{]}{]} <- coords\n",
       "    return(xenium.obj)\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (data.dir, fov = \"fov\", assay = \"Xenium\") \n",
       "{\n",
       "    data <- ReadXenium(data.dir = data.dir, type = c(\"centroids\", \n",
       "        \"segmentations\"), )\n",
       "    segmentations.data <- list(centroids = CreateCentroids(data$centroids), \n",
       "        segmentation = CreateSegmentation(data$segmentations))\n",
       "    coords <- CreateFOV(coords = segmentations.data, type = c(\"segmentation\", \n",
       "        \"centroids\"), molecules = data$microns, assay = assay)\n",
       "    xenium.obj <- CreateSeuratObject(counts = data$matrix[[\"Gene Expression\"]], \n",
       "        assay = assay)\n",
       "    if (\"Blank Codeword\" %in% names(data$matrix)) \n",
       "        xenium.obj[[\"BlankCodeword\"]] <- CreateAssayObject(counts = data$matrix[[\"Blank Codeword\"]])\n",
       "    else xenium.obj[[\"BlankCodeword\"]] <- CreateAssayObject(counts = data$matrix[[\"Unassigned Codeword\"]])\n",
       "    xenium.obj[[\"ControlCodeword\"]] <- CreateAssayObject(counts = data$matrix[[\"Negative Control Codeword\"]])\n",
       "    xenium.obj[[\"ControlProbe\"]] <- CreateAssayObject(counts = data$matrix[[\"Negative Control Probe\"]])\n",
       "    xenium.obj[[fov]] <- coords\n",
       "    return(xenium.obj)\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (data.dir, fov = \"fov\", assay = \"Xenium\") \n",
       "{\n",
       "    data <- ReadXenium(data.dir = data.dir, type = c(\"centroids\", \n",
       "        \"segmentations\"), )\n",
       "    segmentations.data <- list(centroids = CreateCentroids(data$centroids), \n",
       "        segmentation = CreateSegmentation(data$segmentations))\n",
       "    coords <- CreateFOV(coords = segmentations.data, type = c(\"segmentation\", \n",
       "        \"centroids\"), molecules = data$microns, assay = assay)\n",
       "    xenium.obj <- CreateSeuratObject(counts = data$matrix[[\"Gene Expression\"]], \n",
       "        assay = assay)\n",
       "    if (\"Blank Codeword\" %in% names(data$matrix)) \n",
       "        xenium.obj[[\"BlankCodeword\"]] <- CreateAssayObject(counts = data$matrix[[\"Blank Codeword\"]])\n",
       "    else xenium.obj[[\"BlankCodeword\"]] <- CreateAssayObject(counts = data$matrix[[\"Unassigned Codeword\"]])\n",
       "    xenium.obj[[\"ControlCodeword\"]] <- CreateAssayObject(counts = data$matrix[[\"Negative Control Codeword\"]])\n",
       "    xenium.obj[[\"ControlProbe\"]] <- CreateAssayObject(counts = data$matrix[[\"Negative Control Probe\"]])\n",
       "    xenium.obj[[fov]] <- coords\n",
       "    return(xenium.obj)\n",
       "}\n",
       "<bytecode: 0x56407b2624c0>\n",
       "<environment: namespace:Seurat>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Seurat::LoadXenium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "b9ab5f59",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (data.dir, outs = c(\"matrix\", \"microns\"), type = \"centroids\", \n",
       "<span style=white-space:pre-wrap>    mols.qv.threshold = 20) </span>\n",
       "{\n",
       "<span style=white-space:pre-wrap>    type &lt;- match.arg(arg = type, choices = c(\"centroids\", \"segmentations\"), </span>\n",
       "<span style=white-space:pre-wrap>        several.ok = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>    outs &lt;- match.arg(arg = outs, choices = c(\"matrix\", \"microns\"), </span>\n",
       "<span style=white-space:pre-wrap>        several.ok = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>    outs &lt;- c(outs, type)</span>\n",
       "<span style=white-space:pre-wrap>    has_dt &lt;- requireNamespace(\"data.table\", quietly = TRUE) &amp;&amp; </span>\n",
       "<span style=white-space:pre-wrap>        requireNamespace(\"R.utils\", quietly = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>    data &lt;- sapply(outs, function(otype) {</span>\n",
       "<span style=white-space:pre-wrap>        switch(EXPR = otype, matrix = {</span>\n",
       "<span style=white-space:pre-wrap>            pmtx &lt;- progressor()</span>\n",
       "<span style=white-space:pre-wrap>            pmtx(message = \"Reading counts matrix\", class = \"sticky\", </span>\n",
       "<span style=white-space:pre-wrap>                amount = 0)</span>\n",
       "<span style=white-space:pre-wrap>            matrix &lt;- suppressWarnings(Read10X(data.dir = file.path(data.dir, </span>\n",
       "<span style=white-space:pre-wrap>                \"cell_feature_matrix/\")))</span>\n",
       "<span style=white-space:pre-wrap>            pmtx(type = \"finish\")</span>\n",
       "<span style=white-space:pre-wrap>            matrix</span>\n",
       "<span style=white-space:pre-wrap>        }, centroids = {</span>\n",
       "<span style=white-space:pre-wrap>            pcents &lt;- progressor()</span>\n",
       "<span style=white-space:pre-wrap>            pcents(message = \"Loading cell centroids\", class = \"sticky\", </span>\n",
       "<span style=white-space:pre-wrap>                amount = 0)</span>\n",
       "<span style=white-space:pre-wrap>            if (has_dt) {</span>\n",
       "<span style=white-space:pre-wrap>                cell_info &lt;- as.data.frame(data.table::fread(file.path(data.dir, </span>\n",
       "<span style=white-space:pre-wrap>                  \"cells.csv.gz\")))</span>\n",
       "<span style=white-space:pre-wrap>            } else {</span>\n",
       "<span style=white-space:pre-wrap>                cell_info &lt;- read.csv(file.path(data.dir, \"cells.csv.gz\"))</span>\n",
       "<span style=white-space:pre-wrap>            }</span>\n",
       "<span style=white-space:pre-wrap>            cell_centroid_df &lt;- data.frame(x = cell_info$x_centroid, </span>\n",
       "<span style=white-space:pre-wrap>                y = cell_info$y_centroid, cell = cell_info$cell_id, </span>\n",
       "<span style=white-space:pre-wrap>                stringsAsFactors = FALSE)</span>\n",
       "<span style=white-space:pre-wrap>            pcents(type = \"finish\")</span>\n",
       "<span style=white-space:pre-wrap>            cell_centroid_df</span>\n",
       "<span style=white-space:pre-wrap>        }, segmentations = {</span>\n",
       "<span style=white-space:pre-wrap>            psegs &lt;- progressor()</span>\n",
       "<span style=white-space:pre-wrap>            psegs(message = \"Loading cell segmentations\", class = \"sticky\", </span>\n",
       "<span style=white-space:pre-wrap>                amount = 0)</span>\n",
       "<span style=white-space:pre-wrap>            if (has_dt) {</span>\n",
       "<span style=white-space:pre-wrap>                cell_boundaries_df &lt;- as.data.frame(data.table::fread(file.path(data.dir, </span>\n",
       "<span style=white-space:pre-wrap>                  \"cell_boundaries.csv.gz\")))</span>\n",
       "<span style=white-space:pre-wrap>            } else {</span>\n",
       "<span style=white-space:pre-wrap>                cell_boundaries_df &lt;- read.csv(file.path(data.dir, </span>\n",
       "<span style=white-space:pre-wrap>                  \"cell_boundaries.csv.gz\"), stringsAsFactors = FALSE)</span>\n",
       "<span style=white-space:pre-wrap>            }</span>\n",
       "<span style=white-space:pre-wrap>            names(cell_boundaries_df) &lt;- c(\"cell\", \"x\", \"y\")</span>\n",
       "<span style=white-space:pre-wrap>            psegs(type = \"finish\")</span>\n",
       "<span style=white-space:pre-wrap>            cell_boundaries_df</span>\n",
       "<span style=white-space:pre-wrap>        }, microns = {</span>\n",
       "<span style=white-space:pre-wrap>            pmicrons &lt;- progressor()</span>\n",
       "<span style=white-space:pre-wrap>            pmicrons(message = \"Loading molecule coordinates\", </span>\n",
       "<span style=white-space:pre-wrap>                class = \"sticky\", amount = 0)</span>\n",
       "<span style=white-space:pre-wrap>            if (has_dt) {</span>\n",
       "<span style=white-space:pre-wrap>                tx_dt &lt;- as.data.frame(data.table::fread(file.path(data.dir, </span>\n",
       "<span style=white-space:pre-wrap>                  \"transcripts.csv.gz\")))</span>\n",
       "<span style=white-space:pre-wrap>                transcripts &lt;- subset(tx_dt, qv &gt;= mols.qv.threshold)</span>\n",
       "<span style=white-space:pre-wrap>            } else {</span>\n",
       "<span style=white-space:pre-wrap>                transcripts &lt;- read.csv(file.path(data.dir, \"transcripts.csv.gz\"))</span>\n",
       "<span style=white-space:pre-wrap>                transcripts &lt;- subset(transcripts, qv &gt;= mols.qv.threshold)</span>\n",
       "<span style=white-space:pre-wrap>            }</span>\n",
       "<span style=white-space:pre-wrap>            df &lt;- data.frame(x = transcripts$x_location, y = transcripts$y_location, </span>\n",
       "<span style=white-space:pre-wrap>                gene = transcripts$feature_name, stringsAsFactors = FALSE)</span>\n",
       "<span style=white-space:pre-wrap>            pmicrons(type = \"finish\")</span>\n",
       "<span style=white-space:pre-wrap>            df</span>\n",
       "<span style=white-space:pre-wrap>        }, stop(\"Unknown Xenium input type: \", otype))</span>\n",
       "<span style=white-space:pre-wrap>    }, USE.NAMES = TRUE)</span>\n",
       "<span style=white-space:pre-wrap>    return(data)</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (data.dir, outs = c(\"matrix\", \"microns\"), type = \"centroids\", \n",
       "    mols.qv.threshold = 20) \n",
       "\\{\n",
       "    type <- match.arg(arg = type, choices = c(\"centroids\", \"segmentations\"), \n",
       "        several.ok = TRUE)\n",
       "    outs <- match.arg(arg = outs, choices = c(\"matrix\", \"microns\"), \n",
       "        several.ok = TRUE)\n",
       "    outs <- c(outs, type)\n",
       "    has\\_dt <- requireNamespace(\"data.table\", quietly = TRUE) \\&\\& \n",
       "        requireNamespace(\"R.utils\", quietly = TRUE)\n",
       "    data <- sapply(outs, function(otype) \\{\n",
       "        switch(EXPR = otype, matrix = \\{\n",
       "            pmtx <- progressor()\n",
       "            pmtx(message = \"Reading counts matrix\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            matrix <- suppressWarnings(Read10X(data.dir = file.path(data.dir, \n",
       "                \"cell\\_feature\\_matrix/\")))\n",
       "            pmtx(type = \"finish\")\n",
       "            matrix\n",
       "        \\}, centroids = \\{\n",
       "            pcents <- progressor()\n",
       "            pcents(message = \"Loading cell centroids\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            if (has\\_dt) \\{\n",
       "                cell\\_info <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"cells.csv.gz\")))\n",
       "            \\} else \\{\n",
       "                cell\\_info <- read.csv(file.path(data.dir, \"cells.csv.gz\"))\n",
       "            \\}\n",
       "            cell\\_centroid\\_df <- data.frame(x = cell\\_info\\$x\\_centroid, \n",
       "                y = cell\\_info\\$y\\_centroid, cell = cell\\_info\\$cell\\_id, \n",
       "                stringsAsFactors = FALSE)\n",
       "            pcents(type = \"finish\")\n",
       "            cell\\_centroid\\_df\n",
       "        \\}, segmentations = \\{\n",
       "            psegs <- progressor()\n",
       "            psegs(message = \"Loading cell segmentations\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            if (has\\_dt) \\{\n",
       "                cell\\_boundaries\\_df <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"cell\\_boundaries.csv.gz\")))\n",
       "            \\} else \\{\n",
       "                cell\\_boundaries\\_df <- read.csv(file.path(data.dir, \n",
       "                  \"cell\\_boundaries.csv.gz\"), stringsAsFactors = FALSE)\n",
       "            \\}\n",
       "            names(cell\\_boundaries\\_df) <- c(\"cell\", \"x\", \"y\")\n",
       "            psegs(type = \"finish\")\n",
       "            cell\\_boundaries\\_df\n",
       "        \\}, microns = \\{\n",
       "            pmicrons <- progressor()\n",
       "            pmicrons(message = \"Loading molecule coordinates\", \n",
       "                class = \"sticky\", amount = 0)\n",
       "            if (has\\_dt) \\{\n",
       "                tx\\_dt <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"transcripts.csv.gz\")))\n",
       "                transcripts <- subset(tx\\_dt, qv >= mols.qv.threshold)\n",
       "            \\} else \\{\n",
       "                transcripts <- read.csv(file.path(data.dir, \"transcripts.csv.gz\"))\n",
       "                transcripts <- subset(transcripts, qv >= mols.qv.threshold)\n",
       "            \\}\n",
       "            df <- data.frame(x = transcripts\\$x\\_location, y = transcripts\\$y\\_location, \n",
       "                gene = transcripts\\$feature\\_name, stringsAsFactors = FALSE)\n",
       "            pmicrons(type = \"finish\")\n",
       "            df\n",
       "        \\}, stop(\"Unknown Xenium input type: \", otype))\n",
       "    \\}, USE.NAMES = TRUE)\n",
       "    return(data)\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (data.dir, outs = c(\"matrix\", \"microns\"), type = \"centroids\", \n",
       "    mols.qv.threshold = 20) \n",
       "{\n",
       "    type <- match.arg(arg = type, choices = c(\"centroids\", \"segmentations\"), \n",
       "        several.ok = TRUE)\n",
       "    outs <- match.arg(arg = outs, choices = c(\"matrix\", \"microns\"), \n",
       "        several.ok = TRUE)\n",
       "    outs <- c(outs, type)\n",
       "    has_dt <- requireNamespace(\"data.table\", quietly = TRUE) && \n",
       "        requireNamespace(\"R.utils\", quietly = TRUE)\n",
       "    data <- sapply(outs, function(otype) {\n",
       "        switch(EXPR = otype, matrix = {\n",
       "            pmtx <- progressor()\n",
       "            pmtx(message = \"Reading counts matrix\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            matrix <- suppressWarnings(Read10X(data.dir = file.path(data.dir, \n",
       "                \"cell_feature_matrix/\")))\n",
       "            pmtx(type = \"finish\")\n",
       "            matrix\n",
       "        }, centroids = {\n",
       "            pcents <- progressor()\n",
       "            pcents(message = \"Loading cell centroids\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            if (has_dt) {\n",
       "                cell_info <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"cells.csv.gz\")))\n",
       "            } else {\n",
       "                cell_info <- read.csv(file.path(data.dir, \"cells.csv.gz\"))\n",
       "            }\n",
       "            cell_centroid_df <- data.frame(x = cell_info$x_centroid, \n",
       "                y = cell_info$y_centroid, cell = cell_info$cell_id, \n",
       "                stringsAsFactors = FALSE)\n",
       "            pcents(type = \"finish\")\n",
       "            cell_centroid_df\n",
       "        }, segmentations = {\n",
       "            psegs <- progressor()\n",
       "            psegs(message = \"Loading cell segmentations\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            if (has_dt) {\n",
       "                cell_boundaries_df <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"cell_boundaries.csv.gz\")))\n",
       "            } else {\n",
       "                cell_boundaries_df <- read.csv(file.path(data.dir, \n",
       "                  \"cell_boundaries.csv.gz\"), stringsAsFactors = FALSE)\n",
       "            }\n",
       "            names(cell_boundaries_df) <- c(\"cell\", \"x\", \"y\")\n",
       "            psegs(type = \"finish\")\n",
       "            cell_boundaries_df\n",
       "        }, microns = {\n",
       "            pmicrons <- progressor()\n",
       "            pmicrons(message = \"Loading molecule coordinates\", \n",
       "                class = \"sticky\", amount = 0)\n",
       "            if (has_dt) {\n",
       "                tx_dt <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"transcripts.csv.gz\")))\n",
       "                transcripts <- subset(tx_dt, qv >= mols.qv.threshold)\n",
       "            } else {\n",
       "                transcripts <- read.csv(file.path(data.dir, \"transcripts.csv.gz\"))\n",
       "                transcripts <- subset(transcripts, qv >= mols.qv.threshold)\n",
       "            }\n",
       "            df <- data.frame(x = transcripts$x_location, y = transcripts$y_location, \n",
       "                gene = transcripts$feature_name, stringsAsFactors = FALSE)\n",
       "            pmicrons(type = \"finish\")\n",
       "            df\n",
       "        }, stop(\"Unknown Xenium input type: \", otype))\n",
       "    }, USE.NAMES = TRUE)\n",
       "    return(data)\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (data.dir, outs = c(\"matrix\", \"microns\"), type = \"centroids\", \n",
       "    mols.qv.threshold = 20) \n",
       "{\n",
       "    type <- match.arg(arg = type, choices = c(\"centroids\", \"segmentations\"), \n",
       "        several.ok = TRUE)\n",
       "    outs <- match.arg(arg = outs, choices = c(\"matrix\", \"microns\"), \n",
       "        several.ok = TRUE)\n",
       "    outs <- c(outs, type)\n",
       "    has_dt <- requireNamespace(\"data.table\", quietly = TRUE) && \n",
       "        requireNamespace(\"R.utils\", quietly = TRUE)\n",
       "    data <- sapply(outs, function(otype) {\n",
       "        switch(EXPR = otype, matrix = {\n",
       "            pmtx <- progressor()\n",
       "            pmtx(message = \"Reading counts matrix\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            matrix <- suppressWarnings(Read10X(data.dir = file.path(data.dir, \n",
       "                \"cell_feature_matrix/\")))\n",
       "            pmtx(type = \"finish\")\n",
       "            matrix\n",
       "        }, centroids = {\n",
       "            pcents <- progressor()\n",
       "            pcents(message = \"Loading cell centroids\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            if (has_dt) {\n",
       "                cell_info <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"cells.csv.gz\")))\n",
       "            } else {\n",
       "                cell_info <- read.csv(file.path(data.dir, \"cells.csv.gz\"))\n",
       "            }\n",
       "            cell_centroid_df <- data.frame(x = cell_info$x_centroid, \n",
       "                y = cell_info$y_centroid, cell = cell_info$cell_id, \n",
       "                stringsAsFactors = FALSE)\n",
       "            pcents(type = \"finish\")\n",
       "            cell_centroid_df\n",
       "        }, segmentations = {\n",
       "            psegs <- progressor()\n",
       "            psegs(message = \"Loading cell segmentations\", class = \"sticky\", \n",
       "                amount = 0)\n",
       "            if (has_dt) {\n",
       "                cell_boundaries_df <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"cell_boundaries.csv.gz\")))\n",
       "            } else {\n",
       "                cell_boundaries_df <- read.csv(file.path(data.dir, \n",
       "                  \"cell_boundaries.csv.gz\"), stringsAsFactors = FALSE)\n",
       "            }\n",
       "            names(cell_boundaries_df) <- c(\"cell\", \"x\", \"y\")\n",
       "            psegs(type = \"finish\")\n",
       "            cell_boundaries_df\n",
       "        }, microns = {\n",
       "            pmicrons <- progressor()\n",
       "            pmicrons(message = \"Loading molecule coordinates\", \n",
       "                class = \"sticky\", amount = 0)\n",
       "            if (has_dt) {\n",
       "                tx_dt <- as.data.frame(data.table::fread(file.path(data.dir, \n",
       "                  \"transcripts.csv.gz\")))\n",
       "                transcripts <- subset(tx_dt, qv >= mols.qv.threshold)\n",
       "            } else {\n",
       "                transcripts <- read.csv(file.path(data.dir, \"transcripts.csv.gz\"))\n",
       "                transcripts <- subset(transcripts, qv >= mols.qv.threshold)\n",
       "            }\n",
       "            df <- data.frame(x = transcripts$x_location, y = transcripts$y_location, \n",
       "                gene = transcripts$feature_name, stringsAsFactors = FALSE)\n",
       "            pmicrons(type = \"finish\")\n",
       "            df\n",
       "        }, stop(\"Unknown Xenium input type: \", otype))\n",
       "    }, USE.NAMES = TRUE)\n",
       "    return(data)\n",
       "}\n",
       "<bytecode: 0x56407aa82e98>\n",
       "<environment: namespace:Seurat>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Seurat::ReadXenium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "5209f032",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (lhs, rhs) \n",
       "{\n",
       "<span style=white-space:pre-wrap>    lhs &lt;- substitute(lhs)</span>\n",
       "<span style=white-space:pre-wrap>    rhs &lt;- substitute(rhs)</span>\n",
       "<span style=white-space:pre-wrap>    kind &lt;- 1L</span>\n",
       "<span style=white-space:pre-wrap>    env &lt;- parent.frame()</span>\n",
       "<span style=white-space:pre-wrap>    lazy &lt;- TRUE</span>\n",
       "<span style=white-space:pre-wrap>    .External2(magrittr_pipe)</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (lhs, rhs) \n",
       "\\{\n",
       "    lhs <- substitute(lhs)\n",
       "    rhs <- substitute(rhs)\n",
       "    kind <- 1L\n",
       "    env <- parent.frame()\n",
       "    lazy <- TRUE\n",
       "    .External2(magrittr\\_pipe)\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (lhs, rhs) \n",
       "{\n",
       "    lhs <- substitute(lhs)\n",
       "    rhs <- substitute(rhs)\n",
       "    kind <- 1L\n",
       "    env <- parent.frame()\n",
       "    lazy <- TRUE\n",
       "    .External2(magrittr_pipe)\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (lhs, rhs) \n",
       "{\n",
       "    lhs <- substitute(lhs)\n",
       "    rhs <- substitute(rhs)\n",
       "    kind <- 1L\n",
       "    env <- parent.frame()\n",
       "    lazy <- TRUE\n",
       "    .External2(magrittr_pipe)\n",
       "}\n",
       "<bytecode: 0x5640760ffbd8>\n",
       "<environment: namespace:magrittr>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "magrittr::`%>%`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbb6ce4c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:fgf1]",
   "language": "R",
   "name": "conda-env-fgf1-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
