{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d4880c41",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.2 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.6      \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.8      \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.10\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.1      \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.1 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2      \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.2 \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n"
     ]
    }
   ],
   "source": [
    "library(targets)\n",
    "library(tarchetypes)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "2d3c8a96",
   "metadata": {},
   "outputs": [],
   "source": [
    "tar_load(combined_da_results_obob14v14,)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "4c404113",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining, by = \"labels\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[90m# A tibble: 10 × 4\u001b[39m\n",
      "   labels              n_all n_limited  diff\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m               \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m     \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m b1_Tanycytes           15        14     1\n",
      "\u001b[90m 2\u001b[39m MOL                     4         3     1\n",
      "\u001b[90m 3\u001b[39m Microglia               1         2    -\u001b[31m1\u001b[39m\n",
      "\u001b[90m 4\u001b[39m b2_Tany                 1         5    -\u001b[31m4\u001b[39m\n",
      "\u001b[90m 5\u001b[39m a2_Tanycytes            1        \u001b[31mNA\u001b[39m    \u001b[31mNA\u001b[39m\n",
      "\u001b[90m 6\u001b[39m a1_Tany__Astrocytes    \u001b[31mNA\u001b[39m         3    \u001b[31mNA\u001b[39m\n",
      "\u001b[90m 7\u001b[39m Astrocytes             \u001b[31mNA\u001b[39m         2    \u001b[31mNA\u001b[39m\n",
      "\u001b[90m 8\u001b[39m Kisspeptin             \u001b[31mNA\u001b[39m         3    \u001b[31mNA\u001b[39m\n",
      "\u001b[90m 9\u001b[39m NG2_OPC2               \u001b[31mNA\u001b[39m         1    \u001b[31mNA\u001b[39m\n",
      "\u001b[90m10\u001b[39m Sst_Pthlh.sc23         \u001b[31mNA\u001b[39m         1    \u001b[31mNA\u001b[39m\n"
     ]
    }
   ],
   "source": [
    "tar_load(combined_da_results_all_obob14v14)\n",
    "all = combined_da_results_all_obob14v14  %>%\n",
    "filter(SpatialFDR < 0.1) %>%\n",
    "group_by(labels) %>%\n",
    "summarise(n_all = n())\n",
    "\n",
    "tar_load(combined_da_results_Day14_obob14v14)\n",
    "limited = combined_da_results_Day14_obob14v14  %>%\n",
    "filter(SpatialFDR < 0.1) %>%\n",
    "group_by(labels) %>%\n",
    "summarise(n_limited = n())\n",
    "\n",
    "full_join(all, limited) %>% \n",
    "mutate(diff = n_all - n_limited) %>%\n",
    "arrange(desc(diff)) %>%\n",
    "print(n=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "5972afa6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining, by = \"labels\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[90m# A tibble: 28 × 4\u001b[39m\n",
      "   labels                     n_all n_limited  diff\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m     \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m Astrocytes                   196       107    89\n",
      "\u001b[90m 2\u001b[39m b1_Tanycytes                  63        37    26\n",
      "\u001b[90m 3\u001b[39m NG2_OPC2                      42        17    25\n",
      "\u001b[90m 4\u001b[39m a1_Tany__Astrocytes           47        25    22\n",
      "\u001b[90m 5\u001b[39m Sst_Pthlh.sc23                24         7    17\n",
      "\u001b[90m 6\u001b[39m Agrp                          48        34    14\n",
      "\u001b[90m 7\u001b[39m MOL                           54        40    14\n",
      "\u001b[90m 8\u001b[39m a2_Tanycytes__b1_Tanycytes    16         5    11\n",
      "\u001b[90m 9\u001b[39m Htr3b                         85        74    11\n",
      "\u001b[90m10\u001b[39m b2_Tany                        8         1     7\n",
      "\u001b[90m11\u001b[39m Microglia                     10         3     7\n",
      "\u001b[90m12\u001b[39m Kisspeptin                    11         6     5\n",
      "\u001b[90m13\u001b[39m Pomc_Glipr1__Slc17a6_Trhr     16        11     5\n",
      "\u001b[90m14\u001b[39m Unassigned1.sc17.sc28          7         3     4\n",
      "\u001b[90m15\u001b[39m NFOL                          14        11     3\n",
      "\u001b[90m16\u001b[39m Sst_Unc13c__Agrp              37        34     3\n",
      "\u001b[90m17\u001b[39m Rgs16_Dlx1                     3         1     2\n",
      "\u001b[90m18\u001b[39m Nr5a1_Bdnf                     3         2     1\n",
      "\u001b[90m19\u001b[39m Slc17a6_Trhr                   2         1     1\n",
      "\u001b[90m20\u001b[39m a2_Tanycytes                  38        38     0\n",
      "\u001b[90m21\u001b[39m Epend                          5         6    -\u001b[31m1\u001b[39m\n",
      "\u001b[90m22\u001b[39m Ghrh                          15        18    -\u001b[31m3\u001b[39m\n",
      "\u001b[90m23\u001b[39m Trh_Lef1                      57        60    -\u001b[31m3\u001b[39m\n",
      "\u001b[90m24\u001b[39m Pomc_Lepr                      5         9    -\u001b[31m4\u001b[39m\n",
      "\u001b[90m25\u001b[39m Tmem215                       11        16    -\u001b[31m5\u001b[39m\n",
      "\u001b[90m26\u001b[39m Sst_Nts                        1        \u001b[31mNA\u001b[39m    \u001b[31mNA\u001b[39m\n",
      "\u001b[90m27\u001b[39m Rgs16_Nmu                     \u001b[31mNA\u001b[39m         3    \u001b[31mNA\u001b[39m\n",
      "\u001b[90m28\u001b[39m Trh_Cxcl12                    \u001b[31mNA\u001b[39m         5    \u001b[31mNA\u001b[39m\n"
     ]
    }
   ],
   "source": [
    "tar_load(combined_da_results_all_obob5v5)\n",
    "all = combined_da_results_all_obob5v5  %>%\n",
    "filter(SpatialFDR < 0.1) %>%\n",
    "group_by(labels) %>%\n",
    "summarise(n_all = n())\n",
    "\n",
    "tar_load(combined_da_results_Day5_obob5v5)\n",
    "limited = combined_da_results_Day5_obob5v5  %>%\n",
    "filter(SpatialFDR < 0.1) %>%\n",
    "group_by(labels) %>%\n",
    "mutate(n_limited = n()) %>%\n",
    "select(labels, n_limited, n)\n",
    "\n",
    "full_join(all, limited) %>% \n",
    "mutate(diff = n_all - n_limited) %>%\n",
    "arrange(desc(diff)) %>%\n",
    "print(n=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "df1dd4b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_da_results_all.obob5v5 = qs::qread('_targets/objects/combined_da_results_all.obob5v5')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "869d3547",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 9329 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>logFC</th><th scope=col>logCPM</th><th scope=col>F</th><th scope=col>PValue</th><th scope=col>FDR</th><th scope=col>Nhood</th><th scope=col>SpatialFDR</th><th scope=col>polarity</th><th scope=col>labels</th><th scope=col>labels_fraction</th><th scope=col>n_cells</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 5.175528</td><td>12.75419</td><td>65.874861</td><td>2.712965e-11</td><td>6.592504e-09</td><td> 63</td><td>6.856218e-09</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>129</td></tr>\n",
       "\t<tr><td> 4.883802</td><td>12.97953</td><td>50.940929</td><td>1.324004e-09</td><td>1.608665e-07</td><td> 93</td><td>1.592806e-07</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>166</td></tr>\n",
       "\t<tr><td> 3.576113</td><td>12.47521</td><td>33.105950</td><td>2.990955e-07</td><td>2.422673e-05</td><td>214</td><td>2.576412e-05</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td> 99</td></tr>\n",
       "\t<tr><td> 3.401746</td><td>12.73492</td><td>31.074340</td><td>5.936490e-07</td><td>3.606418e-05</td><td>118</td><td>3.781441e-05</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>135</td></tr>\n",
       "\t<tr><td> 3.177160</td><td>13.10128</td><td>23.561109</td><td>8.712993e-06</td><td>4.234514e-04</td><td> 19</td><td>4.260118e-04</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>201</td></tr>\n",
       "\t<tr><td> 2.661283</td><td>13.13157</td><td>21.175818</td><td>2.161802e-05</td><td>8.755300e-04</td><td> 58</td><td>8.618807e-04</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>212</td></tr>\n",
       "\t<tr><td>-3.989157</td><td>12.23840</td><td>20.289935</td><td>3.052978e-05</td><td>1.059819e-03</td><td>200</td><td>1.070892e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td> 97</td></tr>\n",
       "\t<tr><td> 2.180194</td><td>12.94578</td><td>17.482410</td><td>9.390194e-05</td><td>2.852272e-03</td><td>108</td><td>2.821480e-03</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>174</td></tr>\n",
       "\t<tr><td>-3.177985</td><td>12.75694</td><td>17.077316</td><td>1.108626e-04</td><td>2.993290e-03</td><td> 66</td><td>2.903301e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>180</td></tr>\n",
       "\t<tr><td>-2.306381</td><td>13.04335</td><td>15.070167</td><td>2.563977e-04</td><td>6.230464e-03</td><td> 57</td><td>5.944409e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>233</td></tr>\n",
       "\t<tr><td> 2.370110</td><td>12.87178</td><td>14.500229</td><td>3.269528e-04</td><td>7.222684e-03</td><td> 92</td><td>6.870984e-03</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>166</td></tr>\n",
       "\t<tr><td> 2.036045</td><td>12.92883</td><td>14.204678</td><td>3.712173e-04</td><td>7.517151e-03</td><td> 33</td><td>7.127028e-03</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>173</td></tr>\n",
       "\t<tr><td>-2.827849</td><td>12.15033</td><td>13.213011</td><td>5.710871e-04</td><td>8.371871e-03</td><td> 29</td><td>7.963183e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td> 86</td></tr>\n",
       "\t<tr><td> 2.386494</td><td>12.55891</td><td>13.160541</td><td>5.843741e-04</td><td>8.371871e-03</td><td> 99</td><td>7.963183e-03</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>121</td></tr>\n",
       "\t<tr><td>-2.417376</td><td>12.76277</td><td>13.025250</td><td>6.201386e-04</td><td>8.371871e-03</td><td>136</td><td>7.963183e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>183</td></tr>\n",
       "\t<tr><td> 2.133462</td><td>12.55683</td><td>13.398511</td><td>5.265784e-04</td><td>8.371871e-03</td><td>160</td><td>7.963183e-03</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>120</td></tr>\n",
       "\t<tr><td> 1.793196</td><td>12.79742</td><td>13.085623</td><td>6.039056e-04</td><td>8.371871e-03</td><td>165</td><td>7.963183e-03</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>152</td></tr>\n",
       "\t<tr><td>-3.428830</td><td>12.09588</td><td>13.131769</td><td>6.111538e-04</td><td>8.371871e-03</td><td>190</td><td>7.963183e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td> 82</td></tr>\n",
       "\t<tr><td>-2.503424</td><td>12.49668</td><td>12.618147</td><td>7.421542e-04</td><td>8.587784e-03</td><td> 94</td><td>8.122160e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>129</td></tr>\n",
       "\t<tr><td>-3.382299</td><td>12.32172</td><td>12.777130</td><td>7.133974e-04</td><td>8.587784e-03</td><td>218</td><td>8.122160e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>111</td></tr>\n",
       "\t<tr><td>-2.850700</td><td>12.33923</td><td>12.697946</td><td>7.164053e-04</td><td>8.587784e-03</td><td>231</td><td>8.122160e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>110</td></tr>\n",
       "\t<tr><td>-2.279203</td><td>13.20301</td><td>12.282190</td><td>8.615916e-04</td><td>9.516671e-03</td><td>163</td><td>8.940797e-03</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>262</td></tr>\n",
       "\t<tr><td>-2.839669</td><td>12.35562</td><td>11.474459</td><td>1.270543e-03</td><td>1.332662e-02</td><td> 68</td><td>1.265509e-02</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>113</td></tr>\n",
       "\t<tr><td>-4.272532</td><td>12.12827</td><td>11.578755</td><td>1.316209e-03</td><td>1.332662e-02</td><td> 86</td><td>1.266800e-02</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td> 81</td></tr>\n",
       "\t<tr><td>-5.742286</td><td>12.18465</td><td>11.334774</td><td>1.899448e-03</td><td>1.846263e-02</td><td>172</td><td>1.764270e-02</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td> 84</td></tr>\n",
       "\t<tr><td> 1.908189</td><td>12.39935</td><td>10.380648</td><td>2.041626e-03</td><td>1.908135e-02</td><td>242</td><td>1.831509e-02</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>104</td></tr>\n",
       "\t<tr><td> 2.181414</td><td>12.75725</td><td>10.160352</td><td>2.260996e-03</td><td>2.034896e-02</td><td> 85</td><td>1.948854e-02</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>153</td></tr>\n",
       "\t<tr><td> 1.850450</td><td>12.41509</td><td> 9.962382</td><td>2.479158e-03</td><td>2.151555e-02</td><td> 43</td><td>2.082839e-02</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td>103</td></tr>\n",
       "\t<tr><td>-2.174831</td><td>12.48496</td><td> 9.757053</td><td>2.728800e-03</td><td>2.286546e-02</td><td>204</td><td>2.206125e-02</td><td>neg</td><td>a1_Tany__Astrocytes</td><td>1</td><td>135</td></tr>\n",
       "\t<tr><td> 2.095793</td><td>12.24768</td><td> 9.223508</td><td>3.572167e-03</td><td>2.851732e-02</td><td> 39</td><td>2.767170e-02</td><td>pos</td><td>a1_Tany__Astrocytes</td><td>1</td><td> 85</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td> 0.555631832</td><td>16.08223</td><td>0.5982580947</td><td>0.4405852</td><td>0.8242820</td><td> 1</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 73</td></tr>\n",
       "\t<tr><td>-0.431842341</td><td>16.45066</td><td>0.3727559459</td><td>0.5416710</td><td>0.8242820</td><td> 2</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td>100</td></tr>\n",
       "\t<tr><td> 0.551295037</td><td>16.21552</td><td>0.5076834501</td><td>0.4764868</td><td>0.8242820</td><td> 3</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 87</td></tr>\n",
       "\t<tr><td> 0.700026594</td><td>15.96530</td><td>0.6221776818</td><td>0.4304618</td><td>0.8242820</td><td> 4</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 54</td></tr>\n",
       "\t<tr><td> 0.678766967</td><td>16.00659</td><td>0.7110638566</td><td>0.3993309</td><td>0.8242820</td><td> 5</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 67</td></tr>\n",
       "\t<tr><td> 0.525560713</td><td>16.35272</td><td>0.4401633100</td><td>0.5072257</td><td>0.8242820</td><td> 6</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 79</td></tr>\n",
       "\t<tr><td> 0.822935497</td><td>16.01523</td><td>0.6680645342</td><td>0.4139590</td><td>0.8242820</td><td> 7</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 51</td></tr>\n",
       "\t<tr><td>-0.930457943</td><td>16.06482</td><td>1.8719123752</td><td>0.1716221</td><td>0.8242820</td><td> 8</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 54</td></tr>\n",
       "\t<tr><td> 0.987269282</td><td>16.00727</td><td>1.2107956711</td><td>0.2714898</td><td>0.8242820</td><td>10</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 62</td></tr>\n",
       "\t<tr><td>-0.818123984</td><td>15.83660</td><td>0.9477328371</td><td>0.3305773</td><td>0.8242820</td><td>12</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 48</td></tr>\n",
       "\t<tr><td> 0.798806086</td><td>16.16020</td><td>1.1569668441</td><td>0.2824043</td><td>0.8242820</td><td>14</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 86</td></tr>\n",
       "\t<tr><td>-0.492302389</td><td>16.07432</td><td>0.5438116347</td><td>0.4610633</td><td>0.8242820</td><td>18</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 58</td></tr>\n",
       "\t<tr><td> 1.198919460</td><td>15.87275</td><td>1.6293390280</td><td>0.2021473</td><td>0.8242820</td><td>21</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 56</td></tr>\n",
       "\t<tr><td> 0.659272857</td><td>16.08645</td><td>0.9208048899</td><td>0.3375398</td><td>0.8242820</td><td>22</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 69</td></tr>\n",
       "\t<tr><td> 0.740889110</td><td>15.94444</td><td>0.7664462981</td><td>0.3815689</td><td>0.8242820</td><td>23</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 60</td></tr>\n",
       "\t<tr><td> 1.191291104</td><td>16.02297</td><td>2.0260281015</td><td>0.1549956</td><td>0.8242820</td><td>25</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 73</td></tr>\n",
       "\t<tr><td> 0.564786684</td><td>15.85552</td><td>0.3849789677</td><td>0.5351188</td><td>0.8242820</td><td>32</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 46</td></tr>\n",
       "\t<tr><td> 0.994682963</td><td>16.00759</td><td>1.3349327380</td><td>0.2482581</td><td>0.8242820</td><td>35</td><td>0.8253250</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 69</td></tr>\n",
       "\t<tr><td>-0.406219846</td><td>16.27201</td><td>0.3062747452</td><td>0.5801225</td><td>0.8460119</td><td>17</td><td>0.8485754</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 68</td></tr>\n",
       "\t<tr><td> 0.388868523</td><td>16.02849</td><td>0.2318672920</td><td>0.6302681</td><td>0.8823754</td><td>27</td><td>0.8811158</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 65</td></tr>\n",
       "\t<tr><td> 0.283796137</td><td>15.93773</td><td>0.1165066440</td><td>0.7329400</td><td>0.9524475</td><td>28</td><td>0.9559703</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 54</td></tr>\n",
       "\t<tr><td> 0.267700697</td><td>16.15524</td><td>0.1148753312</td><td>0.7347452</td><td>0.9524475</td><td>30</td><td>0.9559703</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 65</td></tr>\n",
       "\t<tr><td> 0.223272009</td><td>16.06103</td><td>0.0872889023</td><td>0.7677255</td><td>0.9553561</td><td>16</td><td>0.9585071</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 57</td></tr>\n",
       "\t<tr><td> 0.427225223</td><td>16.07938</td><td>0.3455399138</td><td>0.7915807</td><td>0.9553561</td><td>26</td><td>0.9585071</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 54</td></tr>\n",
       "\t<tr><td>-0.103405228</td><td>16.17067</td><td>0.0193652356</td><td>0.8893578</td><td>0.9727351</td><td>11</td><td>0.9718001</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 72</td></tr>\n",
       "\t<tr><td>-0.121413863</td><td>16.08554</td><td>0.0350256697</td><td>0.8515871</td><td>0.9727351</td><td>24</td><td>0.9718001</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 65</td></tr>\n",
       "\t<tr><td> 0.105516676</td><td>16.10242</td><td>0.0219448092</td><td>0.8822696</td><td>0.9727351</td><td>31</td><td>0.9718001</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 61</td></tr>\n",
       "\t<tr><td> 0.046865954</td><td>16.17896</td><td>0.0025010343</td><td>0.9601260</td><td>0.9883650</td><td> 9</td><td>0.9908667</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 62</td></tr>\n",
       "\t<tr><td> 0.065684116</td><td>16.13165</td><td>0.0064392222</td><td>0.9360617</td><td>0.9883650</td><td>19</td><td>0.9908667</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 60</td></tr>\n",
       "\t<tr><td> 0.009593624</td><td>16.08802</td><td>0.0001311165</td><td>0.9908667</td><td>0.9908667</td><td>34</td><td>0.9908667</td><td>none</td><td>Unassigned2__Unassigned1</td><td>1</td><td> 59</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 9329 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " logFC & logCPM & F & PValue & FDR & Nhood & SpatialFDR & polarity & labels & labels\\_fraction & n\\_cells\\\\\n",
       " <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t  5.175528 & 12.75419 & 65.874861 & 2.712965e-11 & 6.592504e-09 &  63 & 6.856218e-09 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 129\\\\\n",
       "\t  4.883802 & 12.97953 & 50.940929 & 1.324004e-09 & 1.608665e-07 &  93 & 1.592806e-07 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 166\\\\\n",
       "\t  3.576113 & 12.47521 & 33.105950 & 2.990955e-07 & 2.422673e-05 & 214 & 2.576412e-05 & pos & a1\\_Tany\\_\\_Astrocytes & 1 &  99\\\\\n",
       "\t  3.401746 & 12.73492 & 31.074340 & 5.936490e-07 & 3.606418e-05 & 118 & 3.781441e-05 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 135\\\\\n",
       "\t  3.177160 & 13.10128 & 23.561109 & 8.712993e-06 & 4.234514e-04 &  19 & 4.260118e-04 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 201\\\\\n",
       "\t  2.661283 & 13.13157 & 21.175818 & 2.161802e-05 & 8.755300e-04 &  58 & 8.618807e-04 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 212\\\\\n",
       "\t -3.989157 & 12.23840 & 20.289935 & 3.052978e-05 & 1.059819e-03 & 200 & 1.070892e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 &  97\\\\\n",
       "\t  2.180194 & 12.94578 & 17.482410 & 9.390194e-05 & 2.852272e-03 & 108 & 2.821480e-03 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 174\\\\\n",
       "\t -3.177985 & 12.75694 & 17.077316 & 1.108626e-04 & 2.993290e-03 &  66 & 2.903301e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 180\\\\\n",
       "\t -2.306381 & 13.04335 & 15.070167 & 2.563977e-04 & 6.230464e-03 &  57 & 5.944409e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 233\\\\\n",
       "\t  2.370110 & 12.87178 & 14.500229 & 3.269528e-04 & 7.222684e-03 &  92 & 6.870984e-03 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 166\\\\\n",
       "\t  2.036045 & 12.92883 & 14.204678 & 3.712173e-04 & 7.517151e-03 &  33 & 7.127028e-03 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 173\\\\\n",
       "\t -2.827849 & 12.15033 & 13.213011 & 5.710871e-04 & 8.371871e-03 &  29 & 7.963183e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 &  86\\\\\n",
       "\t  2.386494 & 12.55891 & 13.160541 & 5.843741e-04 & 8.371871e-03 &  99 & 7.963183e-03 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 121\\\\\n",
       "\t -2.417376 & 12.76277 & 13.025250 & 6.201386e-04 & 8.371871e-03 & 136 & 7.963183e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 183\\\\\n",
       "\t  2.133462 & 12.55683 & 13.398511 & 5.265784e-04 & 8.371871e-03 & 160 & 7.963183e-03 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 120\\\\\n",
       "\t  1.793196 & 12.79742 & 13.085623 & 6.039056e-04 & 8.371871e-03 & 165 & 7.963183e-03 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 152\\\\\n",
       "\t -3.428830 & 12.09588 & 13.131769 & 6.111538e-04 & 8.371871e-03 & 190 & 7.963183e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 &  82\\\\\n",
       "\t -2.503424 & 12.49668 & 12.618147 & 7.421542e-04 & 8.587784e-03 &  94 & 8.122160e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 129\\\\\n",
       "\t -3.382299 & 12.32172 & 12.777130 & 7.133974e-04 & 8.587784e-03 & 218 & 8.122160e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 111\\\\\n",
       "\t -2.850700 & 12.33923 & 12.697946 & 7.164053e-04 & 8.587784e-03 & 231 & 8.122160e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 110\\\\\n",
       "\t -2.279203 & 13.20301 & 12.282190 & 8.615916e-04 & 9.516671e-03 & 163 & 8.940797e-03 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 262\\\\\n",
       "\t -2.839669 & 12.35562 & 11.474459 & 1.270543e-03 & 1.332662e-02 &  68 & 1.265509e-02 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 113\\\\\n",
       "\t -4.272532 & 12.12827 & 11.578755 & 1.316209e-03 & 1.332662e-02 &  86 & 1.266800e-02 & neg & a1\\_Tany\\_\\_Astrocytes & 1 &  81\\\\\n",
       "\t -5.742286 & 12.18465 & 11.334774 & 1.899448e-03 & 1.846263e-02 & 172 & 1.764270e-02 & neg & a1\\_Tany\\_\\_Astrocytes & 1 &  84\\\\\n",
       "\t  1.908189 & 12.39935 & 10.380648 & 2.041626e-03 & 1.908135e-02 & 242 & 1.831509e-02 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 104\\\\\n",
       "\t  2.181414 & 12.75725 & 10.160352 & 2.260996e-03 & 2.034896e-02 &  85 & 1.948854e-02 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 153\\\\\n",
       "\t  1.850450 & 12.41509 &  9.962382 & 2.479158e-03 & 2.151555e-02 &  43 & 2.082839e-02 & pos & a1\\_Tany\\_\\_Astrocytes & 1 & 103\\\\\n",
       "\t -2.174831 & 12.48496 &  9.757053 & 2.728800e-03 & 2.286546e-02 & 204 & 2.206125e-02 & neg & a1\\_Tany\\_\\_Astrocytes & 1 & 135\\\\\n",
       "\t  2.095793 & 12.24768 &  9.223508 & 3.572167e-03 & 2.851732e-02 &  39 & 2.767170e-02 & pos & a1\\_Tany\\_\\_Astrocytes & 1 &  85\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t  0.555631832 & 16.08223 & 0.5982580947 & 0.4405852 & 0.8242820 &  1 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  73\\\\\n",
       "\t -0.431842341 & 16.45066 & 0.3727559459 & 0.5416710 & 0.8242820 &  2 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 & 100\\\\\n",
       "\t  0.551295037 & 16.21552 & 0.5076834501 & 0.4764868 & 0.8242820 &  3 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  87\\\\\n",
       "\t  0.700026594 & 15.96530 & 0.6221776818 & 0.4304618 & 0.8242820 &  4 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  54\\\\\n",
       "\t  0.678766967 & 16.00659 & 0.7110638566 & 0.3993309 & 0.8242820 &  5 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  67\\\\\n",
       "\t  0.525560713 & 16.35272 & 0.4401633100 & 0.5072257 & 0.8242820 &  6 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  79\\\\\n",
       "\t  0.822935497 & 16.01523 & 0.6680645342 & 0.4139590 & 0.8242820 &  7 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  51\\\\\n",
       "\t -0.930457943 & 16.06482 & 1.8719123752 & 0.1716221 & 0.8242820 &  8 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  54\\\\\n",
       "\t  0.987269282 & 16.00727 & 1.2107956711 & 0.2714898 & 0.8242820 & 10 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  62\\\\\n",
       "\t -0.818123984 & 15.83660 & 0.9477328371 & 0.3305773 & 0.8242820 & 12 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  48\\\\\n",
       "\t  0.798806086 & 16.16020 & 1.1569668441 & 0.2824043 & 0.8242820 & 14 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  86\\\\\n",
       "\t -0.492302389 & 16.07432 & 0.5438116347 & 0.4610633 & 0.8242820 & 18 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  58\\\\\n",
       "\t  1.198919460 & 15.87275 & 1.6293390280 & 0.2021473 & 0.8242820 & 21 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  56\\\\\n",
       "\t  0.659272857 & 16.08645 & 0.9208048899 & 0.3375398 & 0.8242820 & 22 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  69\\\\\n",
       "\t  0.740889110 & 15.94444 & 0.7664462981 & 0.3815689 & 0.8242820 & 23 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  60\\\\\n",
       "\t  1.191291104 & 16.02297 & 2.0260281015 & 0.1549956 & 0.8242820 & 25 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  73\\\\\n",
       "\t  0.564786684 & 15.85552 & 0.3849789677 & 0.5351188 & 0.8242820 & 32 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  46\\\\\n",
       "\t  0.994682963 & 16.00759 & 1.3349327380 & 0.2482581 & 0.8242820 & 35 & 0.8253250 & none & Unassigned2\\_\\_Unassigned1 & 1 &  69\\\\\n",
       "\t -0.406219846 & 16.27201 & 0.3062747452 & 0.5801225 & 0.8460119 & 17 & 0.8485754 & none & Unassigned2\\_\\_Unassigned1 & 1 &  68\\\\\n",
       "\t  0.388868523 & 16.02849 & 0.2318672920 & 0.6302681 & 0.8823754 & 27 & 0.8811158 & none & Unassigned2\\_\\_Unassigned1 & 1 &  65\\\\\n",
       "\t  0.283796137 & 15.93773 & 0.1165066440 & 0.7329400 & 0.9524475 & 28 & 0.9559703 & none & Unassigned2\\_\\_Unassigned1 & 1 &  54\\\\\n",
       "\t  0.267700697 & 16.15524 & 0.1148753312 & 0.7347452 & 0.9524475 & 30 & 0.9559703 & none & Unassigned2\\_\\_Unassigned1 & 1 &  65\\\\\n",
       "\t  0.223272009 & 16.06103 & 0.0872889023 & 0.7677255 & 0.9553561 & 16 & 0.9585071 & none & Unassigned2\\_\\_Unassigned1 & 1 &  57\\\\\n",
       "\t  0.427225223 & 16.07938 & 0.3455399138 & 0.7915807 & 0.9553561 & 26 & 0.9585071 & none & Unassigned2\\_\\_Unassigned1 & 1 &  54\\\\\n",
       "\t -0.103405228 & 16.17067 & 0.0193652356 & 0.8893578 & 0.9727351 & 11 & 0.9718001 & none & Unassigned2\\_\\_Unassigned1 & 1 &  72\\\\\n",
       "\t -0.121413863 & 16.08554 & 0.0350256697 & 0.8515871 & 0.9727351 & 24 & 0.9718001 & none & Unassigned2\\_\\_Unassigned1 & 1 &  65\\\\\n",
       "\t  0.105516676 & 16.10242 & 0.0219448092 & 0.8822696 & 0.9727351 & 31 & 0.9718001 & none & Unassigned2\\_\\_Unassigned1 & 1 &  61\\\\\n",
       "\t  0.046865954 & 16.17896 & 0.0025010343 & 0.9601260 & 0.9883650 &  9 & 0.9908667 & none & Unassigned2\\_\\_Unassigned1 & 1 &  62\\\\\n",
       "\t  0.065684116 & 16.13165 & 0.0064392222 & 0.9360617 & 0.9883650 & 19 & 0.9908667 & none & Unassigned2\\_\\_Unassigned1 & 1 &  60\\\\\n",
       "\t  0.009593624 & 16.08802 & 0.0001311165 & 0.9908667 & 0.9908667 & 34 & 0.9908667 & none & Unassigned2\\_\\_Unassigned1 & 1 &  59\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 9329 × 11\n",
       "\n",
       "| logFC &lt;dbl&gt; | logCPM &lt;dbl&gt; | F &lt;dbl&gt; | PValue &lt;dbl&gt; | FDR &lt;dbl&gt; | Nhood &lt;dbl&gt; | SpatialFDR &lt;dbl&gt; | polarity &lt;chr&gt; | labels &lt;chr&gt; | labels_fraction &lt;dbl&gt; | n_cells &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "|  5.175528 | 12.75419 | 65.874861 | 2.712965e-11 | 6.592504e-09 |  63 | 6.856218e-09 | pos | a1_Tany__Astrocytes | 1 | 129 |\n",
       "|  4.883802 | 12.97953 | 50.940929 | 1.324004e-09 | 1.608665e-07 |  93 | 1.592806e-07 | pos | a1_Tany__Astrocytes | 1 | 166 |\n",
       "|  3.576113 | 12.47521 | 33.105950 | 2.990955e-07 | 2.422673e-05 | 214 | 2.576412e-05 | pos | a1_Tany__Astrocytes | 1 |  99 |\n",
       "|  3.401746 | 12.73492 | 31.074340 | 5.936490e-07 | 3.606418e-05 | 118 | 3.781441e-05 | pos | a1_Tany__Astrocytes | 1 | 135 |\n",
       "|  3.177160 | 13.10128 | 23.561109 | 8.712993e-06 | 4.234514e-04 |  19 | 4.260118e-04 | pos | a1_Tany__Astrocytes | 1 | 201 |\n",
       "|  2.661283 | 13.13157 | 21.175818 | 2.161802e-05 | 8.755300e-04 |  58 | 8.618807e-04 | pos | a1_Tany__Astrocytes | 1 | 212 |\n",
       "| -3.989157 | 12.23840 | 20.289935 | 3.052978e-05 | 1.059819e-03 | 200 | 1.070892e-03 | neg | a1_Tany__Astrocytes | 1 |  97 |\n",
       "|  2.180194 | 12.94578 | 17.482410 | 9.390194e-05 | 2.852272e-03 | 108 | 2.821480e-03 | pos | a1_Tany__Astrocytes | 1 | 174 |\n",
       "| -3.177985 | 12.75694 | 17.077316 | 1.108626e-04 | 2.993290e-03 |  66 | 2.903301e-03 | neg | a1_Tany__Astrocytes | 1 | 180 |\n",
       "| -2.306381 | 13.04335 | 15.070167 | 2.563977e-04 | 6.230464e-03 |  57 | 5.944409e-03 | neg | a1_Tany__Astrocytes | 1 | 233 |\n",
       "|  2.370110 | 12.87178 | 14.500229 | 3.269528e-04 | 7.222684e-03 |  92 | 6.870984e-03 | pos | a1_Tany__Astrocytes | 1 | 166 |\n",
       "|  2.036045 | 12.92883 | 14.204678 | 3.712173e-04 | 7.517151e-03 |  33 | 7.127028e-03 | pos | a1_Tany__Astrocytes | 1 | 173 |\n",
       "| -2.827849 | 12.15033 | 13.213011 | 5.710871e-04 | 8.371871e-03 |  29 | 7.963183e-03 | neg | a1_Tany__Astrocytes | 1 |  86 |\n",
       "|  2.386494 | 12.55891 | 13.160541 | 5.843741e-04 | 8.371871e-03 |  99 | 7.963183e-03 | pos | a1_Tany__Astrocytes | 1 | 121 |\n",
       "| -2.417376 | 12.76277 | 13.025250 | 6.201386e-04 | 8.371871e-03 | 136 | 7.963183e-03 | neg | a1_Tany__Astrocytes | 1 | 183 |\n",
       "|  2.133462 | 12.55683 | 13.398511 | 5.265784e-04 | 8.371871e-03 | 160 | 7.963183e-03 | pos | a1_Tany__Astrocytes | 1 | 120 |\n",
       "|  1.793196 | 12.79742 | 13.085623 | 6.039056e-04 | 8.371871e-03 | 165 | 7.963183e-03 | pos | a1_Tany__Astrocytes | 1 | 152 |\n",
       "| -3.428830 | 12.09588 | 13.131769 | 6.111538e-04 | 8.371871e-03 | 190 | 7.963183e-03 | neg | a1_Tany__Astrocytes | 1 |  82 |\n",
       "| -2.503424 | 12.49668 | 12.618147 | 7.421542e-04 | 8.587784e-03 |  94 | 8.122160e-03 | neg | a1_Tany__Astrocytes | 1 | 129 |\n",
       "| -3.382299 | 12.32172 | 12.777130 | 7.133974e-04 | 8.587784e-03 | 218 | 8.122160e-03 | neg | a1_Tany__Astrocytes | 1 | 111 |\n",
       "| -2.850700 | 12.33923 | 12.697946 | 7.164053e-04 | 8.587784e-03 | 231 | 8.122160e-03 | neg | a1_Tany__Astrocytes | 1 | 110 |\n",
       "| -2.279203 | 13.20301 | 12.282190 | 8.615916e-04 | 9.516671e-03 | 163 | 8.940797e-03 | neg | a1_Tany__Astrocytes | 1 | 262 |\n",
       "| -2.839669 | 12.35562 | 11.474459 | 1.270543e-03 | 1.332662e-02 |  68 | 1.265509e-02 | neg | a1_Tany__Astrocytes | 1 | 113 |\n",
       "| -4.272532 | 12.12827 | 11.578755 | 1.316209e-03 | 1.332662e-02 |  86 | 1.266800e-02 | neg | a1_Tany__Astrocytes | 1 |  81 |\n",
       "| -5.742286 | 12.18465 | 11.334774 | 1.899448e-03 | 1.846263e-02 | 172 | 1.764270e-02 | neg | a1_Tany__Astrocytes | 1 |  84 |\n",
       "|  1.908189 | 12.39935 | 10.380648 | 2.041626e-03 | 1.908135e-02 | 242 | 1.831509e-02 | pos | a1_Tany__Astrocytes | 1 | 104 |\n",
       "|  2.181414 | 12.75725 | 10.160352 | 2.260996e-03 | 2.034896e-02 |  85 | 1.948854e-02 | pos | a1_Tany__Astrocytes | 1 | 153 |\n",
       "|  1.850450 | 12.41509 |  9.962382 | 2.479158e-03 | 2.151555e-02 |  43 | 2.082839e-02 | pos | a1_Tany__Astrocytes | 1 | 103 |\n",
       "| -2.174831 | 12.48496 |  9.757053 | 2.728800e-03 | 2.286546e-02 | 204 | 2.206125e-02 | neg | a1_Tany__Astrocytes | 1 | 135 |\n",
       "|  2.095793 | 12.24768 |  9.223508 | 3.572167e-03 | 2.851732e-02 |  39 | 2.767170e-02 | pos | a1_Tany__Astrocytes | 1 |  85 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "|  0.555631832 | 16.08223 | 0.5982580947 | 0.4405852 | 0.8242820 |  1 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  73 |\n",
       "| -0.431842341 | 16.45066 | 0.3727559459 | 0.5416710 | 0.8242820 |  2 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 | 100 |\n",
       "|  0.551295037 | 16.21552 | 0.5076834501 | 0.4764868 | 0.8242820 |  3 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  87 |\n",
       "|  0.700026594 | 15.96530 | 0.6221776818 | 0.4304618 | 0.8242820 |  4 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  54 |\n",
       "|  0.678766967 | 16.00659 | 0.7110638566 | 0.3993309 | 0.8242820 |  5 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  67 |\n",
       "|  0.525560713 | 16.35272 | 0.4401633100 | 0.5072257 | 0.8242820 |  6 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  79 |\n",
       "|  0.822935497 | 16.01523 | 0.6680645342 | 0.4139590 | 0.8242820 |  7 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  51 |\n",
       "| -0.930457943 | 16.06482 | 1.8719123752 | 0.1716221 | 0.8242820 |  8 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  54 |\n",
       "|  0.987269282 | 16.00727 | 1.2107956711 | 0.2714898 | 0.8242820 | 10 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  62 |\n",
       "| -0.818123984 | 15.83660 | 0.9477328371 | 0.3305773 | 0.8242820 | 12 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  48 |\n",
       "|  0.798806086 | 16.16020 | 1.1569668441 | 0.2824043 | 0.8242820 | 14 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  86 |\n",
       "| -0.492302389 | 16.07432 | 0.5438116347 | 0.4610633 | 0.8242820 | 18 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  58 |\n",
       "|  1.198919460 | 15.87275 | 1.6293390280 | 0.2021473 | 0.8242820 | 21 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  56 |\n",
       "|  0.659272857 | 16.08645 | 0.9208048899 | 0.3375398 | 0.8242820 | 22 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  69 |\n",
       "|  0.740889110 | 15.94444 | 0.7664462981 | 0.3815689 | 0.8242820 | 23 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  60 |\n",
       "|  1.191291104 | 16.02297 | 2.0260281015 | 0.1549956 | 0.8242820 | 25 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  73 |\n",
       "|  0.564786684 | 15.85552 | 0.3849789677 | 0.5351188 | 0.8242820 | 32 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  46 |\n",
       "|  0.994682963 | 16.00759 | 1.3349327380 | 0.2482581 | 0.8242820 | 35 | 0.8253250 | none | Unassigned2__Unassigned1 | 1 |  69 |\n",
       "| -0.406219846 | 16.27201 | 0.3062747452 | 0.5801225 | 0.8460119 | 17 | 0.8485754 | none | Unassigned2__Unassigned1 | 1 |  68 |\n",
       "|  0.388868523 | 16.02849 | 0.2318672920 | 0.6302681 | 0.8823754 | 27 | 0.8811158 | none | Unassigned2__Unassigned1 | 1 |  65 |\n",
       "|  0.283796137 | 15.93773 | 0.1165066440 | 0.7329400 | 0.9524475 | 28 | 0.9559703 | none | Unassigned2__Unassigned1 | 1 |  54 |\n",
       "|  0.267700697 | 16.15524 | 0.1148753312 | 0.7347452 | 0.9524475 | 30 | 0.9559703 | none | Unassigned2__Unassigned1 | 1 |  65 |\n",
       "|  0.223272009 | 16.06103 | 0.0872889023 | 0.7677255 | 0.9553561 | 16 | 0.9585071 | none | Unassigned2__Unassigned1 | 1 |  57 |\n",
       "|  0.427225223 | 16.07938 | 0.3455399138 | 0.7915807 | 0.9553561 | 26 | 0.9585071 | none | Unassigned2__Unassigned1 | 1 |  54 |\n",
       "| -0.103405228 | 16.17067 | 0.0193652356 | 0.8893578 | 0.9727351 | 11 | 0.9718001 | none | Unassigned2__Unassigned1 | 1 |  72 |\n",
       "| -0.121413863 | 16.08554 | 0.0350256697 | 0.8515871 | 0.9727351 | 24 | 0.9718001 | none | Unassigned2__Unassigned1 | 1 |  65 |\n",
       "|  0.105516676 | 16.10242 | 0.0219448092 | 0.8822696 | 0.9727351 | 31 | 0.9718001 | none | Unassigned2__Unassigned1 | 1 |  61 |\n",
       "|  0.046865954 | 16.17896 | 0.0025010343 | 0.9601260 | 0.9883650 |  9 | 0.9908667 | none | Unassigned2__Unassigned1 | 1 |  62 |\n",
       "|  0.065684116 | 16.13165 | 0.0064392222 | 0.9360617 | 0.9883650 | 19 | 0.9908667 | none | Unassigned2__Unassigned1 | 1 |  60 |\n",
       "|  0.009593624 | 16.08802 | 0.0001311165 | 0.9908667 | 0.9908667 | 34 | 0.9908667 | none | Unassigned2__Unassigned1 | 1 |  59 |\n",
       "\n"
      ],
      "text/plain": [
       "     logFC        logCPM   F            PValue       FDR          Nhood\n",
       "1     5.175528    12.75419 65.874861    2.712965e-11 6.592504e-09  63  \n",
       "2     4.883802    12.97953 50.940929    1.324004e-09 1.608665e-07  93  \n",
       "3     3.576113    12.47521 33.105950    2.990955e-07 2.422673e-05 214  \n",
       "4     3.401746    12.73492 31.074340    5.936490e-07 3.606418e-05 118  \n",
       "5     3.177160    13.10128 23.561109    8.712993e-06 4.234514e-04  19  \n",
       "6     2.661283    13.13157 21.175818    2.161802e-05 8.755300e-04  58  \n",
       "7    -3.989157    12.23840 20.289935    3.052978e-05 1.059819e-03 200  \n",
       "8     2.180194    12.94578 17.482410    9.390194e-05 2.852272e-03 108  \n",
       "9    -3.177985    12.75694 17.077316    1.108626e-04 2.993290e-03  66  \n",
       "10   -2.306381    13.04335 15.070167    2.563977e-04 6.230464e-03  57  \n",
       "11    2.370110    12.87178 14.500229    3.269528e-04 7.222684e-03  92  \n",
       "12    2.036045    12.92883 14.204678    3.712173e-04 7.517151e-03  33  \n",
       "13   -2.827849    12.15033 13.213011    5.710871e-04 8.371871e-03  29  \n",
       "14    2.386494    12.55891 13.160541    5.843741e-04 8.371871e-03  99  \n",
       "15   -2.417376    12.76277 13.025250    6.201386e-04 8.371871e-03 136  \n",
       "16    2.133462    12.55683 13.398511    5.265784e-04 8.371871e-03 160  \n",
       "17    1.793196    12.79742 13.085623    6.039056e-04 8.371871e-03 165  \n",
       "18   -3.428830    12.09588 13.131769    6.111538e-04 8.371871e-03 190  \n",
       "19   -2.503424    12.49668 12.618147    7.421542e-04 8.587784e-03  94  \n",
       "20   -3.382299    12.32172 12.777130    7.133974e-04 8.587784e-03 218  \n",
       "21   -2.850700    12.33923 12.697946    7.164053e-04 8.587784e-03 231  \n",
       "22   -2.279203    13.20301 12.282190    8.615916e-04 9.516671e-03 163  \n",
       "23   -2.839669    12.35562 11.474459    1.270543e-03 1.332662e-02  68  \n",
       "24   -4.272532    12.12827 11.578755    1.316209e-03 1.332662e-02  86  \n",
       "25   -5.742286    12.18465 11.334774    1.899448e-03 1.846263e-02 172  \n",
       "26    1.908189    12.39935 10.380648    2.041626e-03 1.908135e-02 242  \n",
       "27    2.181414    12.75725 10.160352    2.260996e-03 2.034896e-02  85  \n",
       "28    1.850450    12.41509  9.962382    2.479158e-03 2.151555e-02  43  \n",
       "29   -2.174831    12.48496  9.757053    2.728800e-03 2.286546e-02 204  \n",
       "30    2.095793    12.24768  9.223508    3.572167e-03 2.851732e-02  39  \n",
       "⋮    ⋮            ⋮        ⋮            ⋮            ⋮            ⋮    \n",
       "9300  0.555631832 16.08223 0.5982580947 0.4405852    0.8242820     1   \n",
       "9301 -0.431842341 16.45066 0.3727559459 0.5416710    0.8242820     2   \n",
       "9302  0.551295037 16.21552 0.5076834501 0.4764868    0.8242820     3   \n",
       "9303  0.700026594 15.96530 0.6221776818 0.4304618    0.8242820     4   \n",
       "9304  0.678766967 16.00659 0.7110638566 0.3993309    0.8242820     5   \n",
       "9305  0.525560713 16.35272 0.4401633100 0.5072257    0.8242820     6   \n",
       "9306  0.822935497 16.01523 0.6680645342 0.4139590    0.8242820     7   \n",
       "9307 -0.930457943 16.06482 1.8719123752 0.1716221    0.8242820     8   \n",
       "9308  0.987269282 16.00727 1.2107956711 0.2714898    0.8242820    10   \n",
       "9309 -0.818123984 15.83660 0.9477328371 0.3305773    0.8242820    12   \n",
       "9310  0.798806086 16.16020 1.1569668441 0.2824043    0.8242820    14   \n",
       "9311 -0.492302389 16.07432 0.5438116347 0.4610633    0.8242820    18   \n",
       "9312  1.198919460 15.87275 1.6293390280 0.2021473    0.8242820    21   \n",
       "9313  0.659272857 16.08645 0.9208048899 0.3375398    0.8242820    22   \n",
       "9314  0.740889110 15.94444 0.7664462981 0.3815689    0.8242820    23   \n",
       "9315  1.191291104 16.02297 2.0260281015 0.1549956    0.8242820    25   \n",
       "9316  0.564786684 15.85552 0.3849789677 0.5351188    0.8242820    32   \n",
       "9317  0.994682963 16.00759 1.3349327380 0.2482581    0.8242820    35   \n",
       "9318 -0.406219846 16.27201 0.3062747452 0.5801225    0.8460119    17   \n",
       "9319  0.388868523 16.02849 0.2318672920 0.6302681    0.8823754    27   \n",
       "9320  0.283796137 15.93773 0.1165066440 0.7329400    0.9524475    28   \n",
       "9321  0.267700697 16.15524 0.1148753312 0.7347452    0.9524475    30   \n",
       "9322  0.223272009 16.06103 0.0872889023 0.7677255    0.9553561    16   \n",
       "9323  0.427225223 16.07938 0.3455399138 0.7915807    0.9553561    26   \n",
       "9324 -0.103405228 16.17067 0.0193652356 0.8893578    0.9727351    11   \n",
       "9325 -0.121413863 16.08554 0.0350256697 0.8515871    0.9727351    24   \n",
       "9326  0.105516676 16.10242 0.0219448092 0.8822696    0.9727351    31   \n",
       "9327  0.046865954 16.17896 0.0025010343 0.9601260    0.9883650     9   \n",
       "9328  0.065684116 16.13165 0.0064392222 0.9360617    0.9883650    19   \n",
       "9329  0.009593624 16.08802 0.0001311165 0.9908667    0.9908667    34   \n",
       "     SpatialFDR   polarity labels                   labels_fraction n_cells\n",
       "1    6.856218e-09 pos      a1_Tany__Astrocytes      1               129    \n",
       "2    1.592806e-07 pos      a1_Tany__Astrocytes      1               166    \n",
       "3    2.576412e-05 pos      a1_Tany__Astrocytes      1                99    \n",
       "4    3.781441e-05 pos      a1_Tany__Astrocytes      1               135    \n",
       "5    4.260118e-04 pos      a1_Tany__Astrocytes      1               201    \n",
       "6    8.618807e-04 pos      a1_Tany__Astrocytes      1               212    \n",
       "7    1.070892e-03 neg      a1_Tany__Astrocytes      1                97    \n",
       "8    2.821480e-03 pos      a1_Tany__Astrocytes      1               174    \n",
       "9    2.903301e-03 neg      a1_Tany__Astrocytes      1               180    \n",
       "10   5.944409e-03 neg      a1_Tany__Astrocytes      1               233    \n",
       "11   6.870984e-03 pos      a1_Tany__Astrocytes      1               166    \n",
       "12   7.127028e-03 pos      a1_Tany__Astrocytes      1               173    \n",
       "13   7.963183e-03 neg      a1_Tany__Astrocytes      1                86    \n",
       "14   7.963183e-03 pos      a1_Tany__Astrocytes      1               121    \n",
       "15   7.963183e-03 neg      a1_Tany__Astrocytes      1               183    \n",
       "16   7.963183e-03 pos      a1_Tany__Astrocytes      1               120    \n",
       "17   7.963183e-03 pos      a1_Tany__Astrocytes      1               152    \n",
       "18   7.963183e-03 neg      a1_Tany__Astrocytes      1                82    \n",
       "19   8.122160e-03 neg      a1_Tany__Astrocytes      1               129    \n",
       "20   8.122160e-03 neg      a1_Tany__Astrocytes      1               111    \n",
       "21   8.122160e-03 neg      a1_Tany__Astrocytes      1               110    \n",
       "22   8.940797e-03 neg      a1_Tany__Astrocytes      1               262    \n",
       "23   1.265509e-02 neg      a1_Tany__Astrocytes      1               113    \n",
       "24   1.266800e-02 neg      a1_Tany__Astrocytes      1                81    \n",
       "25   1.764270e-02 neg      a1_Tany__Astrocytes      1                84    \n",
       "26   1.831509e-02 pos      a1_Tany__Astrocytes      1               104    \n",
       "27   1.948854e-02 pos      a1_Tany__Astrocytes      1               153    \n",
       "28   2.082839e-02 pos      a1_Tany__Astrocytes      1               103    \n",
       "29   2.206125e-02 neg      a1_Tany__Astrocytes      1               135    \n",
       "30   2.767170e-02 pos      a1_Tany__Astrocytes      1                85    \n",
       "⋮    ⋮            ⋮        ⋮                        ⋮               ⋮      \n",
       "9300 0.8253250    none     Unassigned2__Unassigned1 1                73    \n",
       "9301 0.8253250    none     Unassigned2__Unassigned1 1               100    \n",
       "9302 0.8253250    none     Unassigned2__Unassigned1 1                87    \n",
       "9303 0.8253250    none     Unassigned2__Unassigned1 1                54    \n",
       "9304 0.8253250    none     Unassigned2__Unassigned1 1                67    \n",
       "9305 0.8253250    none     Unassigned2__Unassigned1 1                79    \n",
       "9306 0.8253250    none     Unassigned2__Unassigned1 1                51    \n",
       "9307 0.8253250    none     Unassigned2__Unassigned1 1                54    \n",
       "9308 0.8253250    none     Unassigned2__Unassigned1 1                62    \n",
       "9309 0.8253250    none     Unassigned2__Unassigned1 1                48    \n",
       "9310 0.8253250    none     Unassigned2__Unassigned1 1                86    \n",
       "9311 0.8253250    none     Unassigned2__Unassigned1 1                58    \n",
       "9312 0.8253250    none     Unassigned2__Unassigned1 1                56    \n",
       "9313 0.8253250    none     Unassigned2__Unassigned1 1                69    \n",
       "9314 0.8253250    none     Unassigned2__Unassigned1 1                60    \n",
       "9315 0.8253250    none     Unassigned2__Unassigned1 1                73    \n",
       "9316 0.8253250    none     Unassigned2__Unassigned1 1                46    \n",
       "9317 0.8253250    none     Unassigned2__Unassigned1 1                69    \n",
       "9318 0.8485754    none     Unassigned2__Unassigned1 1                68    \n",
       "9319 0.8811158    none     Unassigned2__Unassigned1 1                65    \n",
       "9320 0.9559703    none     Unassigned2__Unassigned1 1                54    \n",
       "9321 0.9559703    none     Unassigned2__Unassigned1 1                65    \n",
       "9322 0.9585071    none     Unassigned2__Unassigned1 1                57    \n",
       "9323 0.9585071    none     Unassigned2__Unassigned1 1                54    \n",
       "9324 0.9718001    none     Unassigned2__Unassigned1 1                72    \n",
       "9325 0.9718001    none     Unassigned2__Unassigned1 1                65    \n",
       "9326 0.9718001    none     Unassigned2__Unassigned1 1                61    \n",
       "9327 0.9908667    none     Unassigned2__Unassigned1 1                62    \n",
       "9328 0.9908667    none     Unassigned2__Unassigned1 1                60    \n",
       "9329 0.9908667    none     Unassigned2__Unassigned1 1                59    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "combined_da_results_all.obob5v5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "be670463",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: edgeR\n",
      "\n",
      "Loading required package: limma\n",
      "\n",
      "\n",
      "Attaching package: 'limma'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:BiocGenerics':\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'edgeR'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:SingleCellExperiment':\n",
      "\n",
      "    cpm\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(miloR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "378c4c57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for findNhoodGroupMarkers {miloR}\"><tr><td>findNhoodGroupMarkers {miloR}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Identify post-hoc neighbourhood marker genes</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>This function will perform differential gene expression analysis on\n",
       "groups of neighbourhoods. Adjacent and concordantly DA neighbourhoods can be defined using\n",
       "<code>groupNhoods</code> or by the user. Cells <em>between</em> these\n",
       "aggregated groups are compared. For differential gene experession based on an input design\n",
       "<em>within</em> DA neighbourhoods see <code>testDiffExp</code>.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "findNhoodGroupMarkers(\n",
       "  x,\n",
       "  da.res,\n",
       "  assay = \"logcounts\",\n",
       "  aggregate.samples = FALSE,\n",
       "  sample_col = NULL,\n",
       "  subset.row = NULL,\n",
       "  gene.offset = TRUE,\n",
       "  subset.nhoods = NULL,\n",
       "  subset.groups = NULL,\n",
       "  na.function = \"na.pass\"\n",
       ")\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>x</code></td>\n",
       "<td>\n",
       "<p>A <code>Milo</code> object containing single-cell gene expression\n",
       "and neighbourhoods.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>da.res</code></td>\n",
       "<td>\n",
       "<p>A <code>data.frame</code> containing DA results, as expected from running\n",
       "<code>testNhoods</code>, as a <code>NhoodGroup</code> column specifying the grouping of neighbourhoods,\n",
       "as expected from</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>assay</code></td>\n",
       "<td>\n",
       "<p>A character scalar determining which <code>assays</code> slot to extract from the\n",
       "<code>Milo</code> object to use for DGE testing.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>aggregate.samples</code></td>\n",
       "<td>\n",
       "<p>logical indicating wheather the expression values for cells in the same sample\n",
       "and neighbourhood group should be merged for DGE testing. This allows to perform testing exploiting the replication structure\n",
       "in the experimental design, rather than treating single-cells as independent replicates. The function used for aggregation depends on the\n",
       "selected gene expression assay: if <code>assay=\"counts\"</code> the expression values are summed, otherwise we take the mean.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>sample_col</code></td>\n",
       "<td>\n",
       "<p>a character scalar indicating the column in the colData storing sample information\n",
       "(only relevant if <code>aggregate.samples==TRUE</code>)</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset.row</code></td>\n",
       "<td>\n",
       "<p>A logical, integer or character vector indicating the rows\n",
       "of <code>x</code> to use for sumamrizing over cells in neighbourhoods.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>gene.offset</code></td>\n",
       "<td>\n",
       "<p>A logical scalar the determines whether a per-cell offset\n",
       "is provided in the DGE GLM to adjust for the number of detected genes with\n",
       "expression &gt; 0.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset.nhoods</code></td>\n",
       "<td>\n",
       "<p>A logical, integer or character vector indicating which neighbourhoods\n",
       "to subset before aggregation and DGE testing (default: NULL).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset.groups</code></td>\n",
       "<td>\n",
       "<p>A character vector indicating which groups to test for markers (default: NULL)</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>na.function</code></td>\n",
       "<td>\n",
       "<p>A valid NA action function to apply, should be one of\n",
       "<code>na.fail, na.omit, na.exclude, na.pass</code>.</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p>Using a one vs. all approach, each aggregated group of cells is compared to all others\n",
       "using the single-cell log normalized gene expression with a GLM\n",
       "(for details see <code>limma-package</code>), or the single-cell counts using a\n",
       "negative binomial GLM (for details see <code>edgeR-package</code>). When using\n",
       "the latter it is recommended to set <code>gene.offset=TRUE</code> as this behaviour adjusts\n",
       "the model offsets by the number of detected genes in each cell.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<p>A <code>data.frame</code> of DGE results containing a log fold change and adjusted\n",
       "p-value for each aggregated group of neighbourhoods. If <code>return.groups</code> then\n",
       "the return value is a list with the slots <code>groups</code> and <code>dge</code> containing the\n",
       "aggregated neighbourhood groups per single-cell and marker gene results, respectively.\n",
       "</p>\n",
       "<p><em>Warning</em>: If all neighbourhoods are grouped together, then it is impossible to\n",
       "run <code>findNhoodMarkers</code>. In this (hopefully rare) instance, this function will return\n",
       "a warning and return <code>NULL</code>.\n",
       "</p>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>miloR</em> version 1.3.1 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{findNhoodGroupMarkers}{Identify post-hoc neighbourhood marker genes}{findNhoodGroupMarkers}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "This function will perform differential gene expression analysis on\n",
       "groups of neighbourhoods. Adjacent and concordantly DA neighbourhoods can be defined using\n",
       "\\code{groupNhoods} or by the user. Cells \\emph{between} these\n",
       "aggregated groups are compared. For differential gene experession based on an input design\n",
       "\\emph{within} DA neighbourhoods see \\code{\\LinkA{testDiffExp}{testDiffExp}}.\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "findNhoodGroupMarkers(\n",
       "  x,\n",
       "  da.res,\n",
       "  assay = \"logcounts\",\n",
       "  aggregate.samples = FALSE,\n",
       "  sample_col = NULL,\n",
       "  subset.row = NULL,\n",
       "  gene.offset = TRUE,\n",
       "  subset.nhoods = NULL,\n",
       "  subset.groups = NULL,\n",
       "  na.function = \"na.pass\"\n",
       ")\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{x}] A \\code{\\LinkA{Milo}{Milo.Rdash.class}} object containing single-cell gene expression\n",
       "and neighbourhoods.\n",
       "\n",
       "\\item[\\code{da.res}] A \\code{data.frame} containing DA results, as expected from running\n",
       "\\code{testNhoods}, as a \\code{NhoodGroup} column specifying the grouping of neighbourhoods,\n",
       "as expected from\n",
       "\n",
       "\\item[\\code{assay}] A character scalar determining which \\code{assays} slot to extract from the\n",
       "\\code{\\LinkA{Milo}{Milo.Rdash.class}} object to use for DGE testing.\n",
       "\n",
       "\\item[\\code{aggregate.samples}] logical indicating wheather the expression values for cells in the same sample\n",
       "and neighbourhood group should be merged for DGE testing. This allows to perform testing exploiting the replication structure\n",
       "in the experimental design, rather than treating single-cells as independent replicates. The function used for aggregation depends on the\n",
       "selected gene expression assay: if \\code{assay=\"counts\"} the expression values are summed, otherwise we take the mean.\n",
       "\n",
       "\\item[\\code{sample\\_col}] a character scalar indicating the column in the colData storing sample information\n",
       "(only relevant if \\code{aggregate.samples==TRUE})\n",
       "\n",
       "\\item[\\code{subset.row}] A logical, integer or character vector indicating the rows\n",
       "of \\code{x} to use for sumamrizing over cells in neighbourhoods.\n",
       "\n",
       "\\item[\\code{gene.offset}] A logical scalar the determines whether a per-cell offset\n",
       "is provided in the DGE GLM to adjust for the number of detected genes with\n",
       "expression > 0.\n",
       "\n",
       "\\item[\\code{subset.nhoods}] A logical, integer or character vector indicating which neighbourhoods\n",
       "to subset before aggregation and DGE testing (default: NULL).\n",
       "\n",
       "\\item[\\code{subset.groups}] A character vector indicating which groups to test for markers (default: NULL)\n",
       "\n",
       "\\item[\\code{na.function}] A valid NA action function to apply, should be one of\n",
       "\\code{na.fail, na.omit, na.exclude, na.pass}.\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "Using a one vs. all approach, each aggregated group of cells is compared to all others\n",
       "using the single-cell log normalized gene expression with a GLM\n",
       "(for details see \\code{\\LinkA{limma-package}{limma.Rdash.package}}), or the single-cell counts using a\n",
       "negative binomial GLM (for details see \\code{\\LinkA{edgeR-package}{edgeR.Rdash.package}}). When using\n",
       "the latter it is recommended to set \\code{gene.offset=TRUE} as this behaviour adjusts\n",
       "the model offsets by the number of detected genes in each cell.\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Value}\n",
       "A \\code{data.frame} of DGE results containing a log fold change and adjusted\n",
       "p-value for each aggregated group of neighbourhoods. If \\code{return.groups} then\n",
       "the return value is a list with the slots \\code{groups} and \\code{dge} containing the\n",
       "aggregated neighbourhood groups per single-cell and marker gene results, respectively.\n",
       "\n",
       "\\emph{Warning}: If all neighbourhoods are grouped together, then it is impossible to\n",
       "run \\code{findNhoodMarkers}. In this (hopefully rare) instance, this function will return\n",
       "a warning and return \\code{NULL}.\n",
       "\\end{Value}"
      ],
      "text/plain": [
       "findNhoodGroupMarkers          package:miloR           R Documentation\n",
       "\n",
       "_\bI_\bd_\be_\bn_\bt_\bi_\bf_\by _\bp_\bo_\bs_\bt-_\bh_\bo_\bc _\bn_\be_\bi_\bg_\bh_\bb_\bo_\bu_\br_\bh_\bo_\bo_\bd _\bm_\ba_\br_\bk_\be_\br _\bg_\be_\bn_\be_\bs\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     This function will perform differential gene expression analysis\n",
       "     on groups of neighbourhoods. Adjacent and concordantly DA\n",
       "     neighbourhoods can be defined using 'groupNhoods' or by the user.\n",
       "     Cells _between_ these aggregated groups are compared. For\n",
       "     differential gene experession based on an input design _within_ DA\n",
       "     neighbourhoods see 'testDiffExp'.\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     findNhoodGroupMarkers(\n",
       "       x,\n",
       "       da.res,\n",
       "       assay = \"logcounts\",\n",
       "       aggregate.samples = FALSE,\n",
       "       sample_col = NULL,\n",
       "       subset.row = NULL,\n",
       "       gene.offset = TRUE,\n",
       "       subset.nhoods = NULL,\n",
       "       subset.groups = NULL,\n",
       "       na.function = \"na.pass\"\n",
       "     )\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "       x: A 'Milo' object containing single-cell gene expression and\n",
       "          neighbourhoods.\n",
       "\n",
       "  da.res: A 'data.frame' containing DA results, as expected from\n",
       "          running 'testNhoods', as a 'NhoodGroup' column specifying the\n",
       "          grouping of neighbourhoods, as expected from\n",
       "\n",
       "   assay: A character scalar determining which 'assays' slot to extract\n",
       "          from the 'Milo' object to use for DGE testing.\n",
       "\n",
       "aggregate.samples: logical indicating wheather the expression values\n",
       "          for cells in the same sample and neighbourhood group should\n",
       "          be merged for DGE testing. This allows to perform testing\n",
       "          exploiting the replication structure in the experimental\n",
       "          design, rather than treating single-cells as independent\n",
       "          replicates. The function used for aggregation depends on the\n",
       "          selected gene expression assay: if 'assay=\"counts\"' the\n",
       "          expression values are summed, otherwise we take the mean.\n",
       "\n",
       "sample_col: a character scalar indicating the column in the colData\n",
       "          storing sample information (only relevant if\n",
       "          'aggregate.samples==TRUE')\n",
       "\n",
       "subset.row: A logical, integer or character vector indicating the rows\n",
       "          of 'x' to use for sumamrizing over cells in neighbourhoods.\n",
       "\n",
       "gene.offset: A logical scalar the determines whether a per-cell offset\n",
       "          is provided in the DGE GLM to adjust for the number of\n",
       "          detected genes with expression > 0.\n",
       "\n",
       "subset.nhoods: A logical, integer or character vector indicating which\n",
       "          neighbourhoods to subset before aggregation and DGE testing\n",
       "          (default: NULL).\n",
       "\n",
       "subset.groups: A character vector indicating which groups to test for\n",
       "          markers (default: NULL)\n",
       "\n",
       "na.function: A valid NA action function to apply, should be one of\n",
       "          'na.fail, na.omit, na.exclude, na.pass'.\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     Using a one vs. all approach, each aggregated group of cells is\n",
       "     compared to all others using the single-cell log normalized gene\n",
       "     expression with a GLM (for details see 'limma-package'), or the\n",
       "     single-cell counts using a negative binomial GLM (for details see\n",
       "     'edgeR-package'). When using the latter it is recommended to set\n",
       "     'gene.offset=TRUE' as this behaviour adjusts the model offsets by\n",
       "     the number of detected genes in each cell.\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     A 'data.frame' of DGE results containing a log fold change and\n",
       "     adjusted p-value for each aggregated group of neighbourhoods. If\n",
       "     'return.groups' then the return value is a list with the slots\n",
       "     'groups' and 'dge' containing the aggregated neighbourhood groups\n",
       "     per single-cell and marker gene results, respectively.\n",
       "\n",
       "     _Warning_: If all neighbourhoods are grouped together, then it is\n",
       "     impossible to run 'findNhoodMarkers'. In this (hopefully rare)\n",
       "     instance, this function will return a warning and return 'NULL'.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?findNhoodGroupMarkers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "7eadb0aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for findNhoodMarkers {miloR}\"><tr><td>findNhoodMarkers {miloR}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Identify post-hoc neighbourhood marker genes</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>This function will perform differential gene expression analysis on\n",
       "differentially abundant neighbourhoods, by first aggregating adjacent and\n",
       "concordantly DA neighbourhoods, then comparing cells <em>between</em> these\n",
       "aggregated groups. For differential gene experession based on an input design\n",
       "<em>within</em> DA neighbourhoods see <code>testDiffExp</code>.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>x</code></td>\n",
       "<td>\n",
       "<p>A <code>Milo</code> object containing single-cell gene expression\n",
       "and neighbourhoods.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>da.res</code></td>\n",
       "<td>\n",
       "<p>A <code>data.frame</code> containing DA results, as expected from running\n",
       "<code>testNhoods</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>da.fdr</code></td>\n",
       "<td>\n",
       "<p>A numeric scalar that determines at what FDR neighbourhoods are declared\n",
       "DA for the purposes of aggregating across concorantly DA neighbourhoods.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>assay</code></td>\n",
       "<td>\n",
       "<p>A character scalar determining which <code>assays</code> slot to extract from the\n",
       "<code>Milo</code> object to use for DGE testing.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>aggregate.samples</code></td>\n",
       "<td>\n",
       "<p>logical indicating wheather the expression values for cells in the same sample\n",
       "and neighbourhood group should be merged for DGE testing. This allows to perform testing exploiting the replication structure\n",
       "in the experimental design, rather than treating single-cells as independent replicates. The function used for aggregation depends on the\n",
       "selected gene expression assay: if <code>assay=\"counts\"</code> the expression values are summed, otherwise we take the mean.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>sample_col</code></td>\n",
       "<td>\n",
       "<p>a character scalar indicating the column in the colData storing sample information\n",
       "(only relevant if <code>aggregate.samples==TRUE</code>)</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>overlap</code></td>\n",
       "<td>\n",
       "<p>A scalar integer that determines the number of cells that must\n",
       "overlap between adjacent neighbourhoods for merging.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>lfc.threshold</code></td>\n",
       "<td>\n",
       "<p>A scalar that determines the absolute log fold change above\n",
       "which neighbourhoods should be considerd 'DA' for merging. Default=NULL</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>merge.discord</code></td>\n",
       "<td>\n",
       "<p>A logical scalar that overrides the default behaviour and allows\n",
       "adjacent neighbourhoods to be merged if they have discordant log fold change signs. Using\n",
       "this argument is generally discouraged, but may be useful for constructing an empirical null\n",
       "group of cells, regardless of DA sign.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset.row</code></td>\n",
       "<td>\n",
       "<p>A logical, integer or character vector indicating the rows\n",
       "of <code>x</code> to use for sumamrizing over cells in neighbourhoods.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>gene.offset</code></td>\n",
       "<td>\n",
       "<p>A logical scalar the determines whether a per-cell offset\n",
       "is provided in the DGE GLM to adjust for the number of detected genes with\n",
       "expression &gt; 0.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>return.groups</code></td>\n",
       "<td>\n",
       "<p>A logical scalar that returns a <code>data.frame</code> of the\n",
       "aggregated groups per single-cell. Cells that are members of non-DA neighbourhoods contain\n",
       "<code>NA</code> values.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset.nhoods</code></td>\n",
       "<td>\n",
       "<p>A logical, integer or character vector indicating which neighbourhoods\n",
       "to subset before aggregation and DGE testing.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>na.function</code></td>\n",
       "<td>\n",
       "<p>A valid NA action function to apply, should be one of\n",
       "<code>na.fail, na.omit, na.exclude, na.pass</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>compute.new</code></td>\n",
       "<td>\n",
       "<p>A logical scalar indicating whether to force computing a new neighbourhood\n",
       "adjacency matrix if already present.</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p>Louvain clustering is applied to the neighbourhood graph. This graph is first modified\n",
       "based on two criteria: 1) neighbourhoods share at least <code>overlap</code> number of cells,\n",
       "and 2) the DA log fold change sign is concordant.\n",
       "This behaviour can be modulated by setting <code>overlap</code> to be more or less stringent.\n",
       "Additionally, a threshold on the log fold-changes can be set, such that <code>lfc.threshold</code>\n",
       "is required to retain edges between adjacent neighbourhoods. Note: adjacent neighbourhoods will\n",
       "never be merged with opposite signs.\n",
       "</p>\n",
       "<p>Using a one vs. all approach, each aggregated group of cells is compared to all others\n",
       "using the single-cell log normalized gene expression with a GLM\n",
       "(for details see <code>limma-package</code>), or the single-cell counts using a\n",
       "negative binomial GLM (for details see <code>edgeR-package</code>). When using\n",
       "the latter it is recommended to set <code>gene.offset=TRUE</code> as this behaviour adjusts\n",
       "the model offsets by the number of detected genes in each cell.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<p>A <code>data.frame</code> of DGE results containing a log fold change and adjusted\n",
       "p-value for each aggregated group of neighbourhoods. If <code>return.groups</code> then\n",
       "the return value is a list with the slots <code>groups</code> and <code>dge</code> containing the\n",
       "aggregated neighbourhood groups per single-cell and marker gene results, respectively.\n",
       "</p>\n",
       "<p><em>Warning</em>: If all neighbourhoods are grouped together, then it is impossible to\n",
       "run <code>findNhoodMarkers</code>. In this (hopefully rare) instance, this function will return\n",
       "a warning and return <code>NULL</code>.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Author(s)</h3>\n",
       "\n",
       "<p>Mike Morgan &amp; Emma Dann\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "library(SingleCellExperiment)\n",
       "ux.1 &lt;- matrix(rpois(12000, 5), ncol=400)\n",
       "ux.2 &lt;- matrix(rpois(12000, 4), ncol=400)\n",
       "ux &lt;- rbind(ux.1, ux.2)\n",
       "vx &lt;- log2(ux + 1)\n",
       "pca &lt;- prcomp(t(vx))\n",
       "\n",
       "sce &lt;- SingleCellExperiment(assays=list(counts=ux, logcounts=vx),\n",
       "                            reducedDims=SimpleList(PCA=pca$x))\n",
       "colnames(sce) &lt;- paste0(\"Cell\", seq_len(ncol(sce)))\n",
       "milo &lt;- Milo(sce)\n",
       "milo &lt;- buildGraph(milo, k=20, d=10, transposed=TRUE)\n",
       "milo &lt;- makeNhoods(milo, k=20, d=10, prop=0.3)\n",
       "milo &lt;- calcNhoodDistance(milo, d=10)\n",
       "\n",
       "cond &lt;- rep(\"A\", ncol(milo))\n",
       "cond.a &lt;- sample(seq_len(ncol(milo)), size=floor(ncol(milo)*0.25))\n",
       "cond.b &lt;- setdiff(seq_len(ncol(milo)), cond.a)\n",
       "cond[cond.b] &lt;- \"B\"\n",
       "meta.df &lt;- data.frame(Condition=cond, Replicate=c(rep(\"R1\", 132), rep(\"R2\", 132), rep(\"R3\", 136)))\n",
       "meta.df$SampID &lt;- paste(meta.df$Condition, meta.df$Replicate, sep=\"_\")\n",
       "milo &lt;- countCells(milo, meta.data=meta.df, samples=\"SampID\")\n",
       "\n",
       "test.meta &lt;- data.frame(\"Condition\"=c(rep(\"A\", 3), rep(\"B\", 3)), \"Replicate\"=rep(c(\"R1\", \"R2\", \"R3\"), 2))\n",
       "test.meta$Sample &lt;- paste(test.meta$Condition, test.meta$Replicate, sep=\"_\")\n",
       "rownames(test.meta) &lt;- test.meta$Sample\n",
       "da.res &lt;- testNhoods(milo, design=~0 + Condition, design.df=test.meta[colnames(nhoodCounts(milo)), ])\n",
       "\n",
       "nhood.dge &lt;- findNhoodMarkers(milo, da.res, overlap=1, compute.new=TRUE)\n",
       "nhood.dge\n",
       "\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>miloR</em> version 1.3.1 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{findNhoodMarkers}{Identify post-hoc neighbourhood marker genes}{findNhoodMarkers}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "This function will perform differential gene expression analysis on\n",
       "differentially abundant neighbourhoods, by first aggregating adjacent and\n",
       "concordantly DA neighbourhoods, then comparing cells \\emph{between} these\n",
       "aggregated groups. For differential gene experession based on an input design\n",
       "\\emph{within} DA neighbourhoods see \\code{\\LinkA{testDiffExp}{testDiffExp}}.\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{x}] A \\code{\\LinkA{Milo}{Milo.Rdash.class}} object containing single-cell gene expression\n",
       "and neighbourhoods.\n",
       "\n",
       "\\item[\\code{da.res}] A \\code{data.frame} containing DA results, as expected from running\n",
       "\\code{testNhoods}.\n",
       "\n",
       "\\item[\\code{da.fdr}] A numeric scalar that determines at what FDR neighbourhoods are declared\n",
       "DA for the purposes of aggregating across concorantly DA neighbourhoods.\n",
       "\n",
       "\\item[\\code{assay}] A character scalar determining which \\code{assays} slot to extract from the\n",
       "\\code{\\LinkA{Milo}{Milo.Rdash.class}} object to use for DGE testing.\n",
       "\n",
       "\\item[\\code{aggregate.samples}] logical indicating wheather the expression values for cells in the same sample\n",
       "and neighbourhood group should be merged for DGE testing. This allows to perform testing exploiting the replication structure\n",
       "in the experimental design, rather than treating single-cells as independent replicates. The function used for aggregation depends on the\n",
       "selected gene expression assay: if \\code{assay=\"counts\"} the expression values are summed, otherwise we take the mean.\n",
       "\n",
       "\\item[\\code{sample\\_col}] a character scalar indicating the column in the colData storing sample information\n",
       "(only relevant if \\code{aggregate.samples==TRUE})\n",
       "\n",
       "\\item[\\code{overlap}] A scalar integer that determines the number of cells that must\n",
       "overlap between adjacent neighbourhoods for merging.\n",
       "\n",
       "\\item[\\code{lfc.threshold}] A scalar that determines the absolute log fold change above\n",
       "which neighbourhoods should be considerd 'DA' for merging. Default=NULL\n",
       "\n",
       "\\item[\\code{merge.discord}] A logical scalar that overrides the default behaviour and allows\n",
       "adjacent neighbourhoods to be merged if they have discordant log fold change signs. Using\n",
       "this argument is generally discouraged, but may be useful for constructing an empirical null\n",
       "group of cells, regardless of DA sign.\n",
       "\n",
       "\\item[\\code{subset.row}] A logical, integer or character vector indicating the rows\n",
       "of \\code{x} to use for sumamrizing over cells in neighbourhoods.\n",
       "\n",
       "\\item[\\code{gene.offset}] A logical scalar the determines whether a per-cell offset\n",
       "is provided in the DGE GLM to adjust for the number of detected genes with\n",
       "expression > 0.\n",
       "\n",
       "\\item[\\code{return.groups}] A logical scalar that returns a \\code{\\LinkA{data.frame}{data.frame}} of the\n",
       "aggregated groups per single-cell. Cells that are members of non-DA neighbourhoods contain\n",
       "\\code{NA} values.\n",
       "\n",
       "\\item[\\code{subset.nhoods}] A logical, integer or character vector indicating which neighbourhoods\n",
       "to subset before aggregation and DGE testing.\n",
       "\n",
       "\\item[\\code{na.function}] A valid NA action function to apply, should be one of\n",
       "\\code{na.fail, na.omit, na.exclude, na.pass}.\n",
       "\n",
       "\\item[\\code{compute.new}] A logical scalar indicating whether to force computing a new neighbourhood\n",
       "adjacency matrix if already present.\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "Louvain clustering is applied to the neighbourhood graph. This graph is first modified\n",
       "based on two criteria: 1) neighbourhoods share at least \\code{overlap} number of cells,\n",
       "and 2) the DA log fold change sign is concordant.\n",
       "This behaviour can be modulated by setting \\code{overlap} to be more or less stringent.\n",
       "Additionally, a threshold on the log fold-changes can be set, such that \\code{lfc.threshold}\n",
       "is required to retain edges between adjacent neighbourhoods. Note: adjacent neighbourhoods will\n",
       "never be merged with opposite signs.\n",
       "\n",
       "Using a one vs. all approach, each aggregated group of cells is compared to all others\n",
       "using the single-cell log normalized gene expression with a GLM\n",
       "(for details see \\code{\\LinkA{limma-package}{limma.Rdash.package}}), or the single-cell counts using a\n",
       "negative binomial GLM (for details see \\code{\\LinkA{edgeR-package}{edgeR.Rdash.package}}). When using\n",
       "the latter it is recommended to set \\code{gene.offset=TRUE} as this behaviour adjusts\n",
       "the model offsets by the number of detected genes in each cell.\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Value}\n",
       "A \\code{data.frame} of DGE results containing a log fold change and adjusted\n",
       "p-value for each aggregated group of neighbourhoods. If \\code{return.groups} then\n",
       "the return value is a list with the slots \\code{groups} and \\code{dge} containing the\n",
       "aggregated neighbourhood groups per single-cell and marker gene results, respectively.\n",
       "\n",
       "\\emph{Warning}: If all neighbourhoods are grouped together, then it is impossible to\n",
       "run \\code{findNhoodMarkers}. In this (hopefully rare) instance, this function will return\n",
       "a warning and return \\code{NULL}.\n",
       "\\end{Value}\n",
       "%\n",
       "\\begin{Author}\\relax\n",
       "Mike Morgan \\& Emma Dann\n",
       "\\end{Author}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "library(SingleCellExperiment)\n",
       "ux.1 <- matrix(rpois(12000, 5), ncol=400)\n",
       "ux.2 <- matrix(rpois(12000, 4), ncol=400)\n",
       "ux <- rbind(ux.1, ux.2)\n",
       "vx <- log2(ux + 1)\n",
       "pca <- prcomp(t(vx))\n",
       "\n",
       "sce <- SingleCellExperiment(assays=list(counts=ux, logcounts=vx),\n",
       "                            reducedDims=SimpleList(PCA=pca$x))\n",
       "colnames(sce) <- paste0(\"Cell\", seq_len(ncol(sce)))\n",
       "milo <- Milo(sce)\n",
       "milo <- buildGraph(milo, k=20, d=10, transposed=TRUE)\n",
       "milo <- makeNhoods(milo, k=20, d=10, prop=0.3)\n",
       "milo <- calcNhoodDistance(milo, d=10)\n",
       "\n",
       "cond <- rep(\"A\", ncol(milo))\n",
       "cond.a <- sample(seq_len(ncol(milo)), size=floor(ncol(milo)*0.25))\n",
       "cond.b <- setdiff(seq_len(ncol(milo)), cond.a)\n",
       "cond[cond.b] <- \"B\"\n",
       "meta.df <- data.frame(Condition=cond, Replicate=c(rep(\"R1\", 132), rep(\"R2\", 132), rep(\"R3\", 136)))\n",
       "meta.df$SampID <- paste(meta.df$Condition, meta.df$Replicate, sep=\"_\")\n",
       "milo <- countCells(milo, meta.data=meta.df, samples=\"SampID\")\n",
       "\n",
       "test.meta <- data.frame(\"Condition\"=c(rep(\"A\", 3), rep(\"B\", 3)), \"Replicate\"=rep(c(\"R1\", \"R2\", \"R3\"), 2))\n",
       "test.meta$Sample <- paste(test.meta$Condition, test.meta$Replicate, sep=\"_\")\n",
       "rownames(test.meta) <- test.meta$Sample\n",
       "da.res <- testNhoods(milo, design=~0 + Condition, design.df=test.meta[colnames(nhoodCounts(milo)), ])\n",
       "\n",
       "nhood.dge <- findNhoodMarkers(milo, da.res, overlap=1, compute.new=TRUE)\n",
       "nhood.dge\n",
       "\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "findNhoodMarkers             package:miloR             R Documentation\n",
       "\n",
       "_\bI_\bd_\be_\bn_\bt_\bi_\bf_\by _\bp_\bo_\bs_\bt-_\bh_\bo_\bc _\bn_\be_\bi_\bg_\bh_\bb_\bo_\bu_\br_\bh_\bo_\bo_\bd _\bm_\ba_\br_\bk_\be_\br _\bg_\be_\bn_\be_\bs\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     This function will perform differential gene expression analysis\n",
       "     on differentially abundant neighbourhoods, by first aggregating\n",
       "     adjacent and concordantly DA neighbourhoods, then comparing cells\n",
       "     _between_ these aggregated groups. For differential gene\n",
       "     experession based on an input design _within_ DA neighbourhoods\n",
       "     see 'testDiffExp'.\n",
       "\n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "       x: A 'Milo' object containing single-cell gene expression and\n",
       "          neighbourhoods.\n",
       "\n",
       "  da.res: A 'data.frame' containing DA results, as expected from\n",
       "          running 'testNhoods'.\n",
       "\n",
       "  da.fdr: A numeric scalar that determines at what FDR neighbourhoods\n",
       "          are declared DA for the purposes of aggregating across\n",
       "          concorantly DA neighbourhoods.\n",
       "\n",
       "   assay: A character scalar determining which 'assays' slot to extract\n",
       "          from the 'Milo' object to use for DGE testing.\n",
       "\n",
       "aggregate.samples: logical indicating wheather the expression values\n",
       "          for cells in the same sample and neighbourhood group should\n",
       "          be merged for DGE testing. This allows to perform testing\n",
       "          exploiting the replication structure in the experimental\n",
       "          design, rather than treating single-cells as independent\n",
       "          replicates. The function used for aggregation depends on the\n",
       "          selected gene expression assay: if 'assay=\"counts\"' the\n",
       "          expression values are summed, otherwise we take the mean.\n",
       "\n",
       "sample_col: a character scalar indicating the column in the colData\n",
       "          storing sample information (only relevant if\n",
       "          'aggregate.samples==TRUE')\n",
       "\n",
       " overlap: A scalar integer that determines the number of cells that\n",
       "          must overlap between adjacent neighbourhoods for merging.\n",
       "\n",
       "lfc.threshold: A scalar that determines the absolute log fold change\n",
       "          above which neighbourhoods should be considerd 'DA' for\n",
       "          merging. Default=NULL\n",
       "\n",
       "merge.discord: A logical scalar that overrides the default behaviour\n",
       "          and allows adjacent neighbourhoods to be merged if they have\n",
       "          discordant log fold change signs. Using this argument is\n",
       "          generally discouraged, but may be useful for constructing an\n",
       "          empirical null group of cells, regardless of DA sign.\n",
       "\n",
       "subset.row: A logical, integer or character vector indicating the rows\n",
       "          of 'x' to use for sumamrizing over cells in neighbourhoods.\n",
       "\n",
       "gene.offset: A logical scalar the determines whether a per-cell offset\n",
       "          is provided in the DGE GLM to adjust for the number of\n",
       "          detected genes with expression > 0.\n",
       "\n",
       "return.groups: A logical scalar that returns a 'data.frame' of the\n",
       "          aggregated groups per single-cell. Cells that are members of\n",
       "          non-DA neighbourhoods contain 'NA' values.\n",
       "\n",
       "subset.nhoods: A logical, integer or character vector indicating which\n",
       "          neighbourhoods to subset before aggregation and DGE testing.\n",
       "\n",
       "na.function: A valid NA action function to apply, should be one of\n",
       "          'na.fail, na.omit, na.exclude, na.pass'.\n",
       "\n",
       "compute.new: A logical scalar indicating whether to force computing a\n",
       "          new neighbourhood adjacency matrix if already present.\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     Louvain clustering is applied to the neighbourhood graph. This\n",
       "     graph is first modified based on two criteria: 1) neighbourhoods\n",
       "     share at least 'overlap' number of cells, and 2) the DA log fold\n",
       "     change sign is concordant. This behaviour can be modulated by\n",
       "     setting 'overlap' to be more or less stringent. Additionally, a\n",
       "     threshold on the log fold-changes can be set, such that\n",
       "     'lfc.threshold' is required to retain edges between adjacent\n",
       "     neighbourhoods. Note: adjacent neighbourhoods will never be merged\n",
       "     with opposite signs.\n",
       "\n",
       "     Using a one vs. all approach, each aggregated group of cells is\n",
       "     compared to all others using the single-cell log normalized gene\n",
       "     expression with a GLM (for details see 'limma-package'), or the\n",
       "     single-cell counts using a negative binomial GLM (for details see\n",
       "     'edgeR-package'). When using the latter it is recommended to set\n",
       "     'gene.offset=TRUE' as this behaviour adjusts the model offsets by\n",
       "     the number of detected genes in each cell.\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     A 'data.frame' of DGE results containing a log fold change and\n",
       "     adjusted p-value for each aggregated group of neighbourhoods. If\n",
       "     'return.groups' then the return value is a list with the slots\n",
       "     'groups' and 'dge' containing the aggregated neighbourhood groups\n",
       "     per single-cell and marker gene results, respectively.\n",
       "\n",
       "     _Warning_: If all neighbourhoods are grouped together, then it is\n",
       "     impossible to run 'findNhoodMarkers'. In this (hopefully rare)\n",
       "     instance, this function will return a warning and return 'NULL'.\n",
       "\n",
       "_\bA_\bu_\bt_\bh_\bo_\br(_\bs):\n",
       "\n",
       "     Mike Morgan & Emma Dann\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     library(SingleCellExperiment)\n",
       "     ux.1 <- matrix(rpois(12000, 5), ncol=400)\n",
       "     ux.2 <- matrix(rpois(12000, 4), ncol=400)\n",
       "     ux <- rbind(ux.1, ux.2)\n",
       "     vx <- log2(ux + 1)\n",
       "     pca <- prcomp(t(vx))\n",
       "     \n",
       "     sce <- SingleCellExperiment(assays=list(counts=ux, logcounts=vx),\n",
       "                                 reducedDims=SimpleList(PCA=pca$x))\n",
       "     colnames(sce) <- paste0(\"Cell\", seq_len(ncol(sce)))\n",
       "     milo <- Milo(sce)\n",
       "     milo <- buildGraph(milo, k=20, d=10, transposed=TRUE)\n",
       "     milo <- makeNhoods(milo, k=20, d=10, prop=0.3)\n",
       "     milo <- calcNhoodDistance(milo, d=10)\n",
       "     \n",
       "     cond <- rep(\"A\", ncol(milo))\n",
       "     cond.a <- sample(seq_len(ncol(milo)), size=floor(ncol(milo)*0.25))\n",
       "     cond.b <- setdiff(seq_len(ncol(milo)), cond.a)\n",
       "     cond[cond.b] <- \"B\"\n",
       "     meta.df <- data.frame(Condition=cond, Replicate=c(rep(\"R1\", 132), rep(\"R2\", 132), rep(\"R3\", 136)))\n",
       "     meta.df$SampID <- paste(meta.df$Condition, meta.df$Replicate, sep=\"_\")\n",
       "     milo <- countCells(milo, meta.data=meta.df, samples=\"SampID\")\n",
       "     \n",
       "     test.meta <- data.frame(\"Condition\"=c(rep(\"A\", 3), rep(\"B\", 3)), \"Replicate\"=rep(c(\"R1\", \"R2\", \"R3\"), 2))\n",
       "     test.meta$Sample <- paste(test.meta$Condition, test.meta$Replicate, sep=\"_\")\n",
       "     rownames(test.meta) <- test.meta$Sample\n",
       "     da.res <- testNhoods(milo, design=~0 + Condition, design.df=test.meta[colnames(nhoodCounts(milo)), ])\n",
       "     \n",
       "     nhood.dge <- findNhoodMarkers(milo, da.res, overlap=1, compute.new=TRUE)\n",
       "     nhood.dge\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?findNhoodMarkers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57ab994c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:fgf1_ygg]",
   "language": "R",
   "name": "conda-env-fgf1_ygg-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
