{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ef416aca",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.2 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.4.2     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.2.1     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.1.2\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.1     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.2     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.2\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(writexl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8820dd69",
   "metadata": {},
   "outputs": [],
   "source": [
    "split_data_to_excel <- function(input_data, output_dir) {\n",
    "  # group the data by the \"tag\" column and create a list of data frames\n",
    "  grouped_data <- input_data %>% group_split(tag)\n",
    "\n",
    "  # loop through the groups and write each one to a separate Excel file in a subfolder for each combination of values\n",
    "  walk2(grouped_data, unique(input_data$tag), function(df, group_val) {\n",
    "    folder_name <- paste0(output_dir, \"/\", df$fgf1_day[1], \".\", df$fgf1_comparison[1], \"__v__\", df$bl6_day[1], \".\", df$bl6_comparison[1])\n",
    "    if (!dir.exists(folder_name)) {\n",
    "      dir.create(folder_name, recursive = TRUE)\n",
    "    }\n",
    "    write_xlsx(df, paste0(folder_name, \"/\", group_val, \".xlsx\"))\n",
    "  })\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ec50074a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>tag</th><th scope=col>data_day</th><th scope=col>cluster</th><th scope=col>fgf1_day</th><th scope=col>fgf1_comparison</th><th scope=col>bl6_day</th><th scope=col>bl6_comparison</th><th scope=col>grouping</th><th scope=col>cells_a</th><th scope=col>cells_b</th><th scope=col>GeneID</th><th scope=col>p_val</th><th scope=col>avg_log2FC</th><th scope=col>pct.1</th><th scope=col>pct.2</th><th scope=col>p_val_adj</th><th scope=col>ensmusg</th><th scope=col>gsea_sort_score</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>Sgcd  </td><td>1.006914e-19</td><td>0.4943435</td><td>0.729</td><td>0.618</td><td>1.934685e-15</td><td>ENSMUSG00000020354</td><td>7.273468</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>Kcnip4</td><td>3.665186e-14</td><td>0.6583016</td><td>0.705</td><td>0.649</td><td>7.042288e-10</td><td>ENSMUSG00000029088</td><td>6.024965</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>Zfhx3 </td><td>7.390493e-20</td><td>0.3635134</td><td>0.918</td><td>0.863</td><td>1.420009e-15</td><td>ENSMUSG00000038872</td><td>5.397341</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>Cadm1 </td><td>3.323447e-26</td><td>0.2413559</td><td>1.000</td><td>0.999</td><td>6.385672e-22</td><td>ENSMUSG00000032076</td><td>5.115489</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>Lama2 </td><td>2.002614e-18</td><td>0.3048799</td><td>0.878</td><td>0.763</td><td>3.847822e-14</td><td>ENSMUSG00000019899</td><td>4.089898</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>Gria2 </td><td>4.936700e-16</td><td>0.2764835</td><td>0.847</td><td>0.771</td><td>9.485375e-12</td><td>ENSMUSG00000033981</td><td>3.047662</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 18\n",
       "\\begin{tabular}{llllllllllllllllll}\n",
       " tag & data\\_day & cluster & fgf1\\_day & fgf1\\_comparison & bl6\\_day & bl6\\_comparison & grouping & cells\\_a & cells\\_b & GeneID & p\\_val & avg\\_log2FC & pct.1 & pct.2 & p\\_val\\_adj & ensmusg & gsea\\_sort\\_score\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & Sgcd   & 1.006914e-19 & 0.4943435 & 0.729 & 0.618 & 1.934685e-15 & ENSMUSG00000020354 & 7.273468\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & Kcnip4 & 3.665186e-14 & 0.6583016 & 0.705 & 0.649 & 7.042288e-10 & ENSMUSG00000029088 & 6.024965\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & Zfhx3  & 7.390493e-20 & 0.3635134 & 0.918 & 0.863 & 1.420009e-15 & ENSMUSG00000038872 & 5.397341\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & Cadm1  & 3.323447e-26 & 0.2413559 & 1.000 & 0.999 & 6.385672e-22 & ENSMUSG00000032076 & 5.115489\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & Lama2  & 2.002614e-18 & 0.3048799 & 0.878 & 0.763 & 3.847822e-14 & ENSMUSG00000019899 & 4.089898\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & Gria2  & 4.936700e-16 & 0.2764835 & 0.847 & 0.771 & 9.485375e-12 & ENSMUSG00000033981 & 3.047662\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 18\n",
       "\n",
       "| tag &lt;chr&gt; | data_day &lt;chr&gt; | cluster &lt;chr&gt; | fgf1_day &lt;chr&gt; | fgf1_comparison &lt;chr&gt; | bl6_day &lt;chr&gt; | bl6_comparison &lt;chr&gt; | grouping &lt;chr&gt; | cells_a &lt;chr&gt; | cells_b &lt;chr&gt; | GeneID &lt;chr&gt; | p_val &lt;dbl&gt; | avg_log2FC &lt;dbl&gt; | pct.1 &lt;dbl&gt; | pct.2 &lt;dbl&gt; | p_val_adj &lt;dbl&gt; | ensmusg &lt;chr&gt; | gsea_sort_score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | Sgcd   | 1.006914e-19 | 0.4943435 | 0.729 | 0.618 | 1.934685e-15 | ENSMUSG00000020354 | 7.273468 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | Kcnip4 | 3.665186e-14 | 0.6583016 | 0.705 | 0.649 | 7.042288e-10 | ENSMUSG00000029088 | 6.024965 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | Zfhx3  | 7.390493e-20 | 0.3635134 | 0.918 | 0.863 | 1.420009e-15 | ENSMUSG00000038872 | 5.397341 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | Cadm1  | 3.323447e-26 | 0.2413559 | 1.000 | 0.999 | 6.385672e-22 | ENSMUSG00000032076 | 5.115489 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | Lama2  | 2.002614e-18 | 0.3048799 | 0.878 | 0.763 | 3.847822e-14 | ENSMUSG00000019899 | 4.089898 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | Gria2  | 4.936700e-16 | 0.2764835 | 0.847 | 0.771 | 9.485375e-12 | ENSMUSG00000033981 | 3.047662 |\n",
       "\n"
      ],
      "text/plain": [
       "  tag                                                                             \n",
       "1 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "2 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "3 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "4 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "5 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "6 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "  data_day cluster             fgf1_day fgf1_comparison bl6_day bl6_comparison\n",
       "1 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "2 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "3 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "4 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "5 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "6 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "  grouping      cells_a cells_b GeneID p_val        avg_log2FC pct.1 pct.2\n",
       "1 fgf1_grouping neg     none    Sgcd   1.006914e-19 0.4943435  0.729 0.618\n",
       "2 fgf1_grouping neg     none    Kcnip4 3.665186e-14 0.6583016  0.705 0.649\n",
       "3 fgf1_grouping neg     none    Zfhx3  7.390493e-20 0.3635134  0.918 0.863\n",
       "4 fgf1_grouping neg     none    Cadm1  3.323447e-26 0.2413559  1.000 0.999\n",
       "5 fgf1_grouping neg     none    Lama2  2.002614e-18 0.3048799  0.878 0.763\n",
       "6 fgf1_grouping neg     none    Gria2  4.936700e-16 0.2764835  0.847 0.771\n",
       "  p_val_adj    ensmusg            gsea_sort_score\n",
       "1 1.934685e-15 ENSMUSG00000020354 7.273468       \n",
       "2 7.042288e-10 ENSMUSG00000029088 6.024965       \n",
       "3 1.420009e-15 ENSMUSG00000038872 5.397341       \n",
       "4 6.385672e-22 ENSMUSG00000032076 5.115489       \n",
       "5 3.847822e-14 ENSMUSG00000019899 4.089898       \n",
       "6 9.485375e-12 ENSMUSG00000033981 3.047662       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "combined_deg_seurat_formatted = qs::qread('_targets/objects/combined_deg_seurat_formatted')\n",
    "combined_deg_seurat_formatted %>% write_csv('combined_deg_seurat_formatted.csv')\n",
    "# split_data_to_excel(combined_deg_seurat_formatted, \"outputs/deg_seurat\")\n",
    "combined_deg_seurat_formatted %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "052a29ad",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea025958",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "28e510dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 21</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>tag</th><th scope=col>data_day</th><th scope=col>cluster</th><th scope=col>fgf1_day</th><th scope=col>fgf1_comparison</th><th scope=col>bl6_day</th><th scope=col>bl6_comparison</th><th scope=col>grouping</th><th scope=col>cells_a</th><th scope=col>cells_b</th><th scope=col>⋯</th><th scope=col>gs_subcat</th><th scope=col>pathway</th><th scope=col>gs_description</th><th scope=col>pval</th><th scope=col>padj</th><th scope=col>log2err</th><th scope=col>ES</th><th scope=col>NES</th><th scope=col>size</th><th scope=col>leadingEdge</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;list&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>⋯</td><td>GO:MF         </td><td>GOMF_PROTEIN_CONTAINING_COMPLEX_BINDING</td><td>Binding to a macromolecular complex. [GOC:jl]                                                                                                                                                                                                                                 </td><td>0.006235055</td><td>0.06235055</td><td>0.4070179</td><td>-0.8899058</td><td>-1.593066</td><td>16</td><td>ENSMUSG00000063063, ENSMUSG00000029381, ENSMUSG00000015354</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>rest</td><td>⋯</td><td>CGP           </td><td>BENPORATH_PRC2_TARGETS                 </td><td>Set 'PRC2 targets': Polycomb Repression Complex 2 (PRC) targets; identified by ChIP on chip on human embryonic stem cells as genes that: posess the trimethylated H3K27 mark in their promoters and are bound by SUZ12 [GeneID=23512] and EED [GeneID=8726] Polycomb proteins.</td><td>0.010384937</td><td>0.09078952</td><td>0.3807304</td><td> 0.7888536</td><td> 1.504499</td><td>16</td><td>ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000035131</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>rest</td><td>⋯</td><td>CGP           </td><td>BENPORATH_SUZ12_TARGETS                </td><td>Set 'Suz12 targets': genes identified by ChIP on chip as targets of the Polycomb protein SUZ12 [GeneID=23512] in human embryonic stem cells.                                                                                                                                  </td><td>0.002856450</td><td>0.07426771</td><td>0.4317077</td><td> 0.7565117</td><td> 1.609581</td><td>26</td><td>ENSMUSG00000029088, ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000031137, ENSMUSG00000020704, ENSMUSG00000035131</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>rest</td><td>⋯</td><td>CGP           </td><td>WONG_ADULT_TISSUE_STEM_MODULE          </td><td>The 'adult tissue stem' module: genes coordinately up-regulated in a compendium of adult tissue stem cells.                                                                                                                                                                   </td><td>0.010475714</td><td>0.09078952</td><td>0.3807304</td><td> 0.7654812</td><td> 1.532718</td><td>19</td><td>ENSMUSG00000032076, ENSMUSG00000033981, ENSMUSG00000019899, ENSMUSG00000048332, ENSMUSG00000024304</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>rest</td><td>⋯</td><td>MIR:MIR_Legacy</td><td>ACTGTGA_MIR27A_MIR27B                  </td><td>Genes having at least one occurence of the motif ACTGTGA in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNAs hsa-miR-27a and hsa-miR-27b (v7.1 miRBase).                                                       </td><td>0.006172517</td><td>0.01851755</td><td>0.4070179</td><td> 0.7914666</td><td> 1.517202</td><td>17</td><td>ENSMUSG00000032076, ENSMUSG00000025932, ENSMUSG00000036019, ENSMUSG00000020122</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>rest</td><td>⋯</td><td>TFT:GTRD      </td><td>MAFG_TARGET_GENES                      </td><td>Genes containing one or more binding sites for UniProt:O15525 (MAFG) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.                                                                                                </td><td>0.002768658</td><td>0.09136570</td><td>0.4317077</td><td>-0.8908226</td><td>-1.619861</td><td>17</td><td>ENSMUSG00000063063, ENSMUSG00000015354, ENSMUSG00000021136</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 21\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " tag & data\\_day & cluster & fgf1\\_day & fgf1\\_comparison & bl6\\_day & bl6\\_comparison & grouping & cells\\_a & cells\\_b & ⋯ & gs\\_subcat & pathway & gs\\_description & pval & padj & log2err & ES & NES & size & leadingEdge\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <list>\\\\\n",
       "\\hline\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & ⋯ & GO:MF          & GOMF\\_PROTEIN\\_CONTAINING\\_COMPLEX\\_BINDING & Binding to a macromolecular complex. {[}GOC:jl{]}                                                                                                                                                                                                                                  & 0.006235055 & 0.06235055 & 0.4070179 & -0.8899058 & -1.593066 & 16 & ENSMUSG00000063063, ENSMUSG00000029381, ENSMUSG00000015354\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_rest & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & rest & ⋯ & CGP            & BENPORATH\\_PRC2\\_TARGETS                  & Set 'PRC2 targets': Polycomb Repression Complex 2 (PRC) targets; identified by ChIP on chip on human embryonic stem cells as genes that: posess the trimethylated H3K27 mark in their promoters and are bound by SUZ12 {[}GeneID=23512{]} and EED {[}GeneID=8726{]} Polycomb proteins. & 0.010384937 & 0.09078952 & 0.3807304 &  0.7888536 &  1.504499 & 16 & ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000035131\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_rest & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & rest & ⋯ & CGP            & BENPORATH\\_SUZ12\\_TARGETS                 & Set 'Suz12 targets': genes identified by ChIP on chip as targets of the Polycomb protein SUZ12 {[}GeneID=23512{]} in human embryonic stem cells.                                                                                                                                   & 0.002856450 & 0.07426771 & 0.4317077 &  0.7565117 &  1.609581 & 26 & ENSMUSG00000029088, ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000031137, ENSMUSG00000020704, ENSMUSG00000035131\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_rest & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & rest & ⋯ & CGP            & WONG\\_ADULT\\_TISSUE\\_STEM\\_MODULE           & The 'adult tissue stem' module: genes coordinately up-regulated in a compendium of adult tissue stem cells.                                                                                                                                                                    & 0.010475714 & 0.09078952 & 0.3807304 &  0.7654812 &  1.532718 & 19 & ENSMUSG00000032076, ENSMUSG00000033981, ENSMUSG00000019899, ENSMUSG00000048332, ENSMUSG00000024304\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_rest & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & rest & ⋯ & MIR:MIR\\_Legacy & ACTGTGA\\_MIR27A\\_MIR27B                   & Genes having at least one occurence of the motif ACTGTGA in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNAs hsa-miR-27a and hsa-miR-27b (v7.1 miRBase).                                                        & 0.006172517 & 0.01851755 & 0.4070179 &  0.7914666 &  1.517202 & 17 & ENSMUSG00000032076, ENSMUSG00000025932, ENSMUSG00000036019, ENSMUSG00000020122\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_rest & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & rest & ⋯ & TFT:GTRD       & MAFG\\_TARGET\\_GENES                       & Genes containing one or more binding sites for UniProt:O15525 (MAFG) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.                                                                                                 & 0.002768658 & 0.09136570 & 0.4317077 & -0.8908226 & -1.619861 & 17 & ENSMUSG00000063063, ENSMUSG00000015354, ENSMUSG00000021136\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 21\n",
       "\n",
       "| tag &lt;chr&gt; | data_day &lt;chr&gt; | cluster &lt;chr&gt; | fgf1_day &lt;chr&gt; | fgf1_comparison &lt;chr&gt; | bl6_day &lt;chr&gt; | bl6_comparison &lt;chr&gt; | grouping &lt;chr&gt; | cells_a &lt;chr&gt; | cells_b &lt;chr&gt; | ⋯ ⋯ | gs_subcat &lt;chr&gt; | pathway &lt;chr&gt; | gs_description &lt;chr&gt; | pval &lt;dbl&gt; | padj &lt;dbl&gt; | log2err &lt;dbl&gt; | ES &lt;dbl&gt; | NES &lt;dbl&gt; | size &lt;int&gt; | leadingEdge &lt;list&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | ⋯ | GO:MF          | GOMF_PROTEIN_CONTAINING_COMPLEX_BINDING | Binding to a macromolecular complex. [GOC:jl]                                                                                                                                                                                                                                  | 0.006235055 | 0.06235055 | 0.4070179 | -0.8899058 | -1.593066 | 16 | ENSMUSG00000063063, ENSMUSG00000029381, ENSMUSG00000015354 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | rest | ⋯ | CGP            | BENPORATH_PRC2_TARGETS                  | Set 'PRC2 targets': Polycomb Repression Complex 2 (PRC) targets; identified by ChIP on chip on human embryonic stem cells as genes that: posess the trimethylated H3K27 mark in their promoters and are bound by SUZ12 [GeneID=23512] and EED [GeneID=8726] Polycomb proteins. | 0.010384937 | 0.09078952 | 0.3807304 |  0.7888536 |  1.504499 | 16 | ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000035131 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | rest | ⋯ | CGP            | BENPORATH_SUZ12_TARGETS                 | Set 'Suz12 targets': genes identified by ChIP on chip as targets of the Polycomb protein SUZ12 [GeneID=23512] in human embryonic stem cells.                                                                                                                                   | 0.002856450 | 0.07426771 | 0.4317077 |  0.7565117 |  1.609581 | 26 | ENSMUSG00000029088, ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000031137, ENSMUSG00000020704, ENSMUSG00000035131 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | rest | ⋯ | CGP            | WONG_ADULT_TISSUE_STEM_MODULE           | The 'adult tissue stem' module: genes coordinately up-regulated in a compendium of adult tissue stem cells.                                                                                                                                                                    | 0.010475714 | 0.09078952 | 0.3807304 |  0.7654812 |  1.532718 | 19 | ENSMUSG00000032076, ENSMUSG00000033981, ENSMUSG00000019899, ENSMUSG00000048332, ENSMUSG00000024304 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | rest | ⋯ | MIR:MIR_Legacy | ACTGTGA_MIR27A_MIR27B                   | Genes having at least one occurence of the motif ACTGTGA in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNAs hsa-miR-27a and hsa-miR-27b (v7.1 miRBase).                                                        | 0.006172517 | 0.01851755 | 0.4070179 |  0.7914666 |  1.517202 | 17 | ENSMUSG00000032076, ENSMUSG00000025932, ENSMUSG00000036019, ENSMUSG00000020122 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | rest | ⋯ | TFT:GTRD       | MAFG_TARGET_GENES                       | Genes containing one or more binding sites for UniProt:O15525 (MAFG) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.                                                                                                 | 0.002768658 | 0.09136570 | 0.4317077 | -0.8908226 | -1.619861 | 17 | ENSMUSG00000063063, ENSMUSG00000015354, ENSMUSG00000021136 |\n",
       "\n"
      ],
      "text/plain": [
       "  tag                                                                             \n",
       "1 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "2 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest\n",
       "3 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest\n",
       "4 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest\n",
       "5 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest\n",
       "6 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_rest\n",
       "  data_day cluster             fgf1_day fgf1_comparison bl6_day bl6_comparison\n",
       "1 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "2 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "3 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "4 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "5 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "6 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "  grouping      cells_a cells_b ⋯ gs_subcat     \n",
       "1 fgf1_grouping neg     none    ⋯ GO:MF         \n",
       "2 fgf1_grouping neg     rest    ⋯ CGP           \n",
       "3 fgf1_grouping neg     rest    ⋯ CGP           \n",
       "4 fgf1_grouping neg     rest    ⋯ CGP           \n",
       "5 fgf1_grouping neg     rest    ⋯ MIR:MIR_Legacy\n",
       "6 fgf1_grouping neg     rest    ⋯ TFT:GTRD      \n",
       "  pathway                                \n",
       "1 GOMF_PROTEIN_CONTAINING_COMPLEX_BINDING\n",
       "2 BENPORATH_PRC2_TARGETS                 \n",
       "3 BENPORATH_SUZ12_TARGETS                \n",
       "4 WONG_ADULT_TISSUE_STEM_MODULE          \n",
       "5 ACTGTGA_MIR27A_MIR27B                  \n",
       "6 MAFG_TARGET_GENES                      \n",
       "  gs_description                                                                                                                                                                                                                                                                \n",
       "1 Binding to a macromolecular complex. [GOC:jl]                                                                                                                                                                                                                                 \n",
       "2 Set 'PRC2 targets': Polycomb Repression Complex 2 (PRC) targets; identified by ChIP on chip on human embryonic stem cells as genes that: posess the trimethylated H3K27 mark in their promoters and are bound by SUZ12 [GeneID=23512] and EED [GeneID=8726] Polycomb proteins.\n",
       "3 Set 'Suz12 targets': genes identified by ChIP on chip as targets of the Polycomb protein SUZ12 [GeneID=23512] in human embryonic stem cells.                                                                                                                                  \n",
       "4 The 'adult tissue stem' module: genes coordinately up-regulated in a compendium of adult tissue stem cells.                                                                                                                                                                   \n",
       "5 Genes having at least one occurence of the motif ACTGTGA in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNAs hsa-miR-27a and hsa-miR-27b (v7.1 miRBase).                                                       \n",
       "6 Genes containing one or more binding sites for UniProt:O15525 (MAFG) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.                                                                                                \n",
       "  pval        padj       log2err   ES         NES       size\n",
       "1 0.006235055 0.06235055 0.4070179 -0.8899058 -1.593066 16  \n",
       "2 0.010384937 0.09078952 0.3807304  0.7888536  1.504499 16  \n",
       "3 0.002856450 0.07426771 0.4317077  0.7565117  1.609581 26  \n",
       "4 0.010475714 0.09078952 0.3807304  0.7654812  1.532718 19  \n",
       "5 0.006172517 0.01851755 0.4070179  0.7914666  1.517202 17  \n",
       "6 0.002768658 0.09136570 0.4317077 -0.8908226 -1.619861 17  \n",
       "  leadingEdge                                                                                                                                                                       \n",
       "1 ENSMUSG00000063063, ENSMUSG00000029381, ENSMUSG00000015354                                                                                                                        \n",
       "2 ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000035131                                                            \n",
       "3 ENSMUSG00000029088, ENSMUSG00000038872, ENSMUSG00000033981, ENSMUSG00000056258, ENSMUSG00000066687, ENSMUSG00000020709, ENSMUSG00000031137, ENSMUSG00000020704, ENSMUSG00000035131\n",
       "4 ENSMUSG00000032076, ENSMUSG00000033981, ENSMUSG00000019899, ENSMUSG00000048332, ENSMUSG00000024304                                                                                \n",
       "5 ENSMUSG00000032076, ENSMUSG00000025932, ENSMUSG00000036019, ENSMUSG00000020122                                                                                                    \n",
       "6 ENSMUSG00000063063, ENSMUSG00000015354, ENSMUSG00000021136                                                                                                                        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "combined_gsea_seurat_formatted = qs::qread('_targets/objects/combined_gsea_seurat_formatted')\n",
    "split_data_to_excel(combined_gsea_seurat_formatted, \"outputs/gsea_seurat\")\n",
    "combined_gsea_seurat_formatted %>% write_csv('combined_gsea_seurat_formatted')\n",
    "combined_gsea_seurat_formatted %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "40404a35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 27</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>tag</th><th scope=col>data_day</th><th scope=col>cluster</th><th scope=col>fgf1_day</th><th scope=col>fgf1_comparison</th><th scope=col>bl6_day</th><th scope=col>bl6_comparison</th><th scope=col>grouping</th><th scope=col>cells_a</th><th scope=col>cells_b</th><th scope=col>⋯</th><th scope=col>recall</th><th scope=col>term_id</th><th scope=col>source</th><th scope=col>term_name</th><th scope=col>effective_domain_size</th><th scope=col>source_order</th><th scope=col>parents</th><th scope=col>evidence_codes</th><th scope=col>intersection</th><th scope=col>gene_ids</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>⋯</td><td>0.012648221</td><td>GO:0007399</td><td>GO:BP</td><td>nervous system development        </td><td>27334</td><td> 2985</td><td>GO:0048731</td><td>IEA,IDA IMP IEA,IGI IBA,IMP IGI,IDA,IMP,IMP IBA,IMP ISO IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IGI IEA,IMP,ISO,IMP IGI IBA,IEA,IBA,IDA IMP IGI ISO IBA IEA,ISO,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                                                                                                                                              </td><td>ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                          </td><td>Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Ptpro,Egfr,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                                                              </td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>⋯</td><td>0.009796718</td><td>GO:0048731</td><td>GO:BP</td><td>system development                </td><td>27334</td><td>13811</td><td>GO:0007275, GO:0048856</td><td>IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA,IMP IGI,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                               </td><td>ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                  </td><td>Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                 </td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>⋯</td><td>0.008471074</td><td>GO:0007275</td><td>GO:BP</td><td>multicellular organism development</td><td>27334</td><td> 2887</td><td>GO:0032501, GO:0048856</td><td>IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IEA,IDA IMP IGI IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                               </td><td>ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                               </td><td>Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                            </td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>⋯</td><td>0.007100210</td><td>GO:0048856</td><td>GO:BP</td><td>anatomical structure development  </td><td>27334</td><td>13923</td><td>GO:0032502</td><td>IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP,IBA,IMP IBA,IMP,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                    </td><td>ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                      </td><td>Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2           </td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>⋯</td><td>0.006762717</td><td>GO:0032502</td><td>GO:BP</td><td>developmental process             </td><td>27334</td><td> 8073</td><td>GO:0008150</td><td>IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IEA,IDA IBA IEA,ISS ISO IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISS ISO IEA,ISO,IMP IGI IBA IEA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP ISO IEA,IBA,IMP IBA,IMP,IDA IEA,IGI IEA,IMP IGI IEA,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA</td><td>ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000063458,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000035877,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063</td><td>Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Lrmda,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Zhx3,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2</td></tr>\n",
       "\t<tr><td>a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none</td><td>all</td><td>a1_Tany__Astrocytes</td><td>all</td><td>obob5v5</td><td>all</td><td>obob14vBL6</td><td>fgf1_grouping</td><td>neg</td><td>none</td><td>⋯</td><td>0.009389671</td><td>GO:0009653</td><td>GO:BP</td><td>anatomical structure morphogenesis</td><td>27334</td><td> 3664</td><td>GO:0032502, GO:0048856</td><td>IMP,IMP,IGI IBA,IDA IMP IGI IEA,IMP ISO IEA,IEA,IMP IBA,IMP IGI ISO IEA,IEA,IMP,IMP,IBA,IMP IGI ISO IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IEA,IMP,IBA,IEA,IMP IGI,IMP IGI IBA,ISS ISO IBA IEA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IEA,IMP IEA                                                                                                                                                                                                                                                       </td><td>ENSMUSG00000020354,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000004328,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000024304,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000032186,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000029765,ENSMUSG00000053702,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                                                                                                                                            </td><td>Sgcd,Cadm1,Lama2,Eya1,Zbtb16,Hif3a,Ptpro,Egfr,Gpld1,Eya4,Fgf13,Cdh22,Cdh2,Ptch1,Fmnl2,Map2,Slc1a3,Tmod2,Nrxn3,Mecom,Plxna4,Nebl,Nrp1,Cpe,Shroom3,Ctnna2                                                                                                                     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 27\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " tag & data\\_day & cluster & fgf1\\_day & fgf1\\_comparison & bl6\\_day & bl6\\_comparison & grouping & cells\\_a & cells\\_b & ⋯ & recall & term\\_id & source & term\\_name & effective\\_domain\\_size & source\\_order & parents & evidence\\_codes & intersection & gene\\_ids\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <dbl> & <chr> & <chr> & <chr> & <int> & <int> & <list> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & ⋯ & 0.012648221 & GO:0007399 & GO:BP & nervous system development         & 27334 &  2985 & GO:0048731 & IEA,IDA IMP IEA,IGI IBA,IMP IGI,IDA,IMP,IMP IBA,IMP ISO IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IGI IEA,IMP,ISO,IMP IGI IBA,IEA,IBA,IDA IMP IGI ISO IBA IEA,ISO,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                                                                                                                                               & ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                           & Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Ptpro,Egfr,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                                                              \\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & ⋯ & 0.009796718 & GO:0048731 & GO:BP & system development                 & 27334 & 13811 & GO:0007275, GO:0048856 & IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA,IMP IGI,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                                & ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                   & Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                 \\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & ⋯ & 0.008471074 & GO:0007275 & GO:BP & multicellular organism development & 27334 &  2887 & GO:0032501, GO:0048856 & IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IEA,IDA IMP IGI IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                & ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                & Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                            \\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & ⋯ & 0.007100210 & GO:0048856 & GO:BP & anatomical structure development   & 27334 & 13923 & GO:0032502 & IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP,IBA,IMP IBA,IMP,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                     & ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                       & Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2           \\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & ⋯ & 0.006762717 & GO:0032502 & GO:BP & developmental process              & 27334 &  8073 & GO:0008150 & IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IEA,IDA IBA IEA,ISS ISO IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISS ISO IEA,ISO,IMP IGI IBA IEA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP ISO IEA,IBA,IMP IBA,IMP,IDA IEA,IGI IEA,IMP IGI IEA,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA & ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000063458,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000035877,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063 & Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Lrmda,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Zhx3,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2\\\\\n",
       "\t a1\\_Tany\\_\\_Astrocytes\\_\\_\\_all.obob5v5\\_\\_v\\_\\_all.obob14vBL6\\_\\_\\_fgf1\\_grouping.neg\\_vs\\_none & all & a1\\_Tany\\_\\_Astrocytes & all & obob5v5 & all & obob14vBL6 & fgf1\\_grouping & neg & none & ⋯ & 0.009389671 & GO:0009653 & GO:BP & anatomical structure morphogenesis & 27334 &  3664 & GO:0032502, GO:0048856 & IMP,IMP,IGI IBA,IDA IMP IGI IEA,IMP ISO IEA,IEA,IMP IBA,IMP IGI ISO IEA,IEA,IMP,IMP,IBA,IMP IGI ISO IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IEA,IMP,IBA,IEA,IMP IGI,IMP IGI IBA,ISS ISO IBA IEA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IEA,IMP IEA                                                                                                                                                                                                                                                        & ENSMUSG00000020354,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000004328,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000024304,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000032186,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000029765,ENSMUSG00000053702,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                                                                                                                                             & Sgcd,Cadm1,Lama2,Eya1,Zbtb16,Hif3a,Ptpro,Egfr,Gpld1,Eya4,Fgf13,Cdh22,Cdh2,Ptch1,Fmnl2,Map2,Slc1a3,Tmod2,Nrxn3,Mecom,Plxna4,Nebl,Nrp1,Cpe,Shroom3,Ctnna2                                                                                                                     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 27\n",
       "\n",
       "| tag &lt;chr&gt; | data_day &lt;chr&gt; | cluster &lt;chr&gt; | fgf1_day &lt;chr&gt; | fgf1_comparison &lt;chr&gt; | bl6_day &lt;chr&gt; | bl6_comparison &lt;chr&gt; | grouping &lt;chr&gt; | cells_a &lt;chr&gt; | cells_b &lt;chr&gt; | ⋯ ⋯ | recall &lt;dbl&gt; | term_id &lt;chr&gt; | source &lt;chr&gt; | term_name &lt;chr&gt; | effective_domain_size &lt;int&gt; | source_order &lt;int&gt; | parents &lt;list&gt; | evidence_codes &lt;chr&gt; | intersection &lt;chr&gt; | gene_ids &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | ⋯ | 0.012648221 | GO:0007399 | GO:BP | nervous system development         | 27334 |  2985 | GO:0048731 | IEA,IDA IMP IEA,IGI IBA,IMP IGI,IDA,IMP,IMP IBA,IMP ISO IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IGI IEA,IMP,ISO,IMP IGI IBA,IEA,IBA,IDA IMP IGI ISO IBA IEA,ISO,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                                                                                                                                               | ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                           | Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Ptpro,Egfr,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                                                               |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | ⋯ | 0.009796718 | GO:0048731 | GO:BP | system development                 | 27334 | 13811 | GO:0007275, GO:0048856 | IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA,IMP IGI,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                                | ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                   | Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                  |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | ⋯ | 0.008471074 | GO:0007275 | GO:BP | multicellular organism development | 27334 |  2887 | GO:0032501, GO:0048856 | IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IEA,IDA IMP IGI IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                | ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                | Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                             |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | ⋯ | 0.007100210 | GO:0048856 | GO:BP | anatomical structure development   | 27334 | 13923 | GO:0032502 | IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP,IBA,IMP IBA,IMP,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                     | ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                       | Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2            |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | ⋯ | 0.006762717 | GO:0032502 | GO:BP | developmental process              | 27334 |  8073 | GO:0008150 | IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IEA,IDA IBA IEA,ISS ISO IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISS ISO IEA,ISO,IMP IGI IBA IEA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP ISO IEA,IBA,IMP IBA,IMP,IDA IEA,IGI IEA,IMP IGI IEA,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA | ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000063458,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000035877,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063 | Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Lrmda,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Zhx3,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2 |\n",
       "| a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none | all | a1_Tany__Astrocytes | all | obob5v5 | all | obob14vBL6 | fgf1_grouping | neg | none | ⋯ | 0.009389671 | GO:0009653 | GO:BP | anatomical structure morphogenesis | 27334 |  3664 | GO:0032502, GO:0048856 | IMP,IMP,IGI IBA,IDA IMP IGI IEA,IMP ISO IEA,IEA,IMP IBA,IMP IGI ISO IEA,IEA,IMP,IMP,IBA,IMP IGI ISO IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IEA,IMP,IBA,IEA,IMP IGI,IMP IGI IBA,ISS ISO IBA IEA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IEA,IMP IEA                                                                                                                                                                                                                                                        | ENSMUSG00000020354,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000004328,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000024304,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000032186,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000029765,ENSMUSG00000053702,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                                                                                                                                             | Sgcd,Cadm1,Lama2,Eya1,Zbtb16,Hif3a,Ptpro,Egfr,Gpld1,Eya4,Fgf13,Cdh22,Cdh2,Ptch1,Fmnl2,Map2,Slc1a3,Tmod2,Nrxn3,Mecom,Plxna4,Nebl,Nrp1,Cpe,Shroom3,Ctnna2                                                                                                                      |\n",
       "\n"
      ],
      "text/plain": [
       "  tag                                                                             \n",
       "1 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "2 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "3 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "4 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "5 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "6 a1_Tany__Astrocytes___all.obob5v5__v__all.obob14vBL6___fgf1_grouping.neg_vs_none\n",
       "  data_day cluster             fgf1_day fgf1_comparison bl6_day bl6_comparison\n",
       "1 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "2 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "3 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "4 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "5 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "6 all      a1_Tany__Astrocytes all      obob5v5         all     obob14vBL6    \n",
       "  grouping      cells_a cells_b ⋯ recall      term_id    source\n",
       "1 fgf1_grouping neg     none    ⋯ 0.012648221 GO:0007399 GO:BP \n",
       "2 fgf1_grouping neg     none    ⋯ 0.009796718 GO:0048731 GO:BP \n",
       "3 fgf1_grouping neg     none    ⋯ 0.008471074 GO:0007275 GO:BP \n",
       "4 fgf1_grouping neg     none    ⋯ 0.007100210 GO:0048856 GO:BP \n",
       "5 fgf1_grouping neg     none    ⋯ 0.006762717 GO:0032502 GO:BP \n",
       "6 fgf1_grouping neg     none    ⋯ 0.009389671 GO:0009653 GO:BP \n",
       "  term_name                          effective_domain_size source_order\n",
       "1 nervous system development         27334                  2985       \n",
       "2 system development                 27334                 13811       \n",
       "3 multicellular organism development 27334                  2887       \n",
       "4 anatomical structure development   27334                 13923       \n",
       "5 developmental process              27334                  8073       \n",
       "6 anatomical structure morphogenesis 27334                  3664       \n",
       "  parents               \n",
       "1 GO:0048731            \n",
       "2 GO:0007275, GO:0048856\n",
       "3 GO:0032501, GO:0048856\n",
       "4 GO:0032502            \n",
       "5 GO:0008150            \n",
       "6 GO:0032502, GO:0048856\n",
       "  evidence_codes                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "1 IEA,IDA IMP IEA,IGI IBA,IMP IGI,IDA,IMP,IMP IBA,IMP ISO IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IGI IEA,IMP,ISO,IMP IGI IBA,IEA,IBA,IDA IMP IGI ISO IBA IEA,ISO,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                                                                                                                                              \n",
       "2 IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA,IMP IGI,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                                               \n",
       "3 IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IEA,IDA IMP IGI IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP,IMP IEA,IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,IMP IGI ISO IBA IEA,IMP,IMP,IMP IBA,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                                                               \n",
       "4 IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IEA,IEA,IDA IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISO,IMP IGI IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP,IBA,IMP IBA,IMP,IEA,IGI IEA,IMP IGI,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA                                                    \n",
       "5 IMP IBA,IEA,IDA IMP ISO IEA,IGI IBA IC IEA,IDA IMP IGI IBA IEA,IDA IMP ISS ISO IEA,IMP,IEA,IBA NAS IEA,IEA,IMP ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IEA,IDA IBA IEA,ISS ISO IBA IEA,IMP IBA IEA,IMP IEA,IBA IEA,IDA IMP IBA,IDA IMP IGI ISO IBA,IGI,ISS ISO IEA,ISO,IMP IGI IBA IEA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IBA IEA,IMP,IMP ISO IEA,IBA,IMP IBA,IMP,IDA IEA,IGI IEA,IMP IGI IEA,ISO,IMP IGI IBA,IEA,IEA,ISS ISO IBA IEA,IBA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IMP IEA,IEA,IMP ISS ISO IEA\n",
       "6 IMP,IMP,IGI IBA,IDA IMP IGI IEA,IMP ISO IEA,IEA,IMP IBA,IMP IGI ISO IEA,IEA,IMP,IMP,IBA,IMP IGI ISO IBA,IDA IMP IGI ISO IEA,ISS ISO IBA IEA,IMP IGI ISO IEA,IMP,IBA,IEA,IMP IGI,IMP IGI IBA,ISS ISO IBA IEA,IDA IMP IGI ISO IBA IC IEA,ISO IEA,IEA,IMP IEA                                                                                                                                                                                                                                                       \n",
       "  intersection                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
       "1 ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                          \n",
       "2 ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                  \n",
       "3 ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000040957,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                                                                               \n",
       "4 ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063                                      \n",
       "5 ENSMUSG00000020354,ENSMUSG00000038872,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000020704,ENSMUSG00000004328,ENSMUSG00000020709,ENSMUSG00000022419,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000035131,ENSMUSG00000063458,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000034912,ENSMUSG00000024304,ENSMUSG00000037784,ENSMUSG00000035877,ENSMUSG00000075415,ENSMUSG00000014361,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000017548,ENSMUSG00000032186,ENSMUSG00000040957,ENSMUSG00000023802,ENSMUSG00000030111,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000059974,ENSMUSG00000029765,ENSMUSG00000022762,ENSMUSG00000021136,ENSMUSG00000053702,ENSMUSG00000064115,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000024076,ENSMUSG00000029381,ENSMUSG00000063063\n",
       "6 ENSMUSG00000020354,ENSMUSG00000032076,ENSMUSG00000019899,ENSMUSG00000025932,ENSMUSG00000066687,ENSMUSG00000004328,ENSMUSG00000030223,ENSMUSG00000020122,ENSMUSG00000021340,ENSMUSG00000010461,ENSMUSG00000031137,ENSMUSG00000053166,ENSMUSG00000024304,ENSMUSG00000021466,ENSMUSG00000036053,ENSMUSG00000015222,ENSMUSG00000005360,ENSMUSG00000032186,ENSMUSG00000066392,ENSMUSG00000027684,ENSMUSG00000029765,ENSMUSG00000053702,ENSMUSG00000025810,ENSMUSG00000037852,ENSMUSG00000029381,ENSMUSG00000063063                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "  gene_ids                                                                                                                                                                                                                                                                    \n",
       "1 Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Ptpro,Egfr,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                                                              \n",
       "2 Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                                 \n",
       "3 Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Map2,Slc1a3,Suz12,Cables1,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2                            \n",
       "4 Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2           \n",
       "5 Sgcd,Zfhx3,Cadm1,Lama2,Eya1,Zbtb16,Asic2,Hif3a,Adap2,Deptor,Ptpro,Egfr,Gpld1,Brinp3,Lrmda,Eya4,Fgf13,Cdh22,Mdga2,Cdh2,Dzip1l,Zhx3,Fnbp1,Mertk,Ptch1,Fmnl2,Map2,Slc1a3,Suz12,Tmod2,Cables1,Nox3,A2m,Nrxn3,Mecom,Ntm,Plxna4,Ncam2,Smoc1,Nebl,Cadm2,Nrp1,Cpe,Vit,Shroom3,Ctnna2\n",
       "6 Sgcd,Cadm1,Lama2,Eya1,Zbtb16,Hif3a,Ptpro,Egfr,Gpld1,Eya4,Fgf13,Cdh22,Cdh2,Ptch1,Fmnl2,Map2,Slc1a3,Tmod2,Nrxn3,Mecom,Plxna4,Nebl,Nrp1,Cpe,Shroom3,Ctnna2                                                                                                                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "combined_gost_seurat_formatted = qs::qread('_targets/objects/combined_gost_seurat_formatted')\n",
    "combined_gost_seurat_formatted %>% write_csv('combined_gost_seurat_formatted.csv')\n",
    "split_data_to_excel(combined_gost_seurat_formatted, \"outputs/gprofiler_seurat\")\n",
    "combined_gost_seurat_formatted %>% head\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "054ba9c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "input_dir <- \"outputs/plots/deg_summary/\"\n",
    "output_dir <- \"outputs/plots_seurat/\"\n",
    "\n",
    "# create the output directory if it does not exist\n",
    "if (!dir.exists(output_dir)) {\n",
    "  dir.create(output_dir, recursive = TRUE)\n",
    "}\n",
    "\n",
    "# define the regular expression to extract the relevant parts of the tag\n",
    "tag_regex <- \"___(\\\\w+)\\\\.?(\\\\w+)?__v__(\\\\w+)\\\\.?(\\\\w+)?___\"\n",
    "\n",
    "# loop through the PDF files in the input directory and copy them to the output directory with the new subfolder structure\n",
    "pdf_files <- list.files(input_dir, pattern = \"*.pdf\", full.names = TRUE)\n",
    "for (pdf_file in pdf_files) {\n",
    "  file_name <- basename(pdf_file)\n",
    "  tag_match <- str_match(file_name, tag_regex)\n",
    "  fgf1_day <- tag_match[[2]]\n",
    "  fgf1_comparison <- tag_match[[3]]\n",
    "  bl6_day <- tag_match[[4]]\n",
    "  bl6_comparison <- tag_match[[5]]\n",
    "  subfolder_name <- paste0(fgf1_day, \".\", fgf1_comparison, \"__v__\", bl6_day, \".\", bl6_comparison)\n",
    "  subfolder_path <- file.path(output_dir, subfolder_name)\n",
    "  if (!dir.exists(subfolder_path)) {\n",
    "    dir.create(subfolder_path, recursive = TRUE)\n",
    "  }\n",
    "  new_file_path <- file.path(subfolder_path, file_name)\n",
    "  file.copy(pdf_file, new_file_path)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d58019d2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "40b910e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(\"outputs/restored_summaries\", recursive = TRUE):\n",
      "“'outputs/restored_summaries' already exists”\n"
     ]
    }
   ],
   "source": [
    "all_restored_summary = qs::qread('_targets/objects/all_restored_summary')\n",
    "# Split the tibble by the 'comparison' column\n",
    "split_data <- split(all_restored_summary, all_restored_summary$comparison)\n",
    "\n",
    "dir.create(\"outputs/restored_summaries\", recursive = TRUE)\n",
    "\n",
    "# Write out a file for each split\n",
    "walk(split_data, ~ {\n",
    "  comparison_name <- unique(.x$comparison)\n",
    "  file_name <- paste0(\"outputs/restored_summaries/restored_summary_\", comparison_name, \".xlsx\")\n",
    "  write_xlsx(.x, file_name)\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b549eb59",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b5f76b40",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(output_dir, recursive = TRUE):\n",
      "“'outputs/milo_DAbeeswarms' already exists”\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'labels'. You can override using the\n",
      "`.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "process_combined_da_results = function(da_results_all){\n",
    "    da_summary = da_results_all %>%\n",
    "    group_by(labels, polarity) %>%\n",
    "    summarise(n = n()) %>%\n",
    "    arrange(desc(labels))\n",
    "\n",
    "    da_label_groups = da_results_all %>%\n",
    "    group_by(labels) %>%\n",
    "    summarise(n_group = n()) %>%\n",
    "    arrange(desc(labels))\n",
    "\n",
    "    da_summary = merge(da_summary, da_label_groups)\n",
    "\n",
    "    da_summary = da_summary %>% mutate(pct_group = n/n_group)\n",
    "\n",
    "    da_summary_sum = da_summary %>% \n",
    "      filter(polarity == 'none') %>%\n",
    "      mutate(pct_group_changed = (n_group - n)/n_group) %>%\n",
    "      select(-c(polarity, pct_group, n)) %>%\n",
    "      arrange(desc(pct_group_changed), desc(n_group))\n",
    "\n",
    "    da_results_all$labels = da_summary_sum %>% arrange(pct_group_changed, n_group) %>% pull(labels) %>% factor(da_results_all$labels, levels=.)\n",
    "    da_results_all\n",
    "}\n",
    "\n",
    "input_dir <- \"_targets/objects\"\n",
    "output_dir <- \"outputs/milo_DAbeeswarms/\"\n",
    "dir.create(output_dir, recursive = TRUE)\n",
    "da_results_files <- list.files(input_dir, pattern = \"combined_da_results\", full.names = TRUE)\n",
    "for (da_results_file in da_results_files) {\n",
    "    file_name = basename(da_results_file)\n",
    "    da_results = qs::qread(da_results_file) %>% process_combined_da_results\n",
    "    my_plot = miloR::plotDAbeeswarm(da_results, group.by='labels')\n",
    "    plot_path = paste0(output_dir, \"DAbeeswarm_\", file_name, '.pdf')\n",
    "    ggsave(plot_path, plot = my_plot, width = 10, height = 30)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dcdb8e64",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98a9af0a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:fgf1]",
   "language": "R",
   "name": "conda-env-fgf1-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
